Description: Fixes a test failure with Java 8
Author: Emmanuel Bourg <ebourg@apache.org>
Bug: https://hibernate.atlassian.net/browse/HCANN-66
Bug-Debian: https://bugs.debian.org/750754
--- a/src/test/java/org/hibernate/annotations/common/test/annotationfactory/TestAnnotation.java
+++ b/src/test/java/org/hibernate/annotations/common/test/annotationfactory/TestAnnotation.java
@@ -7,7 +7,7 @@
  * @author Paolo Perrotta
  * @author Davide Marchignoli
  */
-@interface TestAnnotation {
+public @interface TestAnnotation {
 	String stringElement();
 
 	String elementWithDefault() default "abc";
