diff --git a/jni/GNUmakefile b/jni/GNUmakefile
index 27737be..fc0ef01 100755
--- a/jni/GNUmakefile
+++ b/jni/GNUmakefile
@@ -251,8 +251,8 @@ endif
 ifneq ($(strip $(findstring $(OS), solaris linux)),)
   # Change the CC/LD instead of CFLAGS/LDFLAGS, incase other things in the flags
   # makes the libffi build choke
-  CC += -m$(MODEL)
-  LD += -m$(MODEL)
+  #CC += -m$(MODEL)
+  #LD += -m$(MODEL)
 endif
 
 LIBJFFI = $(BUILD_DIR)/$(PREFIX)$(LIBNAME)-$(VERSION).$(JNIEXT)
diff --git a/libtest/GNUmakefile b/libtest/GNUmakefile
index 9e70664..07c3950 100644
--- a/libtest/GNUmakefile
+++ b/libtest/GNUmakefile
@@ -177,8 +177,8 @@ endif
 ifneq ($(strip $(findstring $(OS), solaris)),)
   # Change the CC/LD instead of CFLAGS/LDFLAGS, incase other things in the flags
   # makes the libffi build choke
-  CC += -m$(MODEL)
-  LD += -m$(MODEL)
+  #CC += -m$(MODEL)
+  #LD += -m$(MODEL)
 endif
 
 LIBTEST = $(BUILD_DIR)/$(LIBNAME)
