--- a/libtest/GNUmakefile
+++ b/libtest/GNUmakefile
@@ -45,7 +45,7 @@
 #   http://weblogs.java.net/blog/kellyohair/archive/2006/01/compilation_of_1.html
 JFLAGS = -fno-omit-frame-pointer -fno-strict-aliasing
 OFLAGS = -O2 $(JFLAGS)
-WFLAGS = -W -Werror -Wall -Wno-unused -Wno-parentheses
+WFLAGS = -W -Wall -Wno-unused -Wno-parentheses
 PICFLAGS = -fPIC
 SOFLAGS = -shared -Wl,-O1
 LDFLAGS += $(SOFLAGS)
