Description: Use import syntax
Author: Xavier Guimard <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2019-07-17

--- a/dist/index.js
+++ b/dist/index.js
@@ -2,7 +2,8 @@
 'use strict';
 
 
-var Punycode = require('punycode');
+//var Punycode = require('punycode');
+import Punycode from 'punycode';
 
 
 var internals = {};
@@ -11,7 +12,8 @@
 //
 // Read rules from file.
 //
-internals.rules = require('./data/rules.json').map(function (rule) {
+import rules from '../data/rules.json';
+internals.rules = rules.map(function (rule) {
 
   return {
     rule: rule,
