Description: Ignore the bintray plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -13,7 +13,6 @@
 apply plugin: 'eclipse'
 apply plugin: 'groovy'
 apply plugin: 'maven'
-apply plugin: 'com.jfrog.bintray'
 
 apply plugin: 'signing'
 
@@ -119,6 +118,7 @@
   }
 }
 
+/*
 bintray {
     user = System.getProperty('bintrayUser')
     key = System.getProperty('bintrayApiKey')
@@ -156,3 +156,4 @@
     }
 
 }
+*/
