From 1d04a1444a7f66cf55d83d182d04eee27ccccaa7 Mon Sep 17 00:00:00 2001
From: Weng Xuetian <wengxt@gmail.com>
Date: Wed, 5 Aug 2020 10:39:55 -0700
Subject: [PATCH] bump version.

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index c6c1a8b0..d8f00e6d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -2,7 +2,7 @@ cmake_minimum_required(VERSION 3.6)
 
 project(fcitx)
 
-set(version 4.2.9.7)
+set(version 4.2.9.8)
 
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED TRUE)
-- 
2.30.0

