--- a/doc/source/user/formats.rst
+++ b/doc/source/user/formats.rst
@@ -99,6 +99,10 @@
   This indicates what is stored in Glance is an Amazon Machine Image (AMI).
   It is a virtual machine image in raw format.
 
+* **root**
+
+  This indicates what is store in Glance is an LXD image
+
 Container Format
 ----------------
 
--- a/doc/source/user/glanceapi.rst
+++ b/doc/source/user/glanceapi.rst
@@ -516,7 +516,7 @@
 
   This header is required, unless reserving an image. Valid values are one of
   ``aki``, ``ari``, ``ami``, ``raw``, ``iso``, ``vhd``, ``vhdx``, ``vdi``,
-  ``qcow2``, ``vmdk`` or ``ploop``.
+  ``qcow2``, ``vmdk`` ``ploop`` or ``root-tar``.
 
   For more information, see :ref:`formats`.
 
--- a/etc/glance-api.conf
+++ b/etc/glance-api.conf
@@ -4846,7 +4846,7 @@
 
 # Supported values for the 'disk_format' image attribute (list value)
 # Deprecated group/name - [DEFAULT]/disk_formats
-#disk_formats = ami,ari,aki,vhd,vhdx,vmdk,raw,qcow2,vdi,iso,ploop
+disk_formats = ami,ari,aki,vhd,vhdx,vmdk,raw,qcow2,vdi,iso,ploop.root-tar
 
 
 [key_manager]
--- a/glance/common/config.py
+++ b/glance/common/config.py
@@ -95,7 +95,7 @@
                                                    group='DEFAULT')]),
     cfg.ListOpt('disk_formats',
                 default=['ami', 'ari', 'aki', 'vhd', 'vhdx', 'vmdk', 'raw',
-                         'qcow2', 'vdi', 'iso', 'ploop'],
+                         'qcow2', 'vdi', 'iso', 'ploop', 'root-tar'],
                 help=_("Supported values for the 'disk_format' "
                        "image attribute"),
                 deprecated_opts=[cfg.DeprecatedOpt('disk_formats',
--- a/glance/tests/unit/v2/test_images_resource.py
+++ b/glance/tests/unit/v2/test_images_resource.py
@@ -5810,7 +5810,7 @@
     def test_default_disk_formats(self):
         schema = glance.api.v2.images.get_schema()
         expected = [None, 'ami', 'ari', 'aki', 'vhd', 'vhdx', 'vmdk',
-                    'raw', 'qcow2', 'vdi', 'iso', 'ploop']
+                    'raw', 'qcow2', 'vdi', 'iso', 'ploop', 'root-tar']
         actual = schema.properties['disk_format']['enum']
         self.assertEqual(expected, actual)
 
