Index: alsa-driver-1.0.25/alsa-kernel/pci/intel8x0.c
===================================================================
--- alsa-driver-1.0.25.orig/alsa-kernel/pci/intel8x0.c	2012-01-25 18:56:49.000000000 +1100
+++ alsa-driver-1.0.25/alsa-kernel/pci/intel8x0.c	2012-02-08 20:28:45.450539413 +1100
@@ -2653,6 +2653,13 @@
 	 * after S2RAM.  So, don't touch it.
 	 */
 	/* pci_set_power_state(pci, pci_choose_state(pci, state)); */
+
+	/* Quirk support for HP nw8240 and HP NC6220, LP #151111 */
+	if ((pci->subsystem_vendor == 0x103c) &&
+	    ((pci->subsystem_device == 0x0934) ||
+	    (pci->subsystem_device == 0x0944))) {
+		pci_set_power_state(pci, pci_choose_state(pci, state));
+	}
 	return 0;
 }
 
