From: Steve Kowalik <stevenk@debian.org>
Subject: Compile snd-pcm-oss.o with -fno-omit-frame-pointer.
Last-Update: 2010-05-23

Index: alsa-driver-1.0.23+dfsg/alsa-kernel/core/oss/Makefile
===================================================================
--- alsa-driver-1.0.23+dfsg.orig/alsa-kernel/core/oss/Makefile	2010-05-23 13:52:35.000000000 +0200
+++ alsa-driver-1.0.23+dfsg/alsa-kernel/core/oss/Makefile	2010-05-23 13:53:14.000000000 +0200
@@ -3,6 +3,8 @@
 # Copyright (c) 1999 by Jaroslav Kysela <perex@perex.cz>
 #
 
+EXTRA_CFLAGS += -fno-omit-frame-pointer
+
 snd-mixer-oss-objs := mixer_oss.o
 
 snd-pcm-oss-y := pcm_oss.o
