From: Thomas Hood <jdthood@yahoo.co.uk>
Subject: Disable GCC version check in configure.in
Last-Update: 2010-05-23

Index: alsa-driver-1.0.25/configure.in
===================================================================
--- alsa-driver-1.0.25.orig/configure.in	2012-02-08 20:17:08.414521572 +1100
+++ alsa-driver-1.0.25/configure.in	2012-02-08 20:17:10.454521623 +1100
@@ -290,6 +290,7 @@
 AC_MSG_RESULT($kaversion)
 
 dnl Check for GCC version...
+if false; then
 AC_MSG_CHECKING(for GCC version)
 ac_save_CFLAGS="$CFLAGS"
 CFLAGS="$CFLAGS $KERNEL_INC $HACK_KERNEL_INC"
@@ -339,6 +340,7 @@
 kernel_compiler="")
 CFLAGS="$ac_save_CFLAGS"
 my_compiler=`$CROSS_COMPILE$CC --version | head -n 1`
+fi
 dnl
 dnl Test section, uncomment any of these double lines to verify the bellow code:
 dnl
@@ -358,6 +360,7 @@
 dnl my_compiler="2.95.3"
 dnl
 
+if false; then
 AC_DEFUN([CHECK_COMPILER], [
   typevar=$2
   versionvar=$3
@@ -436,6 +439,7 @@
   fi
 fi
 fi
+fi
 dnl compiler version check
 
 AC_DEFUN([CHECK_KERNEL_HEADER], [
