From: Elimar Riesebieter <riesebie@lxtec.de>
Subject: Don't build non DFSG maestro3.c
Last-Update: 2010-05-23

Index: alsa-driver-1.0.25/pci/Makefile
===================================================================
--- alsa-driver-1.0.25.orig/pci/Makefile	2012-01-25 18:56:49.000000000 +1100
+++ alsa-driver-1.0.25/pci/Makefile	2012-02-08 20:17:01.294521387 +1100
@@ -10,7 +10,7 @@
 #
 
 clean-files := atiixp.c atiixp_modem.c bt87x.c cmipci.c ens1370.c \
-	fm801.c intel8x0.c maestro3.c via82xx.c via82xx_modem.c
+	fm801.c intel8x0.c via82xx.c via82xx_modem.c
 
 obj-$(CONFIG_SND) += pdplus/ asihpi/
 
@@ -30,6 +30,5 @@
 ens1371.c: ens1370.c
 fm801.c: fm801.patch $(SND_TOPDIR)/alsa-kernel/pci/fm801.c
 intel8x0.c: intel8x0.patch $(SND_TOPDIR)/alsa-kernel/pci/intel8x0.c
-maestro3.c: maestro3.patch $(SND_TOPDIR)/alsa-kernel/pci/maestro3.c
 via82xx.c: via82xx.patch $(SND_TOPDIR)/alsa-kernel/pci/via82xx.c
 via82xx_modem.c: via82xx_modem.patch $(SND_TOPDIR)/alsa-kernel/pci/via82xx_modem.c
