Description: Fancybox_browser_is_undefined
--- a/fancybox/jquery.fancybox.js
+++ b/fancybox/jquery.fancybox.js
@@ -26,8 +26,8 @@
 
 		titleHeight = 0, titleStr = '', start_pos, final_pos, busy = false, fx = $.extend($('<div/>')[0], { prop: 0 }),
 
-		isIE6 = $.browser.msie && $.browser.version < 7 && !window.XMLHttpRequest,
-
+//		isIE6 = $.browser.msie && $.browser.version < 7 && !window.XMLHttpRequest,
+  isIE6 = navigator.userAgent.match(/msie [6]/i) && !window.XMLHttpRequest,
 		/*
 		 * Private methods 
 		 */
@@ -612,9 +612,9 @@
 			}
 
 			if (currentOpts.type == 'iframe') {
-				$('<iframe id="fancybox-frame" name="fancybox-frame' + new Date().getTime() + '" frameborder="0" hspace="0" ' + ($.browser.msie ? 'allowtransparency="true""' : '') + ' scrolling="' + selectedOpts.scrolling + '" src="' + currentOpts.href + '"></iframe>').appendTo(content);
+//		$('<iframe id="fancybox-frame" name="fancybox-frame' + new Date().getTime() + '" frameborder="0" hspace="0" ' + ($.browser.msie ? 'allowtransparency="true""' : '') + ' scrolling="' + selectedOpts.scrolling + '" src="' + currentOpts.href + '"></iframe>').appendTo(content);
+   	$('<iframe id="fancybox-frame" name="fancybox-frame' + new Date().getTime() + '" frameborder="0" hspace="0" ' + (navigator.userAgent.match(/msie [6]/i) ? 'allowtransparency="true""' : '') + ' scrolling="' + selectedOpts.scrolling + '" src="' + currentOpts.href + '"></iframe>').appendTo(content);
 			}
-
 			wrap.show();
 
 			busy = false;
@@ -812,7 +812,7 @@
 					selectedArray.push(this);
 
 				} else {
-					selectedArray = $("a[rel=" + rel + "], area[rel=" + rel + "]");
+					selectedArray = $("a[rel=" + rel + "], area[rel=" + rel + "], img[rel=" + rel + "]");
 					selectedIndex = selectedArray.index( this );
 				}
 
