Description: Reduce iterations in ReDOS quadratic_lists test
Author: Graham Inggs <ginggs@debian.org>
Last-Update: 2022-04-18

--- a/test/specs/redos/quadratic_lists.cjs
+++ b/test/specs/redos/quadratic_lists.cjs
@@ -1,4 +1,4 @@
 module.exports = {
-  markdown: '- a\n'.repeat(10000),
-  html: `<ul>${'<li>a</li>'.repeat(10000)}</ul>`
+  markdown: '- a\n'.repeat(1000),
+  html: `<ul>${'<li>a</li>'.repeat(1000)}</ul>`
 };
