Description: make build reproducible
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Reviewed-By: Chris Lamb <lamby@debian.org>
Last-Update: 2022-01-12

--- a/rollup.config.esm.js
+++ b/rollup.config.esm.js
@@ -17,7 +17,7 @@
     license({
       banner: `
 marked - a markdown parser
-Copyright (c) 2011-${new Date().getFullYear()}, Christopher Jeffrey. (MIT Licensed)
++Copyright (c) 2011-${(new Date(process.env.SOURCE_DATE_EPOCH ? (process.env.SOURCE_DATE_EPOCH * 1000) : new Date().getTime())).getFullYear()}, Christopher Jeffrey. (MIT Licensed)
 https://github.com/markedjs/marked
 `
     }),
--- a/rollup.config.js
+++ b/rollup.config.js
@@ -19,7 +19,7 @@
     license({
       banner: `
 marked - a markdown parser
-Copyright (c) 2011-${new Date().getFullYear()}, Christopher Jeffrey. (MIT Licensed)
+Copyright (c) 2011-${(new Date(process.env.SOURCE_DATE_EPOCH ? (process.env.SOURCE_DATE_EPOCH * 1000) : new Date().getTime())).getFullYear()}, Christopher Jeffrey. (MIT Licensed)
 https://github.com/markedjs/marked
 `
     }),
@@ -46,7 +46,7 @@
     license({
       banner: `
 marked - a markdown parser
-Copyright (c) 2011-${new Date().getFullYear()}, Christopher Jeffrey. (MIT Licensed)
+Copyright (c) 2011-${(new Date(process.env.SOURCE_DATE_EPOCH ? (process.env.SOURCE_DATE_EPOCH * 1000) : new Date().getTime())).getFullYear()}, Christopher Jeffrey. (MIT Licensed)
 https://github.com/markedjs/marked
 `
     }),
