From: Balint Reczey <balint.reczey@canonical.com>
Date: Tue, 15 Jan 2019 14:11:06 +0700
Subject: Skip installing init.d script

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 5a1eb3a..548cacb 100644
--- a/setup.py
+++ b/setup.py
@@ -19,8 +19,8 @@ setup(name="ec2-hibinit-agent",
       author_email="anchalag@amazon.com",
       tests_require=["pytest"],
       scripts=['agent/hibinit-agent'],
-      data_files=[('/etc', ['etc/hibinit-config.cfg']),
-                  ('/etc/init.d', ['etc/init.d/hibinit-agent'])],
+      data_files=[('/etc', ['etc/hibinit-config.cfg'])],
+#                  ('/etc/init.d', ['etc/init.d/hibinit-agent'])],
       description="Hibernation setup for EC2 Instances",
       long_description=hib_long_description,
       license="Apache 2.0",
