From: Balint Reczey <balint.reczey@canonical.com>
Date: Tue, 15 Jan 2019 15:44:43 +0700
Subject: Use /swap-hibinit instead of too generic /swap

---
 agent/hibinit-agent | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/agent/hibinit-agent b/agent/hibinit-agent
index 9dc2f1d..5461776 100755
--- a/agent/hibinit-agent
+++ b/agent/hibinit-agent
@@ -40,7 +40,7 @@ GRUB2_DIR = '/etc/default/grub.d'
 SWAP_RESERVED_SIZE = 16384
 log_to_syslog = True
 log_to_stderr = True
-SWAP_FILE = '/swap'
+SWAP_FILE = '/swap-hibinit'
 URL = "http://169.254.169.254/latest/meta-data/hibernation/configured"
 
 def log(message):
