From: Balint Reczey <balint.reczey@canonical.com>
Date: Wed, 27 Feb 2019 16:53:29 +0100
Subject: Detect disabled hibernation with Python 3's urllib, too.

LP: #1817926
---
 agent/hibinit-agent | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/agent/hibinit-agent b/agent/hibinit-agent
index 5461776..e2cbed1 100755
--- a/agent/hibinit-agent
+++ b/agent/hibinit-agent
@@ -333,7 +333,7 @@ def hibernationEnabled():
      try:
          response = urlopen(URL)
          data = response.read()
-         if data.lower() == 'false':
+         if data.lower() in ('false', b'false'):
              return False
      except:
          return False
