--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -7,7 +7,13 @@ set(kgendesignerplugin_SRCS
   ${kdesignerplugin_QM_LOADER}
 )
 add_executable(kgendesignerplugin ${kgendesignerplugin_SRCS})
-add_executable(KF5::kgendesignerplugin ALIAS kgendesignerplugin)
+if(CMAKE_CROSSCOMPILING)
+    find_program(KGENDESIGNERPLUGIN_EXECUTABLE kgendesignerplugin)
+    add_executable(KF5::kgendesignerplugin IMPORTED GLOBAL)
+    set_target_properties(KF5::kgendesignerplugin PROPERTIES IMPORTED_LOCATION ${KGENDESIGNERPLUGIN_EXECUTABLE})
+else()
+    add_executable(KF5::kgendesignerplugin ALIAS kgendesignerplugin)
+endif()
 ecm_mark_nongui_executable(kgendesignerplugin)
 
 target_link_libraries(kgendesignerplugin KF5::ConfigCore # KConfig + KConfigGroup
