Description: spelling error in manpage
Last-Update: 2011-06-15
Origin: Fabrizio Regalli <fabreg@fabreg.it>
Forwarded: not-needed

--- a/lib/HTML/Clean.pm
+++ b/lib/HTML/Clean.pm
@@ -42,19 +42,19 @@
 
 =over 8
 
-=item Remove unneeded whitespace (begining of line, etc)
+=item Remove unneeded whitespace (beginning of line, etc)
 
 =item Remove unneeded META elements.
 
 =item Remove HTML comments (except for styles, javascript and SSI)
 
-=item Replace tags with equivilant shorter tags (<strong> --> <b>)
+=item Replace tags with equivalent shorter tags (<strong> --> <b>)
 
 =item etc.
 
 =back
 
-The entire proces is configurable, so you can pick and choose what you want
+The entire process is configurable, so you can pick and choose what you want
 to clean.
 
 =cut
