Description: Fixed double comment tag at end of javascript
Origin: vendor
Bug-Debian: https://bugs.debian.org/303619
Forwarded: no
Author: Gunnar Wolf <gwolf@debian.org>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2019-12-22

--- a/lib/HTML/Clean.pm
+++ b/lib/HTML/Clean.pm
@@ -213,7 +213,9 @@ sub _jscomments {
   # insure javascript is hidden
 
   if ($js =~ m,<!--,) {
-     $js =~ s,</script>,// -->\n</script>,si;
+      unless ($js =~ m,(-->[\s\r\n]*</script>),i) {
+        $js =~ s,</script>,// -->\n</script>,si;
+      }
   }
   return($js);
 }
