From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <marco@ubuntu.com>
Date: Wed, 16 Mar 2022 13:42:56 -0400
Subject: Use the Ubuntu Appearance panel to display settings in the Ubuntu
 session

Forwarded: not-needed
---
 desktopManager.js | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/desktopManager.js b/desktopManager.js
index 7f517da..e1add3c 100644
--- a/desktopManager.js
+++ b/desktopManager.js
@@ -919,7 +919,16 @@ var DesktopManager = class {
         this._menu.add(new Gtk.SeparatorMenuItem());
 
         this._settingsMenuItem = new Gtk.MenuItem({ label: _("Desktop Icons Settings") });
-        this._settingsMenuItem.connect("activate", () => Prefs.showPreferences());
+        if (GLib.getenv('XDG_CURRENT_DESKTOP').split(':').includes('ubuntu')) {
+            this._settingsMenuItem.connect("activate", () => {
+                const desktopFile = Gio.DesktopAppInfo.new('gnome-ubuntu-panel.desktop');
+                const context = Gdk.Display.get_default().get_app_launch_context();
+                context.set_timestamp(Gtk.get_current_event_time());
+                desktopFile.launch([], context);
+            });
+        } else {
+            this._settingsMenuItem.connect("activate", () => Prefs.showPreferences());
+        }
         this._menu.add(this._settingsMenuItem);
 
         this._displaySettingsMenuItem = new Gtk.MenuItem({label: _("Display Settings")});
