Description: update of documentation 
Author: Thorsten Alteholz <debian@alteholz.de>
Last-Update: 2011-09-28

--- feynmf-1.08.orig/Tutorial
+++ feynmf-1.08/Tutorial
@@ -1,20 +1,20 @@
 All three parts of a tuturial on FeynMF are now available from CERN,
 both as PostScript and HTML files:
 
-PostScript:
+PostScript: (as parts of three CERN Computer Newsletters)
 ***********
 
-   ftp://asisftp.cern.ch/cnl/220/textproc.ps.gz    [Part 1]
-   ftp://asisftp.cern.ch/cnl/221/textproc.ps.gz    [Part 2]
-   ftp://asisftp.cern.ch/cnl/222/textproc.ps.gz    [Part 3]
+   http://cnlart.web.cern.ch/cnlart/220/tex/cnl220.ps         [Part 1]
+   http://cnlart.web.cern.ch/cnlart/221/tex/fmfcnl2.ps        [Part 2]
+   http://cnlart.web.cern.ch/cnlart/222/tex/fmf/fmfcnl3.ps    [Part 3]
 
 
 HTML:
 *****
 
-   http://wwwcn1.cern.ch/cnls/220/node57.html      [Part 1]
-   http://wwwcn1.cern.ch/cnls/221/node63.html      [Part 2]
-   http://wwwcn1.cern.ch/cnls/222/node38.html      [Part 3]
+   http://cnlart.web.cern.ch/cnlart/220/node57.html      [Part 1]
+   http://cnlart.web.cern.ch/cnlart/221/node63.html      [Part 2]
+   http://cnlart.web.cern.ch/cnlart/222/node38.html      [Part 3]
 
 A paper has been printed by Computer Physics Communications:
 
--- feynmf-1.08.orig/feynmf.dtx
+++ feynmf-1.08/feynmf.dtx
@@ -587,7 +587,7 @@
 % The most complicated part of using the {\bf FeynMF} style appears to be the
 % proper invocation of {\bf Metafont}.  The {\bf feynmf} script provides a
 % convenient front end and will automagically invoke {\bf Metafont} with the
-% proper mode and magnifincation.  It will also avoid cluttering system font
+% proper mode and magnification.  It will also avoid cluttering system font
 % directories and offers an option to clean them.
 % 
 % \subsubsection*{OPTIONS}%
@@ -7696,6 +7696,7 @@ endinput;
 %<*!209>
 \documentclass[a4paper]{article}
 \usepackage{doc}
+\usepackage[T1]{fontenc}
 %    \end{macrocode}
 % The \MF{} and \MP{} logos come out much nicer if you have |mflogo|
 % installed:
--- feynmf-1.08.orig/feynmf.pl
+++ feynmf-1.08/feynmf.pl
@@ -37,7 +37,7 @@ B<feynmf> [B<--help>] [B<--version>] [B<
 The most complicated part of using the B<FeynMF> style appears to be the
 proper invocation of B<Metafont>.  The B<feynmf> script provides a
 convenient front end and will automagically invoke B<Metafont> with the
-proper mode and magnifincation.  It will also avoid cluttering system font
+proper mode and magnification.  It will also avoid cluttering system font
 directories and offers an option to clean them.
 
 =head1 OPTIONS
@@ -270,10 +270,11 @@ sub fake_tfms {
     my @tfm = @_;
     # Prepare a fake temporary PL file
     # (/dev/null won't do, because the font must not be empty):
-    my ($pl) = "/tmp/feynmf$$.pl";
+    my ($pl) = `/bin/mktemp -u --suffix=.pl /tmp/feynmfXXXXXX`;
     my ($tfm);
     $pltotf_prog
 	or die "feynmf: fatal: pltopf programm required unless -notfm\n";
+    chop ($pl);
     open (PL, ">$pl") or die "feynmf: can't open temporary file $pl: $!\n";
     push @temporay_files, $pl;
     print PL <<__END_PL__;
@@ -373,7 +374,7 @@ sub run_mf {
 			maybe_run "$gftopk_prog $gf" if $gftopk_prog;
 		    }
 		}
-		close (<LOG>);
+		close (LOG);
 	    }
 	}
     }
--- /dev/null
+++ feynmf-1.08/ChangeLog
@@ -0,0 +1,85 @@
+NEWS for Version 1.08
+
+    *  Improved very short gluon and photon propagators by relaxing
+    overly aggressive rounding.
+
+    *  New (still experimental) `feynmf' script automating the
+    invocation of Metafont (requires perl(1)).
+
+NEWS for Version 1.07
+
+    *  Switch off I/O during the first pass in AMS-LaTeX equation
+    environments.  This avoids a nasty race condition (thanks to Axel
+    Thimm for reporting this bug).  A beneficial side effect is that
+    it saves us 50% processing time in these cases.
+
+NEWS for Version 1.06
+
+    *  New abstraction: polygons.  Useful for complex vertices in
+    non-perturbative diagrams.
+
+    *  New commands: \fmfpoly, \fmfpolyn and \fmfrpolyn, implementing
+    polygons.
+
+    *  New option `rubout' for crossing propagators.
+
+    *  Some common typos are detected are appropriate warning messages
+    are issued.
+
+    *  New options `errorstop', `scroll', `nonstop' and `batch' for
+    specifying the desired interactionmode.
+
+    *  Work around an obscure babel/german/feynmf three way
+    destructive interference.
+
+NEWS for Version 1.05
+
+    *  FeynMF now coexists with french.sty and other packages that
+    change the \catcode of ``:''.
+
+    *  Fixed a bug introduced in version 1.03, which could cause
+    problems with font loading.  Unfortunately, the fix has to give up
+    some compatibility with earlier versions.  Sources for version
+    1.02 and earlier which used \noexpand in labels have to upgraded
+    or the option `pre-1.03' has to be used.  Sorry.
+
+NEWS for Version 1.04
+
+    *  Finally, \fmffixed{(dx,whatever)}{v1,v2} and
+    \fmffixed{(whatever,dy)}{v1,v2} do what you would expect them
+    to.  \fmffixedx and \fmffixedy are new synonyms for those.
+
+NEWS for Version 1.03
+
+    *  It is no longer necessary to protect TeX command sequences with
+    \noexpand.  Old files using \noexpand will continue to work.
+
+NEWS for Version 1.02
+
+    *  Read the .tfm file before writing the .mf file destructively.
+    This should allow a transparent two pass processing if MakeTeXTFM
+    is enabled.  Thanks to Thomas Esser <te@informatik.uni-hannover.de> 
+    for the suggestion.
+
+NEWS for Version 1.01
+
+    * new decoration shapes for vertices by popular demand: `cross'
+    (also `triacross', `pentacross' and `hexacross').
+
+    * fixed a rounding bug in cut-off ends (noticeable as a spurious
+    thin diagonal line at the left end of double lines).
+
+    * The maximum size for shaded and hatched areas has been doubled
+    by removing an obsolete speedup hack that was not used anymore.
+
+    * In an unrelated move, shading and hatching has been
+    reimplemented in a much more general fashion.  Now arbitray
+    tilings can be specified as argument to `decoration.filled='.  The
+    old style numerical arguments continue to work, but should be
+    replaced by `empty', `full', `shaded', and `hatched' in new
+    documents.
+
+########################################################################
+# Local Variables:
+# mode:indented-text
+# End:
--- feynmf-1.08.orig/README
+++ feynmf-1.08/README
@@ -112,89 +112,6 @@ WARNING:
     feynmf-announce', `unsubscribe feynmf-bugs', `help', etc.
 
 ########################################################################
-
-NEWS for Version 1.08
-
-    *  Improved very short gluon and photon propagators by relaxing
-    overly aggressive rounding.
-
-    *  New (still experimental) `feynmf' script automating the
-    invocation of Metafont (requires perl(1)).
-
-NEWS for Version 1.07
-
-    *  Switch off I/O during the first pass in AMS-LaTeX equation
-    environments.  This avoids a nasty race condition (thanks to Axel
-    Thimm for reporting this bug).  A beneficial side effect is that
-    it saves us 50% processing time in these cases.
-
-NEWS for Version 1.06
-
-    *  New abstraction: polygons.  Useful for complex vertices in
-    non-perturbative diagrams.
-
-    *  New commands: \fmfpoly, \fmfpolyn and \fmfrpolyn, implementing
-    polygons.
-
-    *  New option `rubout' for crossing propagators.
-
-    *  Some common typos are detected are appropriate warning messages
-    are issued.
-
-    *  New options `errorstop', `scroll', `nonstop' and `batch' for
-    specifying the desired interactionmode.
-
-    *  Work around an obscure babel/german/feynmf three way
-    destructive interference.
-
-NEWS for Version 1.05
-
-    *  FeynMF now coexists with french.sty and other packages that
-    change the \catcode of ``:''.
-
-    *  Fixed a bug introduced in version 1.03, which could cause
-    problems with font loading.  Unfortunately, the fix has to give up
-    some compatibility with earlier versions.  Sources for version
-    1.02 and earlier which used \noexpand in labels have to upgraded
-    or the option `pre-1.03' has to be used.  Sorry.
-
-NEWS for Version 1.04
-
-    *  Finally, \fmffixed{(dx,whatever)}{v1,v2} and
-    \fmffixed{(whatever,dy)}{v1,v2} do what you would expect them
-    to.  \fmffixedx and \fmffixedy are new synonyms for those.
-
-NEWS for Version 1.03
-
-    *  It is no longer necessary to protect TeX command sequences with
-    \noexpand.  Old files using \noexpand will continue to work.
-
-NEWS for Version 1.02
-
-    *  Read the .tfm file before writing the .mf file destructively.
-    This should allow a transparent two pass processing if MakeTeXTFM
-    is enabled.  Thanks to Thomas Esser <te@informatik.uni-hannover.de> 
-    for the suggestion.
-
-NEWS for Version 1.01
-
-    * new decoration shapes for vertices by popular demand: `cross'
-    (also `triacross', `pentacross' and `hexacross').
-
-    * fixed a rounding bug in cut-off ends (noticeable as a spurious
-    thin diagonal line at the left end of double lines).
-
-    * The maximum size for shaded and hatched areas has been doubled
-    by removing an obsolete speedup hack that was not used anymore.
-
-    * In an unrelated move, shading and hatching has been
-    reimplemented in a much more general fashion.  Now arbitray
-    tilings can be specified as argument to `decoration.filled='.  The
-    old style numerical arguments continue to work, but should be
-    replaced by `empty', `full', `shaded', and `hatched' in new
-    documents.
-
-########################################################################
 # Local Variables:
 # mode:indented-text
 # End:
