Description: fix spelling-error-in-manpage
Author: Florian Schlichting <fsfs@debian.org>
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=75956

--- a/lib/Devel/Cycle.pm
+++ b/lib/Devel/Cycle.pm
@@ -340,7 +340,7 @@
    $reference->[$index] eq $reference_value
 
 The first element of the array reference is the $object_reference that
-you pased to find_cycle() and may not be directly involved in the
+you passed to find_cycle() and may not be directly involved in the
 cycle.
 
 If a reference is a weak ref produced using Scalar::Util's weaken()
@@ -376,7 +376,7 @@
    $reference->[$index] eq $reference_value
 
 The first element of the array reference is the $object_reference that
-you pased to find_cycle() and may not be directly involved in the
+you passed to find_cycle() and may not be directly involved in the
 cycle.
 
 =back
