---
 conf.d/Makefile.am |    1 +
 conf.d/Makefile.in |    1 +
 2 files changed, 2 insertions(+)

Index: fontconfig-2.13.1/conf.d/Makefile.am
===================================================================
--- fontconfig-2.13.1.orig/conf.d/Makefile.am
+++ fontconfig-2.13.1/conf.d/Makefile.am
@@ -42,6 +42,7 @@ CONF_LINKS = \
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
Index: fontconfig-2.13.1/conf.d/Makefile.in
===================================================================
--- fontconfig-2.13.1.orig/conf.d/Makefile.in
+++ fontconfig-2.13.1/conf.d/Makefile.in
@@ -385,6 +385,7 @@ CONF_LINKS = \
 	65-fonts-persian.conf \
 	65-nonlatin.conf \
 	69-unifont.conf \
+	70-no-bitmaps.conf \
 	80-delicious.conf \
 	90-synthetic.conf
 
