Description: Recognize CourierStd as a monospace font
Bug-Ubuntu: https://launchpad.net/bugs/1824260
Author: Gunnar Hjalmarsson <gunnarhj@ubuntu.com>
Forwarded: https://gitlab.freedesktop.org/fontconfig/fontconfig/-/issues/262

diff -ru a/conf.d/30-metric-aliases.conf b/conf.d/30-metric-aliases.conf
--- a/conf.d/30-metric-aliases.conf	2018-06-05 12:36:38.000000000 +0200
+++ b/conf.d/30-metric-aliases.conf	2020-10-27 19:00:20.634096378 +0100
@@ -528,6 +528,13 @@
 	</alias>
 
 	<alias binding="same">
+	  <family>CourierStd</family>
+	  <accept>
+	  <family>Courier</family>
+	  </accept>
+	</alias>
+
+	<alias binding="same">
 	  <family>ITC Avant Garde Gothic</family>
 	  <accept>
 	  <family>TeX Gyre Adventor</family>
diff -ru a/conf.d/45-latin.conf b/conf.d/45-latin.conf
--- a/conf.d/45-latin.conf	2018-06-05 12:36:38.000000000 +0200
+++ b/conf.d/45-latin.conf	2020-10-26 20:02:20.612804230 +0100
@@ -197,6 +197,10 @@
 		<default><family>monospace</family></default>
 	</alias>
 	<alias>
+		<family>CourierStd</family>
+		<default><family>monospace</family></default>
+	</alias>
+	<alias>
 		<family>Cumberland AMT</family>
 		<default><family>monospace</family></default>
 	</alias>
