Index: b/libiberty/Makefile.in
===================================================================
--- a/libiberty/Makefile.in
+++ b/libiberty/Makefile.in
@@ -416,7 +416,7 @@ etags tags TAGS: etags-subdir
 demangle: $(ALL) $(srcdir)/cp-demangle.c
 	@echo "The standalone demangler, now named c++filt, is now"
 	@echo "a part of binutils."
-	$(CC) @DEFS@ $(CFLAGS) $(CPPFLAGS) -I. -I$(INCDIR) $(HDEFINES) \
+	$(CC) @DEFS@ $(CFLAGS) $(CPPFLAGS) $(LDFLAGS) -I. -I$(INCDIR) $(HDEFINES) \
 	  $(srcdir)/cp-demangle.c -DSTANDALONE_DEMANGLER $(TARGETLIB) -o $@
 
 ls:
