Description: Remove unneeded task for builds in Debian
Author: Miguel Landaeta <miguel@miguel.cc>
Bug-Debian: http://bugs.debian.org/646071
Forwarded: no
Last-Update: 2013-08-10

--- multiverse-core-0.7.0.orig/build.gradle
+++ multiverse-core-0.7.0/build.gradle
@@ -193,6 +193,7 @@ project(':multiverse-core') {
     }
 
     uploadArchives() {
+        /* This is only relevant to upstream
         repositories.mavenDeployer {
             name = 'sshDeployer'
             uniqueVersion = false
@@ -206,6 +207,7 @@ project(':multiverse-core') {
                 authentication(userName: "$codehaus_loginname", password: "$codehaus_password")
             }
         }
+        */
     }
 }
 
