Description: Remove unused maven repositories
Author: Markus Koschany <apo@gambaru.de>
Bug-Debian: http://bugs.debian.org/795698
Forwarded: no
Last-Update: 2015-08-16

diff --git a/build.gradle b/build.gradle
index 6667789..bd48828 100644
--- a/build.gradle
+++ b/build.gradle
@@ -47,14 +47,6 @@ project(':multiverse-core') {
         title = "Multiverse Core $version"
     }
 
-    repositories {
-        mavenRepo(url: localMavenRepo)
-        mavenCentral()
-        mavenRepo(url: 'http://download.java.net/maven/2/')
-        mavenRepo(url: 'https://maven.atlassian.com/content/groups/public')
-        mavenRepo(url: 'http://snapshots.repository.codehaus.org')
-    }
-
     dependencies {
         testCompile group: 'junit', name: 'junit', version: '4.10'
         testCompile group: 'org.mockito', name: 'mockito-all', version: '1.9.0'
