--- a/multiverse-core/src/main/java/org/multiverse/api/Lock.java
+++ b/multiverse-core/src/main/java/org/multiverse/api/Lock.java
@@ -52,9 +52,9 @@
  * <p>It is possible to upgrade a lock to more strict version, e.g. to upgrade a read-lock to a write-lock.
  * The following upgrades are possible:
  * <ol>
- * <li>LockMode.Read->LockMode.Write: as long as no other transaction has acquired the Lock in LockMode.Read</li>
- * <li>LockMode.Read->LockMode.Exclusive: as long as no other transaction has acquired the Lock in LockMode.Read</li>
- * <li>LockMode.Write->LockMode.Exclusive: will always succeed</li>
+ * <li>LockMode.Read to LockMode.Write: as long as no other transaction has acquired the Lock in LockMode.Read</li>
+ * <li>LockMode.Read to LockMode.Exclusive: as long as no other transaction has acquired the Lock in LockMode.Read</li>
+ * <li>LockMode.Write to LockMode.Exclusive: will always succeed</li>
  * </ol>
  * <p>
  * The Txn is allowed to apply a more strict LockMode than the one specified.
