From: Daniel Baumann <daniel@debian.org>
Date: Sat, 5 Sep 2009 06:48:56 +0100
Subject: Adjust shebang

Last-Update: 2019-10-08
---
 lib/uniname/gen-uninames.lisp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/uniname/gen-uninames.lisp b/lib/uniname/gen-uninames.lisp
index 86cc2f7..524c40b 100755
--- a/lib/uniname/gen-uninames.lisp
+++ b/lib/uniname/gen-uninames.lisp
@@ -1,4 +1,4 @@
-#!/usr/local/bin/clisp -C
+#!/usr/bin/clisp -C
 
 ;;; Creation of gnulib's uninames.h from the UnicodeData.txt table.
 
