From: Jonas Smedegaard <dr@jones.dk>
Date: Fri, 1 Feb 2019 15:14:30 +0100
Subject: Avoid tests broken or depending on unavailable tool cppi

Last-Update: 2019-10-08
---
 Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/Makefile b/Makefile
index ef6f36f..2766f40 100644
--- a/Makefile
+++ b/Makefile
@@ -22,6 +22,11 @@ info html dvi pdf:
 syntax-check-rules := $(sort $(shell sed -n 's/^\(sc_[a-zA-Z0-9_-]*\):.*/\1/p'\
 			Makefile))
 
+# Skip broken tests
+syntax-check-rules := $(filter-out \
+ sc_check_copyright sc_check_sym_list sc_cpp_indent_check sc_maint,\
+ $(syntax-check-rules))
+
 # Perform some platform independent checks on the gnulib code.
 check: $(syntax-check-rules)
 
