Description: Add Files-Excluded field to header stanza
Origin: vendor
Bug-Debian: https://bugs.debian.org/781369
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2016-03-02
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=112637
Bug: https://rt.cpan.org/Ticket/Display.html?id=112637

--- a/t/01.basic.t
+++ b/t/01.basic.t
@@ -1,4 +1,4 @@
-use Test::More tests => 37;
+use Test::More tests => 38;
 
 use Debian::Copyright;
 
@@ -12,6 +12,7 @@
 isa_ok($copyright, 'Debian::Copyright');
 $copyright->read('t/data/copyright');
 like($copyright->header, qr{\AFormat:\s}xms, 'Header stanza');
+is($copyright->header->Files_Excluded, 'none', 'Files-Excluded header');
 is($copyright->files->Length, 2, 'files length');
 is($copyright->files->Keys(0), '*', 'key files(0)');
 is($copyright->files->Values(0)->Files, '*', 'files(0)->Files');
--- a/t/data/copyright
+++ b/t/data/copyright
@@ -2,6 +2,7 @@
 Upstream-Name: pkg-components
 Upstream-Contact: Nicholas Bamber <nicholas@periapt.co.uk>
 Source: http://github.com/periapt/pkg-components
+Files-Excluded: none
 
 Files: *
 Copyright: 
--- a/lib/Debian/Copyright/Stanza/Header.pm
+++ b/lib/Debian/Copyright/Stanza/Header.pm
@@ -43,6 +43,8 @@
 
 =item Copyright
 
+=item Files_Excluded
+
 =back
 
 =cut
@@ -62,6 +64,7 @@
     Comment
     License
     Copyright
+    Files_Excluded
 );
 
 our $VERSION = '0.2';
