--- a/src/main/java/org/apache/maven/plugin/javadoc/AbstractJavadocMojo.java
+++ b/src/main/java/org/apache/maven/plugin/javadoc/AbstractJavadocMojo.java
@@ -170,14 +170,14 @@
      * Archiver manager
      *
      * @since 2.5
-     * @component
+     * @component role="org.codehaus.plexus.archiver.manager.ArchiverManager"
      */
     private ArchiverManager archiverManager;
 
     /**
      * Factory for creating artifact objects
      *
-     * @component
+     * @component role="org.apache.maven.artifact.factory.ArtifactFactory"
      */
     private ArtifactFactory factory;
 
@@ -185,14 +185,14 @@
      * Used to resolve artifacts of aggregated modules
      *
      * @since 2.1
-     * @component
+     * @component role="org.apache.maven.artifact.metadata.ArtifactMetadataSource"
      */
     private ArtifactMetadataSource artifactMetadataSource;
 
     /**
      * Used for resolving artifacts
      *
-     * @component
+     * @component role="org.apache.maven.artifact.resolver.ArtifactResolver"
      */
     private ArtifactResolver resolver;
 
@@ -200,7 +200,7 @@
      * Project builder
      *
      * @since 2.5
-     * @component
+     * @component role="org.apache.maven.project.MavenProjectBuilder"
      */
     private MavenProjectBuilder mavenProjectBuilder;
 
--- a/src/main/java/org/apache/maven/plugin/javadoc/JavadocJar.java
+++ b/src/main/java/org/apache/maven/plugin/javadoc/JavadocJar.java
@@ -73,14 +73,14 @@
     /**
      * Used for attaching the artifact in the project.
      *
-     * @component
+     * @component role="org.apache.maven.project.MavenProjectHelper"
      */
     private MavenProjectHelper projectHelper;
 
     /**
      * The Jar archiver.
      *
-     * @parameter expression="${component.org.codehaus.plexus.archiver.Archiver#jar}"
+     * @component role="org.codehaus.plexus.archiver.Archiver" roleHint="jar"
      * @since 2.5
      */
     private JarArchiver jarArchiver;
