Description: Disable doclint by default
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
+++ b/src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java
@@ -984,7 +984,7 @@
      *
      * @since 3.0.0
      */
-    @Parameter( property = "doclint" )
+    @Parameter( property = "doclint", defaultValue = "none" )
     private String doclint;
 
     /**
