diff --git a/ChangeLog b/ChangeLog
index 1f6e774..b390185 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,60 @@
+2017-03-25 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.6.1
+	* Library changes:
+	*   Performance improvements in suggest()
+	*   Fixes regressions for Hungarian related to compounding.
+	*   Fixes regressions for Korean related to ICONV.
+	* Command line tool:
+	*   Added Tajik translation 
+	*   Fix regarding serching of OOo dicts installed in user folder
+	* Manpages:
+	*   Fix microsoft-cp1251 to cp1251. Dicts should not use the first.
+	*   Typos.
+	
+2016-12-22 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.6.0
+	* Changes in the library:
+	*  Performance improvement in ngsuggest(), suggestions should be faster.
+	*  Revert MAXWORDLEN to 100 as in 1.3.3 for performance reasons.
+	*  MAXWORDLEN can be set during build time with -D defines.
+	*  Fix crash when word with 102 consecutive X is spelled.
+	* Changes in the command line tool:
+	*  -D shows all loaded dictionares insted of only the first.
+	*  -D properly lists all available dictionaries on Windows.
+
+2016-11-30 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.5.4
+	* Fixes the command COMPOUNDSYLLABLE used in Hungarian dictionary.
+
+2016-11-28 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.5.3
+	* Removed a #include from hunspell.hxx that was creating trouble
+
+2016-11-27 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.5.2
+	* Reverted full backward compatibility with 1.4 public API, again
+
+2016-11-27 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.5.1
+	* Reverted full backward compatibility with 1.4 public API
+
+2016-11-26 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.5
+	* Fixed the changelog notice bellow about the API. I had a mistake.
+
+2016-11-18 Dimitrij Mijoski <dmjpp at hotm>:
+	* Version 1.5
+	* Lot of stability fixes
+	* Fixed compilation errors on various systems (Windows, FreeBSD)
+	* Small performance improvement compared to 1.4.0
+	* The C++ API is updated to use modern C++ types (string, vector).
+	  Backward compatibility is kept for most of the functions except for
+	  the following:
+		get_wordchars();
+		get_version();
+		input_conv(string, string);
+		removed get_csconv();
+
 2016-04-29 Caolán McNamara <caolanm at LibO>:
 	* deprecate old api and add new one
 	  old one remains implemented in terms of new one
