From: Henrique de Moraes Holschuh <hmh@debian.org>
Date: Tue, 25 Sep 2018 08:03:32 +1000
Subject: Set a safe path at startup for benefit of the config scripts

---
 amavisd | 4 ++++
 1 file changed, 4 insertions(+)

--- a/amavisd
+++ b/amavisd
@@ -19014,6 +19014,10 @@
 my $desired_user;                       # username or UID
 if ($> != 0) { $desired_user = $> }     # use effective UID if not root
 
+# Use a default, guaranteed safe path during startup, before loading
+# an user-supplied one from the config file
+$ENV{PATH} = "/bin:/usr/bin";
+
 # collect and parse command line options
 my($log_level_override, $max_servers_override);
 my($myhome_override, $tempbase_override, $helpers_home_override);
