From: Alexander Wirt <formorer@debian.org>
Date: Tue, 25 Sep 2018 08:03:32 +1000
Subject: Fix paths of snmp agent and install it

---
 amavisd-snmp-subagent | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/amavisd-snmp-subagent b/amavisd-snmp-subagent
index d4c313a..c0a427b 100755
--- a/amavisd-snmp-subagent
+++ b/amavisd-snmp-subagent
@@ -69,7 +69,7 @@ $syslog_ident = $myproduct_name;
 $syslog_facility = LOG_MAIL;
 
 my($db_home) =  # DB databases directory
-  defined $ENV{'AMAVISD_DB_HOME'} ? $ENV{'AMAVISD_DB_HOME'} : '/var/amavis/db';
+  defined $ENV{'AMAVISD_DB_HOME'} ? $ENV{'AMAVISD_DB_HOME'} : '/var/lib/amavis/db';
 
 my($mta_queue_dir);
 
@@ -98,7 +98,7 @@ my(@databases) = (
 
 my($log_level) = 0;
 my($daemonize) = 1;
-my($pid_filename);  # e.g. "/var/run/amavisd-snmp-subagent.pid";
+my($pid_filename) = "/var/run/amavis/amavisd-snmp-subagent.pid";  # e.g. "/var/run/amavisd-snmp-subagent.pid";
 
 my($pid_file_created) = 0;
 my($syslog_open) = 0;
@@ -740,7 +740,7 @@ Usage:
    -d log_level    debugging level, 0..5, default 0
    -P pid_file     a file name to receive a PID of a damonized process
    -D db_home_dir  amavis database directory ($db_home),
-                     default AMAVISD_DB_HOME or /var/amavis/db
+                     default AMAVISD_DB_HOME or /var/lib/amavis/db
 EOD
 }
 
