From: Alexander Wirt <formorer@debian.org>
Date: Tue, 25 Sep 2018 08:03:32 +1000
Subject: clarify fqdn error message (#451804)

---
 amavisd | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/amavisd
+++ b/amavisd
@@ -19538,7 +19538,8 @@
   The value of variable \$myhostname is \"$myhn\", but should have been
   a fully qualified domain name; perhaps uname(3) did not provide such.
   You must explicitly assign a FQDN of this host to variable \$myhostname
-  in amavisd.conf, or fix what uname(3) provides as a host's network name!
+  in /etc/amavis/conf.d/05-node_id, or fix what uname(3) provides as a host's 
+  network name!
 EOD
 
 $mail_id_size_bits > 0 &&
