Index: gzip-1.10/gzip.c
===================================================================
--- gzip-1.10.orig/gzip.c
+++ gzip-1.10/gzip.c
@@ -645,7 +645,11 @@ int main (int argc, char **argv)
     ALLOC(ush, tab_prefix1, 1L<<(BITS-1));
 #endif
 
+#ifdef SIGPIPE
     exiting_signal = quiet ? SIGPIPE : 0;
+#else
+    exiting_signal = 0;
+#endif
     install_signal_handlers ();
 
     /* And get to work */
