Description: Support native x32 builds
Author: Christian Seiler <christian@iwakd.de>
Forwarded: yes, but upstream doesn't want to apply: http://thread.gmane.org/gmane.linux.lib.dietlibc/1814/focus=1817

--- a/Makefile
+++ b/Makefile
@@ -74,10 +74,14 @@ else
 ifeq ($(MYARCH),armeb)
 ARCH=arm
 else
+ifeq ($(MYARCH),x32)
+ARCH=x32
+else
 $(error unknown architecture, please fix Makefile)
 endif
 endif
 endif
+endif
 endif
 endif
 endif
--- a/diet.c
+++ b/diet.c
@@ -214,8 +214,12 @@ int main(int argc,char *argv[]) {
       shortplatform="parisc";
 #endif
 #ifdef __x86_64__
+#ifdef __ILP32__
+      shortplatform=(m==32?"i386":(m==64?"x86_64":"x32"));
+#else
       shortplatform=(m==32?"i386":(m==33?"x32":"x86_64"));
 #endif
+#endif
 #ifdef __ia64__
       shortplatform="ia64";
 #endif
