Fix bug #776720
--- a/apps/gtk/gprompter/Makefile.am
+++ b/apps/gtk/gprompter/Makefile.am
@@ -253,10 +253,10 @@
 gprompter_CFLAGS =	$(GNOME_CFLAGS) $(GTHREAD_CFLAGS) $(GMODULE_CFLAGS) \
 			-I$(top_srcdir)/src/lib \
 			-I$(srcdir)/scintilla/include 
-gprompter_LDADD = 	$(GNOME_LIBS) $(GTHREAD_LIBS) $(GMODULE_LIBS) \
-			-lm \
+gprompter_LDADD = 	libscintilla.la \
 			../../../src/lib/libpresage.la \
-			libscintilla.la
+			$(GNOME_LIBS) $(GTHREAD_LIBS) $(GMODULE_LIBS) \
+			-lm
 if USE_GCC
 gprompter_LDADD +=	-lstdc++
 endif
