--- a/apps/dbus/presage_dbus_python_demo.in
+++ b/apps/dbus/presage_dbus_python_demo.in
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 ##########
 #  Presage, an extensible predictive text entry system
--- a/apps/dbus/presage_dbus_service
+++ b/apps/dbus/presage_dbus_service
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 ##########
 #  Presage, an extensible predictive text entry system
--- a/apps/dbus/presage_dbus_service.py
+++ b/apps/dbus/presage_dbus_service.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python2
 
 ##########
 #  Presage, an extensible predictive text entry system
