From: Dmitry Shachnev <mitya57@debian.org>
Date: Thu, 8 Oct 2015 13:58:04 -0700
Subject: disable a test that checks versions of JS libraries

Forwarded: not-needed
---
 tests/test_theming.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/test_theming.py b/tests/test_theming.py
index c90e1e9..0a91843 100644
--- a/tests/test_theming.py
+++ b/tests/test_theming.py
@@ -65,6 +65,7 @@ def test_theme_api(app, status, warning):
     assert not os.path.exists(themedir)
 
 
+@pytest.mark.skip('In Debian we use packaged JS libraries')
 @pytest.mark.sphinx(testroot='tocdepth')  # a minimal root
 def test_js_source(app, status, warning):
     # Now sphinx provides non-minified JS files for jquery.js and underscore.js
