From: Dmitry Shachnev <mitya57@debian.org>
Date: Tue, 2 Jun 2020 12:22:52 +0300
Subject: Use local intersphinx inventory

To prevent Sphinx from downloading https://docs.python.org/3/objects.inv
during build.
---
 doc/conf.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/doc/conf.py b/doc/conf.py
index 712cf63..c6451d6 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -107,8 +107,8 @@ texinfo_documents = [
 ]
 
 intersphinx_mapping = {
-    'python': ('https://docs.python.org/3/', None),
-    'requests': ('https://requests.readthedocs.io/en/master', None),
+    'python': ('/usr/share/doc/python3-doc/html', None),
+    'requests': ('/usr/share/doc/python-requests-doc/html', None),
 }
 
 # Sphinx document translation with sphinx gettext feature uses these settings:
