From: Dmitry Shachnev <mitya57@debian.org>
Date: Thu, 6 Feb 2020 23:55:28 +0300
Subject: Skip tests that require sphinxcontrib.serializinghtml module

---
 tests/test_api_translator.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/tests/test_api_translator.py b/tests/test_api_translator.py
index 3e97cbc..863aa1c 100644
--- a/tests/test_api_translator.py
+++ b/tests/test_api_translator.py
@@ -49,6 +49,7 @@ def test_singlehtml_set_translator_for_singlehtml(app, status, warning):
     assert translator_class.__name__ == 'ConfSingleHTMLTranslator'
 
 
+@pytest.mark.skip('Requires sphinxcontrib.serializinghtml module')
 @pytest.mark.sphinx('pickle', testroot='api-set-translator')
 def test_pickle_set_translator_for_pickle(app, status, warning):
     translator_class = app.builder.get_translator_class()
@@ -56,6 +57,7 @@ def test_pickle_set_translator_for_pickle(app, status, warning):
     assert translator_class.__name__ == 'ConfPickleTranslator'
 
 
+@pytest.mark.skip('Requires sphinxcontrib.serializinghtml module')
 @pytest.mark.sphinx('json', testroot='api-set-translator')
 def test_json_set_translator_for_json(app, status, warning):
     translator_class = app.builder.get_translator_class()
