From: Dmitry Shachnev <mitya57@debian.org>
Date: Tue, 4 Feb 2020 00:12:18 +0300
Subject: Use packaged MathJax (for our own documentation)

---
 doc/conf.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/doc/conf.py b/doc/conf.py
index 3eb01b5..712cf63 100644
--- a/doc/conf.py
+++ b/doc/conf.py
@@ -115,6 +115,7 @@ intersphinx_mapping = {
 locale_dirs = ['locale/']
 gettext_compact = False
 
+mathjax_path = 'file:///usr/share/javascript/mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML'
 
 # -- Extension interface -------------------------------------------------------
 
