Description: spelling error (perfrom -> perform)
Author: Damyan Ivanov <dmn@debian.org>
Forwarded: https://sourceforge.net/p/firebird/mailman/message/34934257/

--- a/src/common/isc_sync.cpp
+++ b/src/common/isc_sync.cpp
@@ -950,7 +950,7 @@ ULONG ISC_exception_post(ULONG sig_num,
 	case SIGILL:
 
 		sprintf(log_msg, "%s Illegal Instruction.\n"
-				"\t\tThe code attempted to perfrom an\n"
+				"\t\tThe code attempted to perform an\n"
 				"\t\tillegal operation."
 				"\tThis exception will cause the Firebird server\n"
 				"\tto terminate abnormally.", err_msg);
--- a/src/include/gen/msgs.h
+++ b/src/include/gen/msgs.h
@@ -484,7 +484,7 @@ static const struct {
 	{335544780, "An exception occurred that does not have a description.  Exception number @1."},		/* exception_unknown */
 	{335544781, "Stack overflow.  The resource requirements of the runtime stack have exceeded the memory available to it."},		/* exception_stack_overflow */
 	{335544782, "Segmentation Fault. The code attempted to access memory without privileges."},		/* exception_sigsegv */
-	{335544783, "Illegal Instruction. The Code attempted to perfrom an illegal operation."},		/* exception_sigill */
+	{335544783, "Illegal Instruction. The Code attempted to perform an illegal operation."},		/* exception_sigill */
 	{335544784, "Bus Error. The Code caused a system bus error."},		/* exception_sigbus */
 	{335544785, "Floating Point Error. The Code caused an Arithmetic Exception or a floating point exception."},		/* exception_sigfpe */
 	{335544786, "Cannot delete rows from external files."},		/* ext_file_delete */
--- a/src/msgs/history2.sql
+++ b/src/msgs/history2.sql
@@ -666,7 +666,7 @@ without specifying a character set.', NU
 (667, 'daves', '1994-07-18 10:39:14', 12, 104, '	%sZ		      print version number', NULL, NULL, 'c_pg');
 (668, 'daves', '1994-07-18 10:39:30', 12, 109, '	%sY  <path>           redirect/supress status message output', NULL, NULL, 'c_pg');
 (669, 'daves', '1994-07-18 11:00:17', 13, 382, 'Cannot delete object referencec by another object', NULL, NULL, 'c_pg');
-(670, 'daves', '1994-07-18 11:02:30', 2, 177, 'Unsuccesful attempt to modify trigger relation', NULL, NULL, 'c_pg');
+(670, 'daves', '1994-07-18 11:02:30', 2, 177, 'Unsuccessful attempt to modify trigger relation', NULL, NULL, 'c_pg');
 (671, 'daves', '1994-07-18 11:04:54', 12, 170, 'commiting metadata for table %s', NULL, NULL, 'c_pg');
 (672, 'daves', '1994-07-18 11:05:46', 12, 171, 'error commiting metadata for relation %s', NULL, NULL, 'c_pg');
 (673, 'daves', '1994-07-18 11:07:03', 16, 104, ' Number of archives running concurrently excceds limit.', NULL, NULL, 'c_pg');
@@ -895,7 +895,7 @@ message to appear.', 'Your program encou
 (909, 'SYSDBA', '1998-09-29 12:47:50', 0, 453, 'Floating-point inexact result.  The result of a floating-point operation cannot be represented exactly as a decimal fr', NULL, NULL, 'c_pg');
 (910, 'SYSDBA', '1998-09-29 12:48:33', 0, 459, 'Interger overflow.  The result of an integer operation caused the most significant bit of the result to carry.', NULL, NULL, 'c_pg');
 (911, 'SYSDBA', '1998-09-29 12:49:25', 0, 462, 'Segmentation Fault: code attempted to access memory without privileges.', NULL, NULL, 'c_pg');
-(912, 'SYSDBA', '1998-09-29 12:50:13', 0, 463, 'Illegal Instruction: code attempted to perfrom an illegal operation', NULL, NULL, 'c_pg');
+(912, 'SYSDBA', '1998-09-29 12:50:13', 0, 463, 'Illegal Instruction: code attempted to perform an illegal operation', NULL, NULL, 'c_pg');
 (913, 'SYSDBA', '1998-09-29 12:50:32', 0, 464, 'Bus Error: code caused a system bus error', NULL, NULL, 'c_pg');
 (914, 'SYSDBA', '1998-09-29 12:51:01', 0, 465, 'Floating Point Error: Code caused an Arithmetic Exception or a floating point exception', NULL, NULL, 'c_pg');
 (915, 'duquette', '1998-10-15 17:17:38', 18, 15, 'gsec - unable to open database', NULL, NULL, 'c_pg');
--- a/src/msgs/messages2.sql
+++ b/src/msgs/messages2.sql
@@ -559,7 +559,7 @@ without specifying a character set.', NU
 ('exception_unknown', 'POST_EXCEPTIONS and POST_EXTRENA', 'common.h', 'obsolete', 0, 460, NULL, 'An exception occurred that does not have a description.  Exception number @1.', NULL, NULL);
 ('exception_stack_overflow', 'POST_EXCEPTION', 'common.h', NULL, 0, 461, NULL, 'Stack overflow.  The resource requirements of the runtime stack have exceeded the memory available to it.', NULL, NULL);
 ('exception_sigsegv', 'ISC_exception_post', 'isc_sync.c', NULL, 0, 462, NULL, 'Segmentation Fault. The code attempted to access memory without privileges.', NULL, NULL);
-('exception_sigill', 'ISC_exception_post', 'isc_sync.c', NULL, 0, 463, NULL, 'Illegal Instruction. The Code attempted to perfrom an illegal operation.', NULL, NULL);
+('exception_sigill', 'ISC_exception_post', 'isc_sync.c', NULL, 0, 463, NULL, 'Illegal Instruction. The Code attempted to perform an illegal operation.', NULL, NULL);
 ('exception_sigbus', 'ISC_exception_post', 'isc_sync.c', NULL, 0, 464, NULL, 'Bus Error. The Code caused a system bus error.', NULL, NULL);
 ('exception_sigfpe', 'ISC_exception_post', 'isc_sync.c', NULL, 0, 465, NULL, 'Floating Point Error. The Code caused an Arithmetic Exception or a floating point exception.', NULL, NULL);
 ('ext_file_delete', 'EXT_erase', 'ext.c', NULL, 0, 466, NULL, 'Cannot delete rows from external files.', NULL, NULL);
--- a/src/msgs/transmsgs.de_DE2.sql
+++ b/src/msgs/transmsgs.de_DE2.sql
@@ -608,7 +608,7 @@ segmented) blob.', 'daves', '1993-07-16
 ('An exception occurred that does not have a description.  Exception number @1.', NULL, NULL, 0, 460, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Stack overflow.  The resource requirements of the runtime stack have exceeded the memory available to it.', NULL, NULL, 0, 461, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Segmentation Fault. The code attempted to access memory without privileges.', NULL, NULL, 0, 462, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
-('Illegal Instruction. The Code attempted to perfrom an illegal operation.', NULL, NULL, 0, 463, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
+('Illegal Instruction. The Code attempted to perform an illegal operation.', NULL, NULL, 0, 463, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Bus Error. The Code caused a system bus error.', NULL, NULL, 0, 464, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Floating Point Error. The Code caused an Arithmetic Exception or a floating point exception.', NULL, NULL, 0, 465, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Cannot delete rows from external files.', NULL, NULL, 0, 466, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
--- a/src/msgs/transmsgs.fr_FR2.sql
+++ b/src/msgs/transmsgs.fr_FR2.sql
@@ -608,7 +608,7 @@ segmented) blob.', 'daves', '1993-07-16
 ('An exception occurred that does not have a description.  Exception number @1.', NULL, NULL, 0, 460, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Stack overflow.  The resource requirements of the runtime stack have exceeded the memory available to it.', NULL, NULL, 0, 461, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Segmentation Fault. The code attempted to access memory without privileges.', NULL, NULL, 0, 462, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
-('Illegal Instruction. The Code attempted to perfrom an illegal operation.', NULL, NULL, 0, 463, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
+('Illegal Instruction. The Code attempted to perform an illegal operation.', NULL, NULL, 0, 463, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Bus Error. The Code caused a system bus error.', NULL, NULL, 0, 464, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Floating Point Error. The Code caused an Arithmetic Exception or a floating point exception.', NULL, NULL, 0, 465, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
 ('Cannot delete rows from external files.', NULL, NULL, 0, 466, '#', NULL, NULL, 'truser', '1000-01-01 00:00:00')
--- a/src/jrd/CryptoManager.cpp
+++ b/src/jrd/CryptoManager.cpp
@@ -1148,7 +1148,7 @@ namespace Jrd {
 			}
 
 			// Slow IO - we need exclusive lock on crypto manager.
-			// That may happen only when another process changed DB encyption.
+			// That may happen only when another process changed DB encryption.
 			BarSync::LockGuard lockGuard(tdbb, sync);
 			lockGuard.lock();
 			for (SINT64 previous = slowIO; ; previous = slowIO)
--- a/src/remote/client/interface.cpp
+++ b/src/remote/client/interface.cpp
@@ -1662,7 +1662,7 @@ void Attachment::freeClientData(CheckSta
 
 		if (status->getState() & Firebird::IStatus::STATE_ERRORS)
 		{
-			iscLogStatus("REMOTE INTERFACE/gds__detach: Unsuccesful detach from "
+			iscLogStatus("REMOTE INTERFACE/gds__detach: Unsuccessful detach from "
 					"database.\n\tUncommitted work may have been lost.", status);
 			reset(status);
 		}
--- a/src/remote/protocol.h
+++ b/src/remote/protocol.h
@@ -365,7 +365,7 @@ const UCHAR CNCT_specific_data		= 7;	//
 const UCHAR CNCT_plugin_name		= 8;	// Name of plugin, which generated that data
 const UCHAR CNCT_login				= 9;	// Same data as isc_dpb_user_name
 const UCHAR CNCT_plugin_list		= 10;	// List of plugins, available on client
-const UCHAR CNCT_client_crypt		= 11;	// Client encyption level (DISABLED/ENABLED/REQUIRED)
+const UCHAR CNCT_client_crypt		= 11;	// Client encryption level (DISABLED/ENABLED/REQUIRED)
 
 // Accept Block (Server response to connect block)
 
