From: Jeremy Harris <jgh146exb@wizmail.org>
Date: Thu, 19 May 2022 14:24:48 +0100
Subject: ARC: reset headers before signing for secondary MX. Bug 2886

Origin: upstream, https://git.exim.org/exim.git/commitdiff/5a8015582376ff3cc0c0d034d9237008b10d2164
Bug: https://bugs.exim.org/show_bug.cgi?id=2886
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/exim4/+bug/1974214
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1004740
---
 src/arc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/arc.c b/src/arc.c
index a68ab6e..b150d3a 100644
--- a/src/arc.c
+++ b/src/arc.c
@@ -1531,6 +1531,7 @@ void
 arc_sign_init(void)
 {
 memset(&arc_sign_ctx, 0, sizeof(arc_sign_ctx));
+headers_rlist = NULL;
 }
 
 
