Description: do not use `git ls-files`
Author: HIGUCHI Daisuke (VDR dai) <dai@debian.org>
Forwarded: not-needed
Last-Update: 2020-08-10

Index: coderay/coderay.gemspec
===================================================================
--- coderay.orig/coderay.gemspec
+++ coderay/coderay.gemspec
@@ -24,8 +24,8 @@ Gem::Specification.new do |s|
   
   readme_file = 'README_INDEX.rdoc'
   
-  s.files         = `git ls-files -- lib/* #{readme_file} MIT-LICENSE`.split("\n")
-  s.executables   = `git ls-files -- bin/*`.split("\n").map { |f| File.basename(f) }
+  s.files         = Dir['lib/*'] << readme_file << 'MIT-LICENSE'
+  s.executables   = Dir['bin/*'].map { |f| File.basename(f) }
   s.require_paths = ['lib']
   
   s.rdoc_options      = '-SNw2', "-m#{readme_file}", '-t CodeRay Documentation'
