Description: fix coderay absolute file path test
Author: HIGUCHI Daisuke (VDR dai) <dai@debian.org>
Forwarded: not-needed
Last-Update: 2020-08-20

Index: coderay/spec/coderay_spec.rb
===================================================================
--- coderay.orig/spec/coderay_spec.rb
+++ coderay/spec/coderay_spec.rb
@@ -9,8 +9,7 @@ RSpec.describe CodeRay do
 
   describe '.coderay_path' do
     it 'returns an absolute file path to the given code file' do
-      base = File.expand_path('../..', __FILE__)
-      expect(CodeRay.coderay_path('file')).to eq("#{base}/lib/coderay/file")
+      expect(CodeRay.coderay_path('file')).to eq("#{CodeRay::CODERAY_PATH}/file")
     end
   end
 
