Description: Add compatibility with Gradle 2.5
Author: Miguel Landaeta <nomadium@debian.org>
Forwarded: no
Last-Update: 2015-08-16

--- a/build.gradle
+++ b/build.gradle
@@ -329,7 +329,8 @@
 
 task zipDist(type: Zip) {
     from jar.outputs.files
-    from(runtimeClasspath) {
+    //from(runtimeClasspath) {
+    from(sourceSets.main.runtimeClasspath.asPath) {
         include('jsr166*', 'netty*', 'multiverse*')
     }
     from('licenses') {
