Description: Improve the reproducibility of the documentation
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/build.gradle
+++ b/build.gradle
@@ -257,6 +257,8 @@
     options.overview('overview.html')
     options.showAll()
     options.encoding('UTF-8')
+    options.locale("en")
+    options.noTimestamp(true)
     options.setUse(true)
     options.author(true)
     options.version(true)
