From: Daniel T Chen <crimsun@ubuntu.com>
Subject: Link explicitly against the built libmjpegutils. Fixes FTBFS.

---
 lavtools/Makefile.am |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- mjpegtools.orig/lavtools/Makefile.am
+++ mjpegtools/lavtools/Makefile.am
@@ -144,7 +144,7 @@ lavrec_LDADD = $(LIBMJPEGUTILS) \
 lav2yuv_SOURCES = lav2yuv.c lav_common.c
 lav2yuv_CPPFLAGS = $(AM_CPPFLAGS) $(LIBDV_CFLAGS)
 lav2yuv_LDADD = $(LIBMJPEGUTILS) \
-		liblavfile.la liblavjpeg.la
+		liblavfile.la liblavjpeg.la -ldv
 
 jpeg2yuv_SOURCES = jpeg2yuv.c
 jpeg2yuv_CPPFLAGS = $(AM_CPPFLAGS) $(JPEG_CFLAGS)
