From: Daniel T Chen <crimsun@ubuntu.com>
Subject: Link explicitly against the built libmjpegutils. Fixes FTBFS.

---
 mplex/Makefile.am |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- mjpegtools.orig/mplex/Makefile.am
+++ mjpegtools/mplex/Makefile.am
@@ -74,5 +74,6 @@ mplex_SOURCES = main.cpp
 
 mplex_DEPENDENCIES = libmplex2.la
 
-mplex_LDADD = libmplex2.la @LIBGETOPT_LIB@ $(LIBM_LIBS)
+mplex_LDADD = libmplex2.la @LIBGETOPT_LIB@ $(LIBM_LIBS) \
+	$(top_builddir)/utils/libmjpegutils.la
 
