Description: Do not use -fno-PIC
Author: Sebastian Ramacher <sramacher@debian.org>
Bug-Debian: https://bugs.debian.org/837566
Last-Update: 2016-11-16

--- mjpegtools-2.1.0+debian.orig/y4mdenoise/Makefile.am
+++ mjpegtools-2.1.0+debian/y4mdenoise/Makefile.am
@@ -4,8 +4,8 @@ MAINTAINERCLEANFILES = Makefile.in
 
 EXTRA_DIST = implementation.html
 
-AM_CFLAGS = -DNDEBUG -finline-functions @PROGRAM_NOPIC@
-AM_CXXFLAGS = -DNDEBUG -finline-functions @PROGRAM_NOPIC@
+AM_CFLAGS = -DNDEBUG -finline-functions
+AM_CXXFLAGS = -DNDEBUG -finline-functions
 
 INCLUDES =  -I$(top_srcdir) -I$(top_srcdir)/utils
 
--- mjpegtools-2.1.0+debian.orig/y4munsharp/Makefile.am
+++ mjpegtools-2.1.0+debian/y4munsharp/Makefile.am
@@ -4,8 +4,6 @@ MAINTAINERCLEANFILES = Makefile.in
 
 INCLUDES = -I $(top_srcdir)/utils -I $(top_srcdir)
 
-y4munharp_CFLAGS=@PROGRAM_NOPIC@
-
 bin_PROGRAMS = y4munsharp
 
 y4munsharp_SOURCES = y4munsharp.c
--- mjpegtools-2.1.0+debian.orig/y4mutils/Makefile.am
+++ mjpegtools-2.1.0+debian/y4mutils/Makefile.am
@@ -2,7 +2,6 @@
 
 MAINTAINERCLEANFILES = Makefile.in
 
-AM_CFLAGS   = @PROGRAM_NOPIC@
 AM_CPPFLAGS = -I$(top_srcdir) -I$(top_srcdir)/utils
 
 LIBMJPEGUTILS = $(top_builddir)/utils/libmjpegutils.la
--- mjpegtools-2.1.0+debian.orig/yuvscaler/Makefile.am
+++ mjpegtools-2.1.0+debian/yuvscaler/Makefile.am
@@ -18,6 +18,5 @@ noinst_HEADERS = \
 
 EXTRA_DIST = yuvscaler_implementation.txt
 
-yuvscaler_CFLAGS=@PROGRAM_NOPIC@
 yuvscaler_SOURCES = yuvscaler.c yuvscaler_resample.c yuvscaler_bicubic.c
 yuvscaler_LDADD = $(LIBMJPEGUTILS) $(LIBM_LIBS)
