From:
Subject:

--- libitext-java-2.1.7.orig/core/com/lowagie/text/pdf/RandomAccessFileOrArray.java
+++ libitext-java-2.1.7/core/com/lowagie/text/pdf/RandomAccessFileOrArray.java
@@ -99,7 +99,11 @@ public class RandomAccessFileOrArray imp
                 }
             }
             else {
-                InputStream is = BaseFont.getResourceStream(filename);
+                InputStream is;
+                if( filename.equals("-"))
+                    is = System.in;
+                else
+                    is = BaseFont.getResourceStream(filename);
                 if (is == null)
                     throw new IOException(filename + " not found as file or resource.");
                 try {
