Description: Fixes the compatibility with Java 17
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/src/test/org/apache/commons/httpclient/ssl/SimpleSSLSocketFactory.java
+++ b/src/test/org/apache/commons/httpclient/ssl/SimpleSSLSocketFactory.java
@@ -42,9 +42,9 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
-import com.sun.net.ssl.KeyManager;
-import com.sun.net.ssl.KeyManagerFactory;
-import com.sun.net.ssl.SSLContext;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLContext;
 
 /**
  * Defines a SSL socket factory
--- a/src/test/org/apache/commons/httpclient/ssl/SimpleSSLTestProtocolSocketFactory.java
+++ b/src/test/org/apache/commons/httpclient/ssl/SimpleSSLTestProtocolSocketFactory.java
@@ -46,9 +46,9 @@
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 
-import com.sun.net.ssl.SSLContext;
-import com.sun.net.ssl.TrustManager;
-import com.sun.net.ssl.TrustManagerFactory;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
 
 public class SimpleSSLTestProtocolSocketFactory implements SecureProtocolSocketFactory {
 
