From: =?utf-8?q?H=C3=A5vard_Flaget_Aasen?= <haavard_aasen@yahoo.no>
Date: Tue, 21 Jan 2020 10:22:27 +0100
Subject: Disable test because of missing *.js files

Forwarded: not-needed
---
 jsbeautifier/unpackers/tests/testmyobfuscate.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jsbeautifier/unpackers/tests/testmyobfuscate.py b/jsbeautifier/unpackers/tests/testmyobfuscate.py
index d69df82..a837e05 100644
--- a/jsbeautifier/unpackers/tests/testmyobfuscate.py
+++ b/jsbeautifier/unpackers/tests/testmyobfuscate.py
@@ -14,7 +14,7 @@ OUTPUT = os.path.join(path[0], "test-myobfuscate-output.js")
 
 # pylint: disable=R0904
 
-
+@unittest.skip("missing files")
 class TestMyObfuscate(unittest.TestCase):
     # pylint: disable=C0103
     """MyObfuscate obfuscator testcase."""
