Description: Support m68k
Bug-Debian: https://bugs.debian.org/699849
Author: Thorsten Glaser <tg@mirbsd.de>
Reviewed-by: Sébastien Villemot <sebastien@debian.org>
Last-Update: 2015-04-30

Index: double-conversion/double-conversion/utils.h
===================================================================
--- double-conversion.orig/double-conversion/utils.h
+++ double-conversion/double-conversion/utils.h
@@ -155,6 +155,9 @@ int main(int argc, char** argv) {
 #else
 #undef DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS
 #endif  // _WIN32
+#elif defined(__m68k__)
+// The MC68881 also uses an 80bit wide floating point stack.
+#undef DOUBLE_CONVERSION_CORRECT_DOUBLE_OPERATIONS
 #else
 #error Target architecture was not detected as supported by Double-Conversion.
 #endif
