Description: fix pod errors
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=76019
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=76019
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2012-03-24

--- a/Chart.pod
+++ b/Chart.pod
@@ -689,7 +689,6 @@
 header, to stdout, allowing you to call chart-generating scripts directly
 from your html pages (ie. with a <lt>img src=image.pl<gt> HTML tag). 
 
-=back
 
 =item Produce a png image as a scalar
 
@@ -699,9 +698,6 @@
 
   	$obj-scalar_png($dataref)
 
-=back
-
-
 
 =item Produce a jpeg image as a scalar
 
