From: Simon McVittie <smcv@debian.org>
Date: Fri, 1 Jan 2021 15:03:27 +0000
Subject: Change EPERM error message to show Debian-specific information

Forwarded: not-needed
---
 bubblewrap.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bubblewrap.c b/bubblewrap.c
index fc2edbb..40aca5d 100644
--- a/bubblewrap.c
+++ b/bubblewrap.c
@@ -2652,7 +2652,7 @@ main (int    argc,
           if (errno == EINVAL)
             die ("Creating new namespace failed, likely because the kernel does not support user namespaces.  bwrap must be installed setuid on such systems.");
           else if (errno == EPERM && !is_privileged)
-            die ("No permissions to creating new namespace, likely because the kernel does not allow non-privileged user namespaces. On e.g. debian this can be enabled with 'sysctl kernel.unprivileged_userns_clone=1'.");
+            die ("No permissions to create new namespace, likely because the kernel does not allow non-privileged user namespaces. See <https://deb.li/bubblewrap> or <file:///usr/share/doc/bubblewrap/README.Debian.gz>.");
         }
 
       die_with_error ("Creating new namespace failed");
