From cbb944434e36341d561c3b1e05ff2c50fa014ef7 Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Thu, 21 Oct 2021 13:08:59 +0200
Subject: [PATCH] Enable nonet for xsltproc

---
 asciidoc/a2x.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/asciidoc/a2x.py b/asciidoc/a2x.py
index 4e9d363..8ad958b 100644
--- a/asciidoc/a2x.py
+++ b/asciidoc/a2x.py
@@ -519,7 +519,7 @@ class A2X(AttrDict):
             self.asciidoc_opts.append(('--doctype', self.doctype))
         for attr in self.attributes:
             self.asciidoc_opts.append(('--attribute', attr))
-#        self.xsltproc_opts += ' --nonet'
+        self.xsltproc_opts += ' --nonet'
         if self.verbose:
             self.asciidoc_opts.append(('--verbose',))
             self.dblatex_opts += ' -V'
-- 
2.33.0

