From 47cdcabd598551e81ccc50806b43ed6f5200a7f7 Mon Sep 17 00:00:00 2001
From: Leon Marz <main@lmarz.org>
Date: Thu, 21 Oct 2021 13:06:20 +0200
Subject: [PATCH] testasciidoc path

---
 tests/testasciidoc.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tests/testasciidoc.py b/tests/testasciidoc.py
index 2aace85..2db41fa 100755
--- a/tests/testasciidoc.py
+++ b/tests/testasciidoc.py
@@ -12,6 +12,7 @@ import shutil
 import sys
 
 sys.path.append(str(Path(__file__).resolve().parent.parent))
+sys.path.append('/usr/share/asciidoc')
 from asciidoc import asciidoc  # noqa: E402
 
 # Default backends.
-- 
2.33.0

