From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 14 Apr 2022 18:51:15 +0200
Subject: appIndicator: Fix indicator IDs on error logs

---
 appIndicator.js | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/appIndicator.js b/appIndicator.js
index 07576a3..2a00b36 100644
--- a/appIndicator.js
+++ b/appIndicator.js
@@ -373,7 +373,7 @@ var AppIndicator = class AppIndicatorsAppIndicator {
         // ... and don't seem to have any effect.
         this._proxy.ActivateRemote(x, y, this._cancellable, (_, e) => {
             if (e && !e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.CANCELLED))
-                Util.Logger.critical(`${this._indicator.id}, failed to activate: ${e.message}`);
+                Util.Logger.critical(`${this.id}, failed to activate: ${e.message}`);
         });
     }
 
@@ -384,10 +384,10 @@ var AppIndicator = class AppIndicatorsAppIndicator {
             if (e && e.matches(Gio.DBusError, Gio.DBusError.UNKNOWN_METHOD)) {
                 this._proxy.SecondaryActivateRemote(x, y, cancellable, (_r, error) => {
                     if (error && !error.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.CANCELLED))
-                        Util.Logger.critical(`${this._indicator.id}, failed to secondary activate: ${e.message}`);
+                        Util.Logger.critical(`${this.id}, failed to secondary activate: ${e.message}`);
                 });
             } else if (e && !e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.CANCELLED)) {
-                Util.Logger.critical(`${this._indicator.id}, failed to secondary activate: ${e.message}`);
+                Util.Logger.critical(`${this.id}, failed to secondary activate: ${e.message}`);
             }
         });
     }
@@ -398,14 +398,14 @@ var AppIndicator = class AppIndicatorsAppIndicator {
         if (dx !== 0) {
             this._proxy.ScrollRemote(Math.floor(dx), 'horizontal', cancellable, (_, e) => {
                 if (e && !e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.CANCELLED))
-                    Util.Logger.critical(`${this._indicator.id}, failed to scroll horizontally: ${e.message}`);
+                    Util.Logger.critical(`${this.id}, failed to scroll horizontally: ${e.message}`);
             });
         }
 
         if (dy !== 0) {
             this._proxy.ScrollRemote(Math.floor(dy), 'vertical', cancellable, (_, e) => {
                 if (e && !e.matches(Gio.IOErrorEnum, Gio.IOErrorEnum.CANCELLED))
-                    Util.Logger.critical(`${this._indicator.id}, failed to scroll vertically: ${e.message}`);
+                    Util.Logger.critical(`${this.id}, failed to scroll vertically: ${e.message}`);
             });
         }
     }
