--- a/StandAlone/CMakeLists.txt
+++ b/StandAlone/CMakeLists.txt
@@ -47,7 +47,7 @@ add_custom_command(
 
 #add_custom_target(glslangValidator DEPENDS ${GLSLANG_INTRINSIC_H})
 
-add_library(glslang-default-resource-limits
+add_library(glslang-default-resource-limits STATIC
             ${CMAKE_CURRENT_SOURCE_DIR}/ResourceLimits.cpp
             ${CMAKE_CURRENT_SOURCE_DIR}/resource_limits_c.cpp)
 set_property(TARGET glslang-default-resource-limits PROPERTY FOLDER glslang)
