From 9e609af6caea5996d6b0f7e5a6b9872dc18b3800 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Nov=C3=BD?= <onovy@debian.org>
Date: Thu, 3 Nov 2016 21:43:46 +0100
Subject: Remove upper constrains from upstream requirements

=== modified file 'setup.cfg'
--- a/setup.cfg
+++ b/setup.cfg
@@ -35,9 +35,9 @@ packages = find:
 package_dir = 
 	=src
 install_requires = 
-	mccabe>=0.6.0,<0.7.0
-	pycodestyle>=2.8.0,<2.9.0
-	pyflakes>=2.4.0,<2.5.0
+	mccabe>=0.6.0
+	pycodestyle>=2.8.0
+	pyflakes>=2.4.0
 	importlib-metadata<4.3;python_version<"3.8"
 python_requires = >=3.6

