Description: clean gruntfile
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2021-12-13

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -219,30 +219,9 @@
 
   this.registerTask('test', ['test:bin', 'test:cov']);
 
-  grunt.registerTask('bench', ['metrics']);
-
-  if (process.env.SAUCE_ACCESS_KEY) {
-    grunt.registerTask('sauce', ['concat:tests', 'connect', 'saucelabs-mocha']);
-  } else {
-    grunt.registerTask('sauce', []);
-  }
-
-  // Requires secret properties (saucelabs-credentials etc.) from .travis.yaml
-  grunt.registerTask('extensive-tests-and-publish-to-aws', [
-    'default',
-    'sauce',
-    'metrics',
-    'publish-to-aws'
-  ]);
-  grunt.registerTask('on-file-change', [
-    'build',
-    'concat:tests',
-    'test'
-  ]);
-
   // === Primary tasks ===
   grunt.registerTask('dev', ['clean', 'connect', 'watch']);
-  grunt.registerTask('default', ['clean', 'build', 'test', 'release']);
+  grunt.registerTask('default', ['clean', 'build']);
   grunt.registerTask('integration-tests', [
     'default'
   ]);
