Description: Disable bg-shell plugin
Author: Xavier Guimard <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2020-02-03

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -193,7 +193,7 @@
   grunt.loadNpmTasks('grunt-contrib-copy');
   grunt.loadNpmTasks('grunt-contrib-uglify');
   grunt.loadNpmTasks('grunt-babel');
-  grunt.loadNpmTasks('grunt-bg-shell');
+  //grunt.loadNpmTasks('grunt-bg-shell');
   grunt.loadNpmTasks('grunt-webpack');
 
   grunt.task.loadTasks('tasks');
@@ -228,7 +228,6 @@
   // Requires secret properties (saucelabs-credentials etc.) from .travis.yaml
   grunt.registerTask('extensive-tests-and-publish-to-aws', [
     'default',
-    'bgShell:integrationTests',
     'sauce',
     'metrics',
     'publish-to-aws'
@@ -243,7 +242,6 @@
   grunt.registerTask('dev', ['clean', 'connect', 'watch']);
   grunt.registerTask('default', ['clean', 'build', 'test', 'release']);
   grunt.registerTask('integration-tests', [
-    'default',
-    'bgShell:integrationTests'
+    'default'
   ]);
 };
