--- a/gradle/filter.gradle
+++ b/gradle/filter.gradle
@@ -18,7 +18,7 @@
  */
 import java.text.SimpleDateFormat
 
-ext.buildTime = new Date()
+ext.buildTime = new Date(1575590041000L)
 
 ext.propertiesFilter = [
         beginToken: '#',
--- a/gradle/docs.gradle
+++ b/gradle/docs.gradle
@@ -197,6 +197,7 @@
 
 if (JavaVersion.current().isJava7Compatible()) {
     javadocAll.options.source = '1.7'
+    javadocAll.options.noTimestamp()
 }
 
 if (JavaVersion.current().isJava8Compatible()) {
@@ -204,6 +205,7 @@
         tasks.withType(Javadoc) {
             // disable the crazy super-strict doclint tool in Java 8
             options.addStringOption('Xdoclint:none', '-quiet')
+            options.noTimestamp()
         }
     }
 }
--- a/subprojects/groovy-nio/build.gradle
+++ b/subprojects/groovy-nio/build.gradle
@@ -36,5 +36,6 @@
 }
 tasks.withType(Javadoc) {
         options.source = "1.7"
+        options.noTimestamp()
 }
 
