Description: Fixes the compatibility with Java 11
Origin: backport, https://github.com/apache/groovy/commit/90a852b1#diff-8e84f74cdcd98790e1e699f245b18ba1
--- a/subprojects/groovy-console/src/main/groovy/groovy/ui/text/StructuredSyntaxResources.java
+++ b/subprojects/groovy-console/src/main/groovy/groovy/ui/text/StructuredSyntaxResources.java
@@ -41,7 +41,7 @@
             // a security exception
             SecurityManager mgr = System.getSecurityManager();
             if (mgr != null) {
-                mgr.checkSystemClipboardAccess();
+                mgr.checkPermission(new AWTPermission("accessClipboard"));
             }
             systemClipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
         }
