From: Markus Koschany <apo@debian.org>
Date: Wed, 20 Feb 2019 13:54:09 +0100
Subject: detectJavaApiLink

Bug-Debian: https://bugs.debian.org/922755
---
 gradle/docs.gradle | 1 +
 1 file changed, 1 insertion(+)

diff --git a/gradle/docs.gradle b/gradle/docs.gradle
index 9088a73..51eaaf5 100644
--- a/gradle/docs.gradle
+++ b/gradle/docs.gradle
@@ -36,6 +36,7 @@ def javadocSpec = {
         links('file:///usr/share/doc/ant/api/', 'file:///usr/share/doc/default-jdk/api/',
               'file:///usr/share/doc/libservlet3.1-java-doc/api',
               'file:///usr/share/doc/junit4/api/')
+        failOnError = false
     }
 }
 
