Description: fix a spelling mistake
Origin: vendor
Bug: https://rt.cpan.org/Ticket/Display.html?id=88657
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=88657
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2013-09-13

--- a/lib/File/Type.pm
+++ b/lib/File/Type.pm
@@ -1572,7 +1572,7 @@
 L<File::MimeInfo> uses an external magic file, and relies on file extensions
 rather than magic to determine the mime type.
 
-As a result, File::Type uses a seperate module and script to create the
+As a result, File::Type uses a separate module and script to create the
 code at the core of this module, which means that there is no need to
 include a magic database at all, and that it is safe to run under mod_perl.
 
