Description: Fixes the compatibility with Java 5
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/api/src/java/org/apache/avalon/framework/Enum.java
+++ b/api/src/java/org/apache/avalon/framework/Enum.java
@@ -126,11 +126,11 @@
         if( !(o instanceof Enum) )
             return false;
 
-        final Enum enum = (Enum)o;
+        final Enum enum_ = (Enum)o;
 
-        if( !getClass().equals( enum.getClass() ) )
+        if( !getClass().equals( enum_.getClass() ) )
             return false;
-        if( m_name != null ? !m_name.equals( enum.m_name ) : enum.m_name != null )
+        if( m_name != null ? !m_name.equals( enum_.m_name ) : enum_.m_name != null )
             return false;
 
         return true;
