--- a/src/test/java/org/apache/maven/reporting/exec/TestDefaultMavenReportExecutor.java
+++ b/src/test/java/org/apache/maven/reporting/exec/TestDefaultMavenReportExecutor.java
@@ -231,7 +231,7 @@
             return localArtifactRepository;
         }
         String localRepoPath =
-            System.getProperty( "localRepository", MavenCli.userMavenConfigurationHome.getPath() + "/repository" );
+            System.getProperty( "localRepository", MavenCli.USER_MAVEN_CONFIGURATION_HOME.getPath() + "/repository" );
 
         localArtifactRepository = lookup( RepositorySystem.class ).createLocalRepository( new File( localRepoPath ) );
         return localArtifactRepository;
@@ -243,9 +243,9 @@
 
         SettingsBuildingRequest settingsBuildingRequest = new DefaultSettingsBuildingRequest();
 
-        settingsBuildingRequest.setGlobalSettingsFile( MavenCli.DEFAULT_GLOBAL_SETTINGS_FILE );
+        settingsBuildingRequest.setGlobalSettingsFile( org.apache.maven.cli.configuration.SettingsXmlConfigurationProcessor.DEFAULT_GLOBAL_SETTINGS_FILE );
 
-        settingsBuildingRequest.setUserSettingsFile( MavenCli.DEFAULT_USER_SETTINGS_FILE );
+        settingsBuildingRequest.setUserSettingsFile( org.apache.maven.cli.configuration.SettingsXmlConfigurationProcessor.DEFAULT_USER_SETTINGS_FILE );
 
         settingsBuildingRequest.getSystemProperties().putAll( System.getProperties() );
 
