Index: b/bnd.bnd
===================================================================
--- a/bnd.bnd
+++ b/bnd.bnd
@@ -8,6 +8,7 @@
 	org.osgi.impl.bundle.obr
 
 Include-Resource:       {LICENSE.txt},{about.html},{NOTICE.txt},{README}
+Bundle-SymbolicName: org.osgi.impl.bundle.bindex
 Bundle-Version: 		2.2
 Bundle-Vendor: 			OSGi Alliance
 Bundle-Description: 	Provides a command line utility to index a set of bundles. This code can be run from the command line.
@@ -27,4 +28,4 @@
 Export-Package: org.osgi.service.bindex
 Import-Package: org.apache.tools.ant;resolution:=optional,\
 	org.apache.tools.ant.types;resolution:=optional,\
-	*
\ No newline at end of file
+	*
