Description: Ignore the sporadic failures of IMAPIdleUntaggedResponseTest
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/769267
--- a/mail/src/test/java/com/sun/mail/imap/IMAPIdleUntaggedResponseTest.java
+++ b/mail/src/test/java/com/sun/mail/imap/IMAPIdleUntaggedResponseTest.java
@@ -44,6 +44,7 @@
     public Timeout deadlockTimeout = Timeout.seconds(20);
 
     @Test
+    @org.junit.Ignore
     public void test() {
         TestServer server = null;
         try {
