Description: Disable (Ignore) SMTPWriteTimeoutTest due to network access
Author: tony mancill <tmancill@debian.org>
Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/770744
--- a/mail/src/test/java/com/sun/mail/smtp/SMTPWriteTimeoutTest.java
+++ b/mail/src/test/java/com/sun/mail/smtp/SMTPWriteTimeoutTest.java
@@ -31,6 +31,7 @@
 
 import org.junit.Test;
 import org.junit.Rule;
+import org.junit.Ignore;
 import org.junit.rules.Timeout;
 import static org.junit.Assert.fail;
 
@@ -45,6 +46,7 @@
 
     private static final int TIMEOUT = 200;	// write timeout, in millis
 
+    @Ignore("requires external network access; see Debian #770744")
     @Test
     public void test() throws Exception {
         TestServer server = null;
--- a/mail/src/test/java/com/sun/mail/util/WriteTimeoutSocketTest.java
+++ b/mail/src/test/java/com/sun/mail/util/WriteTimeoutSocketTest.java
@@ -37,6 +37,7 @@
 import com.sun.mail.test.TestSocketFactory;
 import com.sun.mail.test.TestSSLSocketFactory;
 
+import org.junit.Ignore;
 import org.junit.Test;
 import org.junit.Rule;
 import org.junit.rules.Timeout;
@@ -60,6 +61,7 @@
      * Test write timeouts with plain sockets.
      */
     @Test
+    @Ignore
     public void test() {
 	final Properties properties = new Properties();
 	properties.setProperty("mail.imap.host", "localhost");
@@ -71,6 +73,7 @@
      * Test write timeouts with custom socket factory.
      */
     @Test
+    @Ignore
     public void testSocketFactory() {
 	final Properties properties = new Properties();
 	properties.setProperty("mail.imap.host", "localhost");
@@ -87,6 +90,7 @@
      * Test write timeouts with SSL sockets.
      */
     @Test
+    @Ignore
     public void testSSL() {
 	final Properties properties = new Properties();
 	properties.setProperty("mail.imap.host", "localhost");
@@ -100,6 +104,7 @@
      * Test write timeouts with a custom SSL socket factory.
      */
     @Test
+    @Ignore
     public void testSSLSocketFactory() throws Exception {
 	final Properties properties = new Properties();
 	properties.setProperty("mail.imap.host", "localhost");
