From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Sun, 18 Dec 2011 00:46:54 -0500
Subject: Disable login sound by default

Forwarded: not-needed
---
 src/libcanberra-login-sound.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libcanberra-login-sound.desktop.in b/src/libcanberra-login-sound.desktop.in
index 8baa7e3..2ccd560 100644
--- a/src/libcanberra-login-sound.desktop.in
+++ b/src/libcanberra-login-sound.desktop.in
@@ -7,3 +7,4 @@ OnlyShowIn=GNOME;Unity;
 AutostartCondition=GSettings org.gnome.desktop.sound event-sounds
 X-GNOME-Autostart-Phase=Application
 X-GNOME-Provides=login-sound
+X-GNOME-Autostart-enabled=false
