From: Michael Terry <michael.terry@canonical.com>
Date: Fri, 1 Jul 2011 15:03:38 +0100
Subject: Clean up Startup Applications dialog by hiding default apps

Bug-Ubuntu: https://launchpad.net/bugs/803917
Forwarded: not-needed
---
 src/libcanberra-login-sound.desktop.in | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libcanberra-login-sound.desktop.in b/src/libcanberra-login-sound.desktop.in
index 2ccd560..67cb28a 100644
--- a/src/libcanberra-login-sound.desktop.in
+++ b/src/libcanberra-login-sound.desktop.in
@@ -8,3 +8,4 @@ AutostartCondition=GSettings org.gnome.desktop.sound event-sounds
 X-GNOME-Autostart-Phase=Application
 X-GNOME-Provides=login-sound
 X-GNOME-Autostart-enabled=false
+NoDisplay=true
