From: Bart Martens <bartm@debian.org>
Date: Sun, 9 Oct 2011 13:38:25 +0200
Subject: Fix FTBFS with binutils-gold

Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=555081
---
 src/Makefile.am | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/Makefile.am b/src/Makefile.am
index 56ed96d..dc3602f 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -342,6 +342,7 @@ libcanberra_gtk3_la_CFLAGS = \
 	$(GTK3_CFLAGS)
 libcanberra_gtk3_la_LIBADD = \
 	$(GTK3_LIBS) \
+	-lX11 \
 	libcanberra.la
 libcanberra_gtk3_la_LDFLAGS = \
 	-export-dynamic -version-info $(LIBCANBERRA_GTK_VERSION_INFO)
@@ -352,6 +353,8 @@ libcanberra_gtk3_module_la_CFLAGS = \
 	$(GTK3_CFLAGS)
 libcanberra_gtk3_module_la_LIBADD = \
 	$(GTK3_LIBS) \
+	-lX11 \
+	-lgmodule-2.0 \
 	libcanberra.la \
 	libcanberra-gtk3.la
 libcanberra_gtk3_module_la_LDFLAGS = \
@@ -380,6 +383,7 @@ libcanberra_gtk_la_CFLAGS = \
 	$(GTK_CFLAGS)
 libcanberra_gtk_la_LIBADD = \
 	$(GTK_LIBS) \
+	-lX11 \
 	libcanberra.la
 libcanberra_gtk_la_LDFLAGS = \
 	-export-dynamic -version-info $(LIBCANBERRA_GTK_VERSION_INFO)
@@ -390,6 +394,8 @@ libcanberra_gtk_module_la_CFLAGS = \
 	$(GTK_CFLAGS)
 libcanberra_gtk_module_la_LIBADD = \
 	$(GTK_LIBS) \
+	-lX11 \
+	-lgmodule-2.0 \
 	libcanberra.la \
 	libcanberra-gtk.la
 libcanberra_gtk_module_la_LDFLAGS = \
