Description: Test installed script when run under autopkgtest
Origin: vendor
Forwarded: not-needed
Author: Salvatore Bonaccorso <carnil@debian.org>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2020-10-31

--- a/t/11mimeinfo.t
+++ b/t/11mimeinfo.t
@@ -16,7 +16,7 @@
 
 my $empty_dir = File::Temp::tempdir();
 
-my $mimetype_file = File::Spec->catfile($Bin, '..', 'mimetype');
+my $mimetype_file = $ENV{AUTOPKGTEST_TMP} ? '/usr/bin/mimetype' : File::Spec->catfile($Bin, '..', 'mimetype');
 
 my %tests = (
     'mimeopen', 'application/x-perl',
