Description: skip a test when run under autopkgtest
Origin: vendor
Forwarded: not-needed
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-09-07

--- a/t/10-basics.t
+++ b/t/10-basics.t
@@ -20,7 +20,7 @@
 
 is_deeply([ read_lines($0) ], \@content, 'read_lines returns the right thing (chomp)');
 
-is_deeply([ read_dir('lib') ], [ 'File' ], 'read_dir appears to work');
+is_deeply([ read_dir('lib') ], [ 'File' ], 'read_dir appears to work') unless $ENV{AUTOPKGTEST_TMP};
 
 my ($fh, $filename) = tempfile(UNLINK => 1);
 
