Description: Restore the compatibility with the previous releases
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/src/main/java/com/googlecode/javaewah/EWAHCompressedBitmap.java
+++ b/src/main/java/com/googlecode/javaewah/EWAHCompressedBitmap.java
@@ -2133,6 +2133,10 @@
      */
     public static final int WORD_IN_BITS = 64;
 
+    /** The Constant wordinbits represents the number of bits in a long. */
+    @Deprecated
+    public static final int wordinbits = 64;
+
     static final long serialVersionUID = 1L;
 
 }
