--- a/tools/cpp/src/cpp-build/generate_geocoding_data.cc
+++ b/tools/cpp/src/cpp-build/generate_geocoding_data.cc
@@ -91,6 +91,7 @@
 
   const std::string& name() const { return name_; }
   DirEntryKinds kind() const { return kind_; }
+  bool operator<(const DirEntry& d) { return name_ < d.name() || kind_ < d.kind(); }
 
  private:
   std::string name_;
@@ -118,6 +119,7 @@
     errno = 0;
     entry = readdir(dir);
     if (entry == NULL) {
+      std::sort(entries->begin(), entries->end());
       return errno == 0;
     }
     if (strcmp(entry->d_name, ".") == 0 || strcmp(entry->d_name, "..") == 0) {
@@ -125,6 +127,7 @@
     }
     const string entry_path = path + "/" + entry->d_name;
     if (stat(entry_path.c_str(), &entry_stat)) {
+      std::sort(entries->begin(), entries->end());
       return false;
     }
     DirEntryKinds kind = kFile;
