Description: use Debian packaging of abseil during build
Author: tony mancill <tmancill@debian.org>
Forwarded: not-needed

--- a/tools/cpp/CMakeLists.txt
+++ b/tools/cpp/CMakeLists.txt
@@ -28,27 +28,33 @@
 include (gtest.cmake)
 include (FetchContent)
 
-# Downloading the abseil sources.
-FetchContent_Declare(
-    abseil-cpp
-    GIT_REPOSITORY  https://github.com/abseil/abseil-cpp.git
-    GIT_TAG         origin/master
-)
-
-# Building the abseil binaries
-FetchContent_GetProperties(abseil-cpp)
-if (NOT abseil-cpp_POPULATED)
-    FetchContent_Populate(abseil-cpp)
-endif ()
-
-if (NOT abseil-cpp_POPULATED)
-   message (FATAL_ERROR "Could not build abseil-cpp binaries.")
-endif ()
+# Downloading and building abseil are disabled for the Debian packaging
+# in favor of https://tracker.debian.org/pkg/abseil.
+#
+## Downloading the abseil sources.
+#FetchContent_Declare(
+#    abseil-cpp
+#    GIT_REPOSITORY  https://github.com/abseil/abseil-cpp.git
+#    GIT_TAG         origin/master
+#)
+#
+## Building the abseil binaries
+#FetchContent_GetProperties(abseil-cpp)
+#if (NOT abseil-cpp_POPULATED)
+#    FetchContent_Populate(abseil-cpp)
+#endif ()
+#
+#if (NOT abseil-cpp_POPULATED)
+#   message (FATAL_ERROR "Could not build abseil-cpp binaries.")
+#endif ()
 
 # Safeguarding against any potential link errors as mentioned in
 # https://github.com/abseil/abseil-cpp/issues/225
 set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
-add_subdirectory(${abseil-cpp_SOURCE_DIR} ${abseil-cpp_BINARY_DIR})
+#add_subdirectory(${abseil-cpp_SOURCE_DIR} ${abseil-cpp_BINARY_DIR})
+
+set(absl "/usr/lib/*/cmake/absl")
+find_package(absl REQUIRED)
 
 find_or_build_gtest ()
 set (
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -26,6 +26,9 @@
 set (libphonenumber_VERSION_MINOR 12)
 set (libphonenumber_VERSION_PATCH 0)
 
+set(absl "/usr/lib/*/cmake/absl")
+find_package(absl REQUIRED)
+
 #if (32BIT)
 #  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32")
 #  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32")
