--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -26,12 +26,12 @@
 set (libphonenumber_VERSION_MINOR 12)
 set (libphonenumber_VERSION_PATCH 0)
 
-if (32BIT)
-  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32")
-  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32")
-  set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -m32")
-  set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -m32")
-endif ()
+#if (32BIT)
+#  set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -m32")
+#  set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -m32")
+#  set (CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -m32")
+#  set (CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -m32")
+#endif ()
 
 # Helper functions dealing with finding libraries and programs this library
 # depends on.
@@ -471,14 +471,14 @@
   list (APPEND LIBRARY_DEPS ${RE2_LIB})
 endif ()
 
-if (${USE_POSIX_THREAD} STREQUAL "ON" OR ((APPLE OR UNIX) AND ${USE_BOOST} STREQUAL "OFF" AND ${USE_STDMUTEX} STREQUAL "OFF"))
-  if(CMAKE_USE_PTHREADS_INIT)
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
-  endif()
-  if(CMAKE_THREAD_LIBS_INIT)
-    list (APPEND LIBRARY_DEPS ${CMAKE_THREAD_LIBS_INIT})
-  endif()
-endif ()
+#if (${USE_POSIX_THREAD} STREQUAL "ON" OR ((APPLE OR UNIX) AND ${USE_BOOST} STREQUAL "OFF" AND ${USE_STDMUTEX} STREQUAL "OFF"))
+#  if(CMAKE_USE_PTHREADS_INIT)
+#    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
+#  endif()
+#  if(CMAKE_THREAD_LIBS_INIT)
+#    list (APPEND LIBRARY_DEPS ${CMAKE_THREAD_LIBS_INIT})
+#  endif()
+#endif ()
 
 # Safeguarding against any potential link errors as mentioned in
 # https://github.com/abseil/abseil-cpp/issues/225
