From a3a2762105495f7a9e04ba7dd619e35805f74461 Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Sun, 14 Jun 2020 14:49:28 +0800
Subject: [PATCH 5/7] Force INI_HANDLER_LINENO

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index 5f13ccd..ddc571b 100644
--- a/meson.build
+++ b/meson.build
@@ -9,6 +9,7 @@ distro_install = get_option('distro_install')
 
 if distro_install
     pkg = import('pkgconfig')
+    arg_static += ['-DINI_HANDLER_LINENO=1']
 else
     if not get_option('multi-line_entries')
         arg_static += ['-DINI_ALLOW_MULTILINE=0']
-- 
2.29.2

