From b6f088c7fd1c7f602f9f0bd2095fe092206939bc Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Tue, 9 Feb 2021 19:44:21 +0800
Subject: [PATCH 6/7] Fix cpp type of INI_HANDLER_LINENO

---
 cpp/INIReader.cpp | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/cpp/INIReader.cpp b/cpp/INIReader.cpp
index 1bdac40..c203a86 100644
--- a/cpp/INIReader.cpp
+++ b/cpp/INIReader.cpp
@@ -12,6 +12,10 @@
 #include <algorithm>
 #include <cctype>
 #include <cstdlib>
+#ifdef INI_HANDLER_LINENO
+#undef INI_HANDLER_LINENO
+#endif
+#define INI_HANDLER_LINENO 0
 #include "../ini.h"
 #include "INIReader.h"
 
-- 
2.29.2

