From 097166a5b2ce34efd109ed47de94b9cc7bb51c83 Mon Sep 17 00:00:00 2001
From: yangfl <yangfl@users.noreply.github.com>
Date: Tue, 9 Feb 2021 20:07:22 +0800
Subject: [PATCH 7/7] Force include stdlib

---
 ini.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/ini.c b/ini.c
index 0c0318f..a483463 100644
--- a/ini.c
+++ b/ini.c
@@ -21,7 +21,6 @@ https://github.com/benhoyt/inih
 
 #include "ini.h"
 
-#if !INI_USE_STACK
 #if INI_CUSTOM_ALLOCATOR
 #include <stddef.h>
 void* ini_malloc(size_t size);
@@ -33,7 +32,6 @@ void* ini_realloc(void* ptr, size_t size);
 #define ini_free free
 #define ini_realloc realloc
 #endif
-#endif
 
 #define MAX_SECTION 50
 #define MAX_NAME 50
-- 
2.29.2

