--- ldp-docbook-stylesheets-0.0.20040321.orig/XSL/fo/tldp-print.xsl
+++ ldp-docbook-stylesheets-0.0.20040321/XSL/fo/tldp-print.xsl
@@ -3,7 +3,7 @@
 <xsl:stylesheet version="1.0"
 xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
 
-<xsl:import href="docbook.xsl"/>
+<xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/fo/docbook.xsl"/>
 
 <!-- Number all sections in the style of 'CH.S1.S2 Section Title' where
      CH is the chapter number,  S1 is the section number and S2 is the
--- ldp-docbook-stylesheets-0.0.20040321.orig/XSL/html/tldp-chapters.xsl
+++ ldp-docbook-stylesheets-0.0.20040321/XSL/html/tldp-chapters.xsl
@@ -3,7 +3,7 @@
 <xsl:stylesheet version="1.0"
 xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
 
-<xsl:import href="chunk.xsl"/>
+<xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/html/chunk.xsl"/>
 <xsl:import href="tldp-common.xsl"/>
 
 <!-- Generate a separate HTML page for each preface, chapter or
--- ldp-docbook-stylesheets-0.0.20040321.orig/XSL/html/tldp-one-page.xsl
+++ ldp-docbook-stylesheets-0.0.20040321/XSL/html/tldp-one-page.xsl
@@ -3,7 +3,7 @@
 <xsl:stylesheet version="1.0"
 xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
 
-<xsl:import href="docbook.xsl"/>
+<xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/html/docbook.xsl"/>
 <xsl:import href="tldp-common.xsl"/>
 
 <!-- This set of customizations is used to generate the entire XML
--- ldp-docbook-stylesheets-0.0.20040321.orig/XSL/html/tldp-sections.xsl
+++ ldp-docbook-stylesheets-0.0.20040321/XSL/html/tldp-sections.xsl
@@ -3,7 +3,7 @@
 <xsl:stylesheet version="1.0"
 xmlns:xsl="http://www.w3.org/1999/XSL/Transform">
 
-<xsl:import href="chunk.xsl"/>
+<xsl:import href="/usr/share/xml/docbook/stylesheet/nwalsh/html/chunk.xsl"/>
 <xsl:import href="tldp-common.xsl"/>
 
 <!-- Generate a separate HTML page for each section.  Doing this
