Description: Make crc32 error message unambiguous
Bug-Debian: https://bugs.debian.org/762454
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=118886
Author: Florian Schlichting <fsfs@debian.org>
Reviewed-by: Salvatore Bonaccorso <carnil@debian.org>
Last-Update: 2017-12-20

--- a/script/crc32
+++ b/script/crc32
@@ -21,7 +21,7 @@
     }
     my $fh = FileHandle->new();
     if ( !$fh->open( $file, 'r' ) ) {
-        warn "$0: $!\n";
+        warn "$0: Cannot open $file: $!\n";
         next;
     }
     binmode($fh);
