Description: Disable doclint by default when building a Debian package
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/src/main/org/apache/tools/ant/taskdefs/Javadoc.java
+++ b/src/main/org/apache/tools/ant/taskdefs/Javadoc.java
@@ -1864,6 +1864,11 @@
             log("Debian build on Java 9+ detected: Adding the --ignore-source-errors option");
         }
 
+        if (doclet == null && LanguageLevel.isDebianBuild() && JavaEnvUtils.isAtLeastJavaVersion("9")) {
+            toExecute.createArgument().setValue("-Xdoclint:none");
+            log("Debian build on Java 9+ detected: Adding the -Xdoclint:none option");
+        }
+
         doLinkSource(toExecute);
         doNoqualifier(toExecute);
 		
