--- a/jnlp/core.jnlp
+++ b/jnlp/core.jnlp
@@ -13,15 +13,14 @@
       <all-permissions/> 
   </security> 
   <resources> 
-    <jar href="AppleJavaExtensions.jar"/>
     <jar href="bcel.jar"/>
-    <jar href="dom4j-1.6.1.jar"/>
-    <jar href="asm-3.3.jar"/>
-    <jar href="asm-tree-3.3.jar"/>
-    <jar href="asm-commons-3.3.jar"/>
-    <jar href="jaxen-1.1.6.jar"/>
+    <jar href="dom4j.jar"/>
+    <jar href="asm.jar"/>
+    <jar href="asm-tree.jar"/>
+    <jar href="asm-commons.jar"/>
+    <jar href="jaxen.jar"/>
     <jar href="jFormatString.jar"/>
-    <jar href="commons-lang-2.6.jar"/>
+    <jar href="commons-lang.jar"/>
     <jar href="jsr305.jar"/>
   </resources> 
   <component-desc />
--- a/jnlp/findbugs.jnlp
+++ b/jnlp/findbugs.jnlp
@@ -18,15 +18,14 @@
   <resources> 
     <j2se version="1.5+" initial-heap-size="300m" max-heap-size="600m"/>
     <jar href="findbugs.jar"/>
-    <jar href="AppleJavaExtensions.jar"/>
     <jar href="bcel.jar"/>
-    <jar href="dom4j-1.6.1.jar"/>
-    <jar href="asm-3.3.jar"/>
-    <jar href="asm-tree-3.3.jar"/>
-    <jar href="asm-commons-3.3.jar"/>
-    <jar href="jaxen-1.1.6.jar"/>
+    <jar href="dom4j.jar"/>
+    <jar href="asm.jar"/>
+    <jar href="asm-tree.jar"/>
+    <jar href="asm-commons.jar"/>
+    <jar href="jaxen.jar"/>
     <jar href="jFormatString.jar"/>
-    <jar href="commons-lang-2.6.jar"/>
+    <jar href="commons-lang.jar"/>
     <jar href="jsr305.jar"/>
     <property name="findbugs.jaws" value="true"/>
     <property name="findbugs.noSummary" value="true"/>
