Index: findbugs-2.0.3/src/scripts/standard/fb
===================================================================
--- findbugs-2.0.3.orig/src/scripts/standard/fb	2011-12-20 16:14:10.000000000 +0800
+++ findbugs-2.0.3/src/scripts/standard/fb	2013-12-17 20:29:11.166178876 +0800
@@ -24,7 +24,7 @@
 debug_arg=''
 conservespace_arg=''
 workhard_arg=''
-user_props=''
+user_props='-Dswing.defaultlaf=com.sun.java.swing.plaf.gtk.GTKLookAndFeel'
 
 # Handle command line arguments.
 while [ $# -gt 0 ]; do
Index: findbugs-2.0.3/src/scripts/standard/findbugs
===================================================================
--- findbugs-2.0.3.orig/src/scripts/standard/findbugs	2013-12-17 17:59:30.746249087 +0800
+++ findbugs-2.0.3/src/scripts/standard/findbugs	2013-12-17 20:29:23.566178779 +0800
@@ -23,7 +23,7 @@
 debug_arg=''
 conservespace_arg=''
 workhard_arg=''
-user_props=''
+user_props='-Dswing.defaultlaf=com.sun.java.swing.plaf.gtk.GTKLookAndFeel'
 
 # Handle command line arguments.
 while [ $# -gt 0 ]; do
Index: findbugs-2.0.3/src/scripts/standard/findbugs2
===================================================================
--- findbugs-2.0.3.orig/src/scripts/standard/findbugs2	2009-07-31 10:23:38.000000000 +0800
+++ findbugs-2.0.3/src/scripts/standard/findbugs2	2013-12-17 20:34:15.814176494 +0800
@@ -19,7 +19,7 @@
 jvm_maxheap="-Xmx768m"
 jvm_ea=""
 jvm_conservespace=""
-jvm_user_props=""
+jvm_user_props="-Dswing.defaultlaf=com.sun.java.swing.plaf.gtk.GTKLookAndFeel"
 
 #
 # Process command line args until we hit one we don't recognize.
