--- a/etc/MANIFEST-findbugsGUI.MF
+++ b/etc/MANIFEST-findbugsGUI.MF
@@ -1,3 +1,3 @@
 Manifest-Version: 1.0
 Main-Class: edu.umd.cs.findbugs.LaunchAppropriateUI
-Class-Path: bcel-6.0-SNAPSHOT.jar dom4j-1.6.1.jar jaxen-1.1.6.jar asm-debug-all-6.0-SNAPSHOT.jar jsr305.jar jFormatString.jar commons-lang-2.6.jar plastic.jar
+Class-Path: bcel.jar dom4j.jar jaxen.jar asm.jar asm-tree.jar asm-commons.jar jsr305.jar jFormatString.jar commons-lang.jar plastic.jar
--- a/etc/MANIFEST-findbugs.MF
+++ b/etc/MANIFEST-findbugs.MF
@@ -1,3 +1,3 @@
 Manifest-Version: 1.0
 Main-Class: edu.umd.cs.findbugs.LaunchAppropriateUI
-Class-Path: bcel-6.0-SNAPSHOT.jar dom4j-1.6.1.jar jaxen-1.1.6.jar asm-debug-all-6.0-SNAPSHOT.jar jsr305.jar jFormatString.jar commons-lang-2.6.jar
+Class-Path: bcel.jar dom4j.jar jaxen.jar asm.jar asm-tree.jar asm-commons.jar jsr305.jar jFormatString.jar commons-lang.jar
