--- a/src/java/edu/umd/cs/findbugs/visitclass/BetterVisitor.java
+++ b/src/java/edu/umd/cs/findbugs/visitclass/BetterVisitor.java
@@ -56,7 +56,7 @@
 import org.apache.bcel.classfile.StackMapEntry;
 import org.apache.bcel.classfile.Synthetic;
 import org.apache.bcel.classfile.Unknown;
-import org.apache.bcel.classfile.Visitor;
+import org.apache.bcel.classfile.EmptyVisitor;
 
 /**
  * Fixedup of from org.apache.bcel.classfile.Visitor
@@ -64,7 +64,7 @@
  * @author <A HREF="http://www.cs.umd.edu/~pugh">William Pugh</A>
  * @version 980818
  */
-public abstract class BetterVisitor implements Visitor {
+public abstract class BetterVisitor extends EmptyVisitor {
 
     /**
      * clone() is overridden to change access control from protected to public.
--- a/src/java/edu/umd/cs/findbugs/ba/AbstractFrameModelingVisitor.java
+++ b/src/java/edu/umd/cs/findbugs/ba/AbstractFrameModelingVisitor.java
@@ -297,6 +297,9 @@
     public void visitStackConsumer(StackConsumer obj) {
     }
 
+    public void visitNameSignatureInstruction(NameSignatureInstruction obj) {
+    }
+
     /*
      * ----------------------------------------------------------------------
      * General instruction handlers
