--- a/src/java/edu/umd/cs/findbugs/visitclass/BetterVisitor.java
+++ b/src/java/edu/umd/cs/findbugs/visitclass/BetterVisitor.java
@@ -50,6 +50,7 @@
 import org.apache.bcel.classfile.LocalVariableTable;
 import org.apache.bcel.classfile.LocalVariableTypeTable;
 import org.apache.bcel.classfile.Method;
+import org.apache.bcel.classfile.MethodParameters;
 import org.apache.bcel.classfile.Signature;
 import org.apache.bcel.classfile.SourceFile;
 import org.apache.bcel.classfile.StackMap;
@@ -352,6 +353,11 @@
     }
 
     @Override
+    public void visitMethodParameters(MethodParameters obj) {
+      // TODO: should we do something here - implement visit(MethodParameters)
+    }
+
+    @Override
     public void visitSignature(Signature obj) {
         visit(obj);
     }
