--- a/src/java/edu/umd/cs/findbugs/detect/CheckRelaxingNullnessAnnotation.java
+++ b/src/java/edu/umd/cs/findbugs/detect/CheckRelaxingNullnessAnnotation.java
@@ -202,7 +202,8 @@
                         bug.addClassAndMethod(xmethod);
                         LocalVariableAnnotation lva = null;
                         if (localVariables != null) {
-                            for(LocalVariableNode lvn : localVariables) {
+                            for(Object obj : localVariables) {
+                                LocalVariableNode lvn = (LocalVariableNode)obj;
                                 if (lvn.index == i+1) {
                                     lva = new LocalVariableAnnotation(lvn.name, i+1, 0);
                                     lva.setDescription(LocalVariableAnnotation.PARAMETER_NAMED_ROLE);
