From: Debian/Kubuntu Qt/KDE Maintainers <debian-qt-kde@lists.debian.org>
Date: Wed, 13 Apr 2016 11:36:15 +0200
Subject: postgresql-data-checksums

---
 src/server/storage/dbconfigpostgresql.cpp | 1 +
 1 file changed, 1 insertion(+)

--- a/src/server/storage/dbconfigpostgresql.cpp
+++ b/src/server/storage/dbconfigpostgresql.cpp
@@ -495,7 +495,7 @@
         }
 #endif
         // call 'initdb --pgdata=/home/user/.local/share/akonadi/db_data'
-        execute(mInitDbPath, {QStringLiteral("--pgdata=%1").arg(mPgData), QStringLiteral("--encoding=UTF8"), QStringLiteral("--no-locale")});
+        execute(mInitDbPath, {QStringLiteral("--pgdata=%1").arg(mPgData), QStringLiteral("--data-checksums"), QStringLiteral("--encoding=UTF8"), QStringLiteral("--no-locale")});
     } else {
         const auto versions = checkPgVersion();
         if (versions.has_value() && (versions->clusterVersion < versions->pgServerVersion)) {
