Description: Adapt to changed output in junit 4.13
Author: Jerry James <loganjerry@gmail.com>
Origin: upstream, https://github.com/jmock-developers/jmock-library/commit/1c454c07
--- a/jmock-junit4/src/test/java/org/jmock/test/acceptance/junit4/JUnit4TestRunnerTests.java
+++ b/jmock-junit4/src/test/java/org/jmock/test/acceptance/junit4/JUnit4TestRunnerTests.java
@@ -47,9 +47,8 @@
     public void testDetectsNonPublicBeforeMethodsCorrectly() {
         listener.runTestIn(JUnit4TestWithNonPublicBeforeMethod.class);
         listener.assertTestFailedWith(Throwable.class);
-        assertEquals("should have detected non-public before method",
-                "Method before() should be public",
-                       listener.failure.getMessage());
+        assertTrue("should have detected non-public before method",
+                listener.failure.getMessage().contains("Method before() should be public"));
     }
     
     public void testAutoInstantiatesMocks() {
