Description: Adjust testsuite expectations, now that bunzip2 is fixed.
Author: Dimitri John Ledkov <xnox@ubuntu.com>
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1828282


--- busybox-1.30.1.orig/testsuite/bunzip2.tests
+++ busybox-1.30.1/testsuite/bunzip2.tests
@@ -562,7 +562,7 @@ if test "${0##*/}" = "bunzip2.tests"; th
     fi
 
     errout="`${bb}bunzip2 <bz2_issue_12.bz2 2>&1 >/dev/null`"
-    if test x"$errout:$?" = x"bunzip2: bunzip error -3:1"; then
+    if test x"$errout:$?" = x"bunzip2: bunzip error -5:1"; then
 	echo "PASS: $unpack: bz2_issue_12.bz2 corrupted example"
     else
 	echo "FAIL: $unpack: bz2_issue_12.bz2 corrupted example"
