Description: Makes the date of the @Generated annotations reproducible
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/processor/src/main/java/org/jboss/logging/processor/generator/model/ClassModelHelper.java
+++ b/processor/src/main/java/org/jboss/logging/processor/generator/model/ClassModelHelper.java
@@ -52,7 +52,8 @@
      */
     static String generatedDateValue() {
         final SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
-        return sdf.format(new Date());
+        sdf.setTimeZone(java.util.TimeZone.getTimeZone("UTC"));
+        return sdf.format(System.getenv("SOURCE_DATE_EPOCH") != null ? new Date(1000 * Long.valueOf(System.getenv("SOURCE_DATE_EPOCH"))) : new Date());
     }
 
     /**
