diff --git a/example/Makefile.am b/example/Makefile.am
index 877a5e2..3930e10 100644
--- a/example/Makefile.am
+++ b/example/Makefile.am
@@ -1,5 +1,5 @@
 
-AM_CPPFLAGS = -I$(top_srcdir)/include -DNDEBUG
+AM_CPPFLAGS = $(shell pkg-config --cflags mdds-2.0) -DNDEBUG
 
 EXTRA_PROGRAMS = \
 	flat-segment-tree \
diff --git a/configure.ac b/configure.ac
index f2ad3e8..588db4e 100644
--- a/configure.ac
+++ b/configure.ac
@@ -13,7 +13,6 @@ PACKAGE_TARNAME=AC_PACKAGE_TARNAME
 AC_SUBST(PACKAGE_TARNAME)
 
 OBJDIR=obj
-INCDIR=include
 MISCDIR=misc
 QUICKCHECKDIR=quickcheck
 AC_SUBST(OBJDIR)
@@ -134,11 +133,6 @@ AC_SUBST(CXXFLAGS_UNITTESTS)
 
 AC_CONFIG_FILES([
 Makefile
-include/Makefile
-include/mdds/Makefile
-include/mdds/multi_type_vector/Makefile
-include/mdds/multi_type_vector/aos/Makefile
-include/mdds/multi_type_vector/soa/Makefile
 example/Makefile
 test/Makefile
 test/gdb/Makefile
