Description: adjust location of gamemoderun for Debian
Author: Stephan Lachnit <stephanlachnit@protonmail.com>
Forwarded: not-needed
Last-Update: 2020-09-11

--- a/daemon/gamemode-tests.c
+++ b/daemon/gamemode-tests.c
@@ -268,7 +268,7 @@
 		/* Close stdout, we don't care if sh prints anything */
 		fclose(stdout);
 		/* Preload into sh and then kill it */
-		if (execl("/usr/bin/gamemoderun", "/usr/bin/gamemoderun", "sleep", "5", (char *)NULL) ==
+		if (execl("/usr/games/gamemoderun", "/usr/games/gamemoderun", "sleep", "5", (char *)NULL) ==
 		    -1) {
 			LOG_ERROR("failed to launch gamemoderun with execl: %s\n", strerror(errno));
 			return -1;
