Description: spelling error
Author: Yadd <yadd@debian.org>
Forwarded: no
Last-Update: 2022-01-20

--- a/packages/babel-minify/src/cli.js
+++ b/packages/babel-minify/src/cli.js
@@ -119,7 +119,7 @@
     --tdz                   Detect usages of variables in the Temporal Dead Zone
 
   Nested Options:
-    To use nested options (plugin specfic options) simply use the pattern
+    To use nested options (plugin specific options) simply use the pattern
     --pluginName.featureName.
 
     For example,
