From: =?utf-8?b?IkNoYW5nWmh1byBDaGVuICjpmbPmmIzlgKwpIg==?=
 <czchen@debian.org>
Date: Sun, 4 Nov 2018 22:38:18 +0800
Subject: Do not use venderized oniguruma

---
 Makefile.am | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/Makefile.am b/Makefile.am
index df020b1..bbb49eb 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -138,13 +138,6 @@ jq.1: $(srcdir)/jq.1.prebuilt
 endif
 
 
-### Build oniguruma
-
-if BUILD_ONIGURUMA
-libjq_la_LIBADD += modules/oniguruma/src/.libs/libonig.la
-SUBDIRS = modules/oniguruma
-endif
-
 AM_CFLAGS += $(onig_CFLAGS)
 
 ### Packaging
