From: =?utf-8?b?IkNoYW5nWmh1byBDaGVuICjpmbPmmIzlgKwpIg==?=
 <czchen@debian.org>
Date: Sun, 27 Jan 2019 20:50:11 +0800
Subject: Hardcode version to 1.6

---
 scripts/version | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/scripts/version b/scripts/version
index 0e8094c..ff9722c 100755
--- a/scripts/version
+++ b/scripts/version
@@ -1,10 +1,2 @@
 #!/bin/sh
-set -e
-cd `dirname "$0"`
-if git rev-parse --verify -q jq-1.0 > /dev/null 2>&1; then
-    git describe --tags --match 'jq-*' --dirty | sed 's/^jq-//'
-else
-    b=`git rev-parse --abbrev-ref HEAD`
-    c=`git describe --always --dirty`
-    echo "${b}-${c}"
-fi
+echo "1.6"
