Description: Fix uninitialized variables
Author: Reiner Herrmann <reiner@reiner-h.de>
Bug-Debian: https://bugs.debian.org/925722
--- jq-1.6.orig/src/jq_test.c
+++ jq-1.6/src/jq_test.c
@@ -54,8 +54,8 @@ static void test_err_cb(void *data, jv e
 }
 
 static void run_jq_tests(jv lib_dirs, int verbose, FILE *testdata) {
-  char prog[4096];
-  char buf[4096];
+  char prog[4096] = {0};
+  char buf[4096] = {0};
   struct err_data err_msg;
   int tests = 0, passed = 0, invalid = 0;
   unsigned int lineno = 0;
