From: =?utf-8?b?IkNoYW5nWmh1byBDaGVuICjpmbPmmIzlgKwpIg==?=
 <czchen@debian.org>
Date: Sun, 4 Nov 2018 21:34:10 +0800
Subject: Do not distribute COPYING, fixes lintian warning

Forwarded: not-needed
Last-Update: 2015-10-11
---
 Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index 5337577..cd16690 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -184,7 +184,7 @@ EXTRA_DIST = $(DOC_FILES) $(man_MANS) $(TESTS) $(TEST_LOG_COMPILER)     \
 # README.md is expected in Github projects, good stuff in it, so we'll
 # distribute it and install it with the package in the doc directory.
 docdir = ${datadir}/doc/${PACKAGE}
-dist_doc_DATA = README.md COPYING AUTHORS README
+dist_doc_DATA = AUTHORS README
 
 RELEASE ?= 1
 rpm: dist jq.spec
