From: =?utf-8?b?IkNoYW5nWmh1byBDaGVuICjpmbPmmIzlgKwpIg==?=
 <czchen@debian.org>
Date: Sun, 4 Nov 2018 21:34:10 +0800
Subject: patch Makefile to always generate the manpage.

Forwarded: not-needed
Last-Update: 2014-06-21
---
 Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 6344b4e..5337577 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -126,6 +126,8 @@ TESTS_ENVIRONMENT = NO_VALGRIND=$(NO_VALGRIND)
 ### Building the manpage
 
 man_MANS = jq.1
+.PHONY: real_docs
+ENABLE_DOCS=1
 if ENABLE_DOCS
 jq.1: $(srcdir)/docs/content/3.manual/manual.yml
 	$(AM_V_GEN) ( cd ${abs_srcdir}/docs; '$(BUNDLER)' exec rake manpage ) > $@ || { rm -f $@; false; }
