Subject: FTBFS with '.' removed from perl's @INC
Bug-Debian: https://bugs.debian.org/835181
From: Dominic Hargreaves <dom@earth.li>
Last-Update: 2016-09-04
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=134338
Bug: https://rt.cpan.org/Ticket/Display.html?id=134338

--- a/Makefile.PL
+++ b/Makefile.PL
@@ -199,7 +199,7 @@
 	my $fragment = $self->SUPER::tools_other (@_);

 	$fragment =~ s/^MOD_INSTALL\s*=\s*(.*?)-MExtUtils::Install
-		/MOD_INSTALL =$1-MMyInstall/msx;
+		/MOD_INSTALL =$1-I. -MMyInstall/msx;

 	return $fragment;
 }
