From ba4f472e7372dbf453444c7764d1c342e7af12b8 Mon Sep 17 00:00:00 2001
From: Kenichi Ishigaki <ishigaki@cpan.org>
Date: Thu, 2 Dec 2021 01:25:48 +0900
Subject: [PATCH] Tweak for 3.37.0

Bug: https://github.com/DBD-SQLite/DBD-SQLite/issues/92
Bug-Debian: https://bugs.debian.org/1003127

---
 t/51_table_column_metadata.t | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/t/51_table_column_metadata.t b/t/51_table_column_metadata.t
index f6a6d70..7c63b59 100644
--- a/t/51_table_column_metadata.t
+++ b/t/51_table_column_metadata.t
@@ -19,7 +19,7 @@ for my $call_func (@CALL_FUNCS) {
 		my $data = $dbh->$call_func(undef, 'foo', 'id', 'table_column_metadata');
 		ok $data && ref $data eq ref {}, "got a metadata";
 		ok $data->{auto_increment}, "id is auto incremental";
-		is $data->{data_type} => 'integer', "data type is correct";
+		is lc($data->{data_type}) => 'integer', "data type is correct";
 		ok $data->{primary}, "id is a primary key";
 		ok !$data->{not_null}, "id is not null";
 	}
-- 
2.34.1

