Author: Roland Rosenfeld <roland@debian.org>
Description: Fix spelling error in man page.
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=133851

--- a/lib/Net/SNMP.pm
+++ b/lib/Net/SNMP.pm
@@ -1865,7 +1865,7 @@ A reference to an array can be passed to
 define the translation mode on a per ASN.1 type basis.  The array is expected
 to contain a list of named argument pairs for each ASN.1 type that is to
 be modified.  The arguments in the list are applied in the order that they
-are passed in via the array.  Arguments at the end of the list supercede 
+are passed in via the array.  Arguments at the end of the list supersede 
 those passed earlier in the list.  The argument "-all" can be used to specify
 that the mode is to apply to all ASN.1 types.  Only the arguments for the 
 ASN.1 types that are to be modified need to be included in the list.
