--- a/test/lib/TestGyp.py
+++ b/test/lib/TestGyp.py
@@ -573,7 +573,8 @@ class TestGypMake(TestGypBase):
       message_target = 'all'
     else:
       message_target = target
-    kw['stdout'] = "make: Nothing to be done for '%s'.\n" % message_target
+    kw['stdout'] = r"(^|.*\n)make(\[\d+\])?: Nothing to be done for '%s'.\n.*" % message_target
+    kw['match'] = TestCmd.match_re_dotall
     return self.build(gyp_file, target, **kw)
   def run_built_executable(self, name, *args, **kw):
     """
