--- a/test/errors/gyptest-errors.py
+++ b/test/errors/gyptest-errors.py
@@ -43,7 +43,11 @@ stderr = (".*target0.*target1.*target2.*
 test.run_gyp('dependency_cycle.gyp', status=1, stderr=stderr,
              match=TestCmd.match_re_dotall)
 
-stderr = (".*file_cycle0.*file_cycle1.*file_cycle0.*")
+cycles = [
+  r'Cycle: file_cycle0\.gyp -> file_cycle1\.gyp -> file_cycle0\.gyp',
+  r'Cycle: file_cycle1\.gyp -> file_cycle0\.gyp -> file_cycle1\.gyp',
+]
+stderr = (r".*\n(" + r'|'.join(cycles) + r")")
 test.run_gyp('file_cycle0.gyp', status=1, stderr=stderr,
              match=TestCmd.match_re_dotall)
 
