--- a/test/generator-output/gyptest-top-all.py
+++ b/test/generator-output/gyptest-top-all.py
@@ -26,6 +26,7 @@ test.writable(test.workpath('src/subdir2
 test.writable(test.workpath('src/subdir3/build'), True)
 
 test.build('prog1.gyp', test.ALL, chdir='gypfiles')
+test.writable(test.workpath('src'), True)
 
 chdir = 'gypfiles'
 
--- a/test/generator-output/gyptest-actions.py
+++ b/test/generator-output/gyptest-actions.py
@@ -38,6 +38,7 @@ test.writable(test.workpath('relocate/ac
 test.writable(test.workpath('relocate/actions/subdir2/actions-out'), True)
 
 test.build('actions.gyp', test.ALL, chdir='relocate/gypfiles')
+test.writable(test.workpath('relocate/actions'), True)
 
 expect = """\
 Hello from program.c
--- a/test/generator-output/gyptest-copies.py
+++ b/test/generator-output/gyptest-copies.py
@@ -33,6 +33,7 @@ test.writable(test.workpath('relocate/co
 test.writable(test.workpath('relocate/copies/subdir/copies-out'), True)
 
 test.build('copies.gyp', test.ALL, chdir='relocate/gypfiles')
+test.writable(test.workpath('relocate/copies'), True)
 
 test.must_match(['relocate', 'copies', 'copies-out', 'file1'],
                 "file1 contents\n")
--- a/test/generator-output/gyptest-relocate.py
+++ b/test/generator-output/gyptest-relocate.py
@@ -32,6 +32,7 @@ test.writable(test.workpath('relocate/sr
 test.writable(test.workpath('relocate/src/subdir3/build'), True)
 
 test.build('prog1.gyp', test.ALL, chdir='relocate/gypfiles')
+test.writable(test.workpath('relocate/src'), True)
 
 chdir = 'relocate/gypfiles'
 
--- a/test/generator-output/gyptest-rules.py
+++ b/test/generator-output/gyptest-rules.py
@@ -31,6 +31,7 @@ test.writable(test.workpath('relocate/ru
 test.writable(test.workpath('relocate/rules/subdir2/rules-out'), True)
 
 test.build('rules.gyp', test.ALL, chdir='relocate/gypfiles')
+test.writable(test.workpath('relocate/rules'), True)
 
 expect = """\
 Hello from program.c
--- a/test/generator-output/gyptest-depth.py
+++ b/test/generator-output/gyptest-depth.py
@@ -31,6 +31,7 @@ test.writable(test.workpath('src/subdir2
 test.writable(test.workpath('src/subdir3/build'), True)
 
 test.build('prog1.gyp', test.ALL, chdir='gypfiles')
+test.writable(test.workpath('src'), True)
 
 chdir = 'gypfiles'
 
--- a/test/generator-output/gyptest-symlink.py
+++ b/test/generator-output/gyptest-symlink.py
@@ -35,6 +35,7 @@ test.run_gyp('deeper.gyp',
 
 chdir = 'symlink'
 test.build('deeper.gyp', test.ALL, chdir=chdir)
+test.writable(test.workpath('src'), True)
 
 if test.format == 'xcode':
   chdir = 'src/subdir2/deeper'
--- a/test/generator-output/gyptest-subdir2-deep.py
+++ b/test/generator-output/gyptest-subdir2-deep.py
@@ -24,6 +24,7 @@ test.run_gyp('deeper.gyp',
              chdir='src/subdir2/deeper')
 
 test.build('deeper.gyp', test.ALL, chdir='gypfiles')
+test.writable(test.workpath('src'), True)
 
 chdir = 'gypfiles'
 
