--- a/test/linux/gyptest-implicit-rpath.py
+++ b/test/linux/gyptest-implicit-rpath.py
@@ -23,7 +23,7 @@ if sys.platform.startswith('linux'):
 
   def GetRpaths(p):
     p = test.built_file_path(p, chdir=CHDIR)
-    r = re.compile(r'Library rpath: \[([^\]]+)\]')
+    r = re.compile(r'Library runpath: \[([^\]]+)\]')
     proc = subprocess.Popen(['readelf', '-d', p], stdout=subprocess.PIPE)
     o = proc.communicate()[0].decode('utf-8')
     assert not proc.returncode
--- a/test/linux/gyptest-target-rpath.py
+++ b/test/linux/gyptest-target-rpath.py
@@ -23,7 +23,7 @@ if sys.platform.startswith('linux'):
 
   def GetRpaths(p):
     p = test.built_file_path(p, chdir=CHDIR)
-    r = re.compile(r'Library rpath: \[([^\]]+)\]')
+    r = re.compile(r'Library runpath: \[([^\]]+)\]')
     proc = subprocess.Popen(['readelf', '-d', p], stdout=subprocess.PIPE)
     o = proc.communicate()[0].decode('utf-8')
     assert not proc.returncode
