--- a/test/assembly/src/lib1.S
+++ b/test/assembly/src/lib1.S
@@ -5,11 +5,21 @@
 #endif
 
 .globl IDENTIFIER(lib1_function)
+.type  IDENTIFIER(lib1_function), %function
 IDENTIFIER(lib1_function):
-#if !defined(PLATFORM_ANDROID)
+#if defined(__x86_64__) || defined(__amd64__) || defined(__i386__)
   movl $42, %eax
   ret
-#else /* Android (assuming ARM) */
-  mov r0, #42
+#elif defined(__arm__)
+  movs r0, #42
   bx lr
+#elif defined(__aarch64__)
+  mov w0, 42
+  ret
+#elif defined(__powerpc64__)
+  li 3, 42
+  blr
+#elif defined(__s390x__)
+  lhi  %r2, 42
+  br   %r14
 #endif
--- a/test/rules/src/subdir4/asm-function.assem
+++ b/test/rules/src/subdir4/asm-function.assem
@@ -5,6 +5,21 @@
 #endif
 
 .globl IDENTIFIER(asm_function)
+.type IDENTIFIER(asm_function), %function
 IDENTIFIER(asm_function):
+#if defined(__x86_64__) || defined(__amd64__) || defined(__i386__)
   movl $41, %eax
   ret
+#elif defined(__arm__)
+  movs r0, #41
+  bx lr
+#elif defined(__aarch64__)
+  mov w0, 41
+  ret
+#elif defined(__powerpc64__)
+  li 3, 41
+  blr
+#elif defined(__s390x__)
+  lhi  %r2, 41
+  br   %r14
+#endif
