Description: drop useless plugin
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2022-01-28

--- a/loupe/rollup.conf.js
+++ b/loupe/rollup.conf.js
@@ -1,7 +1,6 @@
-import coverage from 'rollup-plugin-istanbul'
 import common from '@rollup/plugin-commonjs'
 import resolve from '@rollup/plugin-node-resolve'
-import transform from 'rollup-plugin-babel'
+import transform from '@rollup/plugin-babel'
 
 const plugins = [
   common(),
@@ -41,6 +40,6 @@
       format: 'umd',
       exports: 'named',
     },
-    plugins: [...plugins, coverage({ exclude: 'test/*.js' })],
+    plugins: plugins,
   },
 ]
