Author: Reiner Herrmann <reiner@reiner-h.de>
Description: Fix build with Python3-enabled scons
Bug-Debian: https://bugs.debian.org/947556

--- create-resources.orig/SConstruct
+++ create-resources/SConstruct
@@ -2,17 +2,17 @@
 #prefix = "/usr/share"
 
 
-opts = Options()
-opts.Add(PathOption('PREFIX', 'Install prefix', '/usr'))
+opts = Variables()
+opts.Add(PathVariable('PREFIX', 'Install prefix', '/usr'))
 env = Environment(options = opts)
 
 #prefix=ARGUMENTS.get('prefix', '/usr/share')
 
 
-print "Preparing Create Resource Files ..."
+print("Preparing Create Resource Files ...")
 Export('env')
 
-print 'Preparing Package Documentation ...'
+print('Preparing Package Documentation ...')
 packagedocs = ['COPYING', 'INSTALL']
 installdir=env['PREFIX'] + '/share/doc/create'
 env.Install(installdir, packagedocs)
--- create-resources.orig/brushes/gimp/SConscript
+++ create-resources/brushes/gimp/SConscript
@@ -1,5 +1,5 @@
 Import('env')
-print 'Preparing Brushes (Gimp)...'
+print('Preparing Brushes (Gimp)...')
 brushesgimp = [
 '10x10square16.gbr',
 '10x10squareBlur16.gbr',
--- create-resources.orig/docs/SConscript
+++ create-resources/docs/SConscript
@@ -1,5 +1,5 @@
 Import('env')
-print 'Preparing Documentation ...'
+print('Preparing Documentation ...')
 docs = ['spec_0.1.1.html', 'styles.css']
 
 installdir=env['PREFIX'] + '/share/doc/create'
--- create-resources.orig/gradients/gimp/SConscript
+++ create-resources/gradients/gimp/SConscript
@@ -1,5 +1,5 @@
 Import('env')
-print 'Preparing Gradients (Gimp) ...'
+print('Preparing Gradients (Gimp) ...')
 gradgimp = [
 'Abstract_1.ggr',
 'Abstract_2.ggr',
--- create-resources.orig/gradients/karbon14/SConscript
+++ create-resources/gradients/karbon14/SConscript
@@ -1,5 +1,5 @@
 Import('env')
-print 'Preparing Gradients (Karbon14) ...'
+print('Preparing Gradients (Karbon14) ...')
 gradk14 = ['allcolors.kgr', 'simple.kgr']
 
 installdir=env['PREFIX'] + '/share/create/gradients/karbon14'
--- create-resources.orig/patterns/bitmap/16bit/SConscript
+++ create-resources/patterns/bitmap/16bit/SConscript
@@ -1,5 +1,5 @@
 Import('env')
-print 'Preparing Patterns (16 bit Bitmap) ...'
+print('Preparing Patterns (16 bit Bitmap) ...')
 patterns16bit = [
 '3dgreen.pat',
 'amethyst.pat',
--- create-resources.orig/patterns/bitmap/8bit/SConscript
+++ create-resources/patterns/bitmap/8bit/SConscript
@@ -1,5 +1,5 @@
 Import('env')
-print 'Preparing Patterns (8 bit Bitmap) ...'
+print('Preparing Patterns (8 bit Bitmap) ...')
 patterns8bit = [
 '3dgreen.pat',
 'amethyst.pat',
--- create-resources.orig/swatches/SConscript
+++ create-resources/swatches/SConscript
@@ -1,5 +1,5 @@
 Import('env')
-print 'Preparing Swatches ...'
+print('Preparing Swatches ...')
 swatches = ['Bears', 'Bgold', 'Blues', 'Borders', 'Browns_And_Yellows', 'Caramel', 'Cascade',
             'China', 'Coldfire', 'Cool_Colors', 'Cranes', 'Dark_pastels', 'Default', 'Ega', 'Firecode', 
 				'GNOME_HIG', 'Gold', 'Grayblue', 'Grays', 'GrayViolet', 'Greens', 'Hilite', 'Kahki',
