diff --git a/src/LiberationMono-Bold.sfd b/src/LiberationMono-Bold.sfd
index 47784dc..300e897 100644
--- a/src/LiberationMono-Bold.sfd
+++ b/src/LiberationMono-Bold.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 13598569]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1095817380
 ModificationTime: 1399616628
 PfmFamily: 17
diff --git a/src/LiberationMono-BoldItalic.sfd b/src/LiberationMono-BoldItalic.sfd
index 2b4b105..80e93f2 100644
--- a/src/LiberationMono-BoldItalic.sfd
+++ b/src/LiberationMono-BoldItalic.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 8302692]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1095817380
 ModificationTime: 1399621376
 PfmFamily: 17
diff --git a/src/LiberationMono-Italic.sfd b/src/LiberationMono-Italic.sfd
index 9ff3617..7e53cc0 100644
--- a/src/LiberationMono-Italic.sfd
+++ b/src/LiberationMono-Italic.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 1812088]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1095817380
 ModificationTime: 1399615336
 PfmFamily: 17
diff --git a/src/LiberationMono-Regular.sfd b/src/LiberationMono-Regular.sfd
index 3251b6c..52b3dcd 100644
--- a/src/LiberationMono-Regular.sfd
+++ b/src/LiberationMono-Regular.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 5879191]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1095817380
 ModificationTime: 1399615248
 PfmFamily: 17
diff --git a/src/LiberationSans-Bold.sfd b/src/LiberationSans-Bold.sfd
index 6153304..44abc8d 100644
--- a/src/LiberationSans-Bold.sfd
+++ b/src/LiberationSans-Bold.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 3251725]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627167
 PfmFamily: 17
diff --git a/src/LiberationSans-BoldItalic.sfd b/src/LiberationSans-BoldItalic.sfd
index f694262..0766c9a 100644
--- a/src/LiberationSans-BoldItalic.sfd
+++ b/src/LiberationSans-BoldItalic.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 951244]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627185
 PfmFamily: 17
diff --git a/src/LiberationSans-Italic.sfd b/src/LiberationSans-Italic.sfd
index 167705e..64dfa59 100644
--- a/src/LiberationSans-Italic.sfd
+++ b/src/LiberationSans-Italic.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 2032059]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627137
 PfmFamily: 17
diff --git a/src/LiberationSans-Regular.sfd b/src/LiberationSans-Regular.sfd
index 36528ff..daec3bf 100644
--- a/src/LiberationSans-Regular.sfd
+++ b/src/LiberationSans-Regular.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 16587066]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627105
 PfmFamily: 17
diff --git a/src/LiberationSansNarrow-Bold.sfd b/src/LiberationSansNarrow-Bold.sfd
index 04f47be..3aeb6ef 100644
--- a/src/LiberationSansNarrow-Bold.sfd
+++ b/src/LiberationSansNarrow-Bold.sfd
@@ -16,7 +16,7 @@ Layer: 1 1 "Fore"  0
 FSType: 0
 OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627334
 PfmFamily: 17
diff --git a/src/LiberationSansNarrow-BoldItalic.sfd b/src/LiberationSansNarrow-BoldItalic.sfd
index b42d4a4..eb71d11 100644
--- a/src/LiberationSansNarrow-BoldItalic.sfd
+++ b/src/LiberationSansNarrow-BoldItalic.sfd
@@ -16,7 +16,7 @@ Layer: 1 1 "Fore"  0
 FSType: 0
 OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627489
 PfmFamily: 17
diff --git a/src/LiberationSansNarrow-Italic.sfd b/src/LiberationSansNarrow-Italic.sfd
index 5cf6e6f..98d97ff 100644
--- a/src/LiberationSansNarrow-Italic.sfd
+++ b/src/LiberationSansNarrow-Italic.sfd
@@ -16,7 +16,7 @@ Layer: 1 1 "Fore"  0
 FSType: 0
 OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627309
 PfmFamily: 17
diff --git a/src/LiberationSansNarrow-Regular.sfd b/src/LiberationSansNarrow-Regular.sfd
index 7ab8cc6..7906b37 100644
--- a/src/LiberationSansNarrow-Regular.sfd
+++ b/src/LiberationSansNarrow-Regular.sfd
@@ -16,7 +16,7 @@ Layer: 1 1 "Fore"  0
 FSType: 0
 OS2Version: 1
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123372800
 ModificationTime: 1399627238
 PfmFamily: 17
diff --git a/src/LiberationSerif-Bold.sfd b/src/LiberationSerif-Bold.sfd
index 400f9f1..d32ed5c 100644
--- a/src/LiberationSerif-Bold.sfd
+++ b/src/LiberationSerif-Bold.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 11337752]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123398000
 ModificationTime: 1386048810
 PfmFamily: 17
diff --git a/src/LiberationSerif-BoldItalic.sfd b/src/LiberationSerif-BoldItalic.sfd
index f85ab72..1af6375 100644
--- a/src/LiberationSerif-BoldItalic.sfd
+++ b/src/LiberationSerif-BoldItalic.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 1085738]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123398000
 ModificationTime: 1385979924
 PfmFamily: 17
diff --git a/src/LiberationSerif-Italic.sfd b/src/LiberationSerif-Italic.sfd
index b827082..c0f6439 100644
--- a/src/LiberationSerif-Italic.sfd
+++ b/src/LiberationSerif-Italic.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 13298560]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123398000
 ModificationTime: 1385979879
 PfmFamily: 17
diff --git a/src/LiberationSerif-Regular.sfd b/src/LiberationSerif-Regular.sfd
index 81e2b09..d21f811 100644
--- a/src/LiberationSerif-Regular.sfd
+++ b/src/LiberationSerif-Regular.sfd
@@ -17,7 +17,7 @@ XUID: [1021 131 222397055 16509670]
 FSType: 0
 OS2Version: 3
 OS2_WeightWidthSlopeOnly: 0
-OS2_UseTypoMetrics: 1
+OS2_UseTypoMetrics: 0
 CreationTime: 1123398000
 ModificationTime: 1386048774
 PfmFamily: 17
