Change GObject Introspection files to a fixed API version
--- a/src/libical-glib/CMakeLists.txt
+++ b/src/libical-glib/CMakeLists.txt
@@ -118,17 +118,13 @@
   endforeach()
 
   include(GObjectIntrospectionMacros)
-  set(LIBICAL_GLIB_GIR_VERSION_STRING "${LIBICAL_LIB_MAJOR_VERSION}.${LIBICAL_LIB_MINOR_VERSION}")
+  set(LIBICAL_GLIB_GIR_VERSION_STRING "${LIBICAL_LIB_MAJOR_VERSION}.0")
+  set(LIB_VERSION "${LIBICAL_LIB_MAJOR_VERSION}_0")
 
   set(GObjectIntrospection_GIRS)
   set(GObjectIntrospection_SCANNER_FLAGS)
   set(GObjectIntrospection_SCANNER_ARGS)
   set(GObjectIntrospection_COMPILER_ARGS)
-  set(LIB_VERSION)
-
-  set(LIB_VERSION ${LIBICAL_GLIB_GIR_VERSION_STRING})
-  string(REPLACE "." "_" LIB_VERSION "${LIB_VERSION}")
-  string(REPLACE "-" "_" LIB_VERSION "${LIB_VERSION}")
 
   set(GObjectIntrospection_SCANNER_ARGS
     --add-include-path=${CMAKE_CURRENT_BINARY_DIR}
@@ -139,8 +135,8 @@
     --add-include-path=${CMAKE_SOURCE_DIR}/src/libical
     --library-path=${LIBRARY_OUTPUT_PATH}
     --identifier-prefix=ICal
-    --pkg-export libical-glib-${LIBICAL_LIB_VERSION_STRING}
-    --c-include="libical-glib/libical-glib.h"
+    --pkg-export libical-glib
+    --c-include=libical-glib/libical-glib.h
     --cflags-begin
     -I${CMAKE_BINARY_DIR}/src
     -I${CMAKE_BINARY_DIR}/src/libical
--- a/src/libical/CMakeLists.txt
+++ b/src/libical/CMakeLists.txt
@@ -165,20 +165,23 @@
   #set(GObjectIntrospection_SCANNER_ARGS "--add-include-path=${CMAKE_CURRENT_SOURCE_DIR}")
   #set(GObjectIntrospection_COMPILER_ARGS "--includedir=${CMAKE_CURRENT_SOURCE_DIR}")
 
-  set(LIB_VERSION ${LIBICAL_LIB_VERSION_STRING})
-  string(REPLACE "." "_" LIB_VERSION "${LIB_VERSION}")
-  string(REPLACE "-" "_" LIB_VERSION "${LIB_VERSION}")
-
-  set(libical_${LIB_VERSION}_gir "libical")
-  set(libical_${LIB_VERSION}_gir_VERSION ${LIBICAL_LIB_VERSION_STRING})
-  set(GObjectIntrospection_SCANNER_ARGS -I${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_BINARY_DIR}/src/libical/ical.h)
-  set(libical_${LIB_VERSION}_gir_LIBRARY "ical")
-  set(libical_${LIB_VERSION}_gir_INCLUDES GObject-2.0)
+  set(LIBICAL_GIR_VERSION_STRING "${LIBICAL_LIB_MAJOR_VERSION}.0")
+  set(LIB_VERSION "${LIBICAL_LIB_MAJOR_VERSION}_0")
+  set(ICal_${LIB_VERSION}_gir "libical")
+  set(ICal_${LIB_VERSION}_gir_VERSION ${LIBICAL_GIR_VERSION_STRING})
+  set(GObjectIntrospection_SCANNER_ARGS
+    --c-include=libical/ical.h
+    --pkg-export libical
+    --identifier-prefix=ical
+    -I${CMAKE_CURRENT_SOURCE_DIR}
+    ${CMAKE_BINARY_DIR}/src/libical/ical.h)
+  set(ICal_${LIB_VERSION}_gir_LIBRARY "ical")
+  set(ICal_${LIB_VERSION}_gir_INCLUDES GObject-2.0)
   set(_includes ${_includes} "-L${LIBRARY_OUTPUT_PATH}")
-  set(libical_${LIB_VERSION}_gir_CFLAGS ${_includes})
-  set(libical_${LIB_VERSION}_gir_LIBS ical)
+  set(ICal_${LIB_VERSION}_gir_CFLAGS ${_includes})
+  set(ICal_${LIB_VERSION}_gir_LIBS ical)
 
-  list(APPEND GObjectIntrospection_GIRS libical-${LIBICAL_LIB_VERSION_STRING}.gir)
+  list(APPEND GObjectIntrospection_GIRS ICal-${LIBICAL_GIR_VERSION_STRING}.gir)
 
   gir_add_introspections(GObjectIntrospection_GIRS)
 endif()
