Description: fix lintian spelling-error-in-binary
Author: Nicolas Mora <nicolas@babelouest.org>
Index: libical3-3.0.3/src/libicalss/icalbdbset.c
===================================================================
--- libical3-3.0.3.orig/src/libicalss/icalbdbset.c	2018-02-27 10:34:25.000000000 -0500
+++ libical3-3.0.3/src/libicalss/icalbdbset.c	2018-07-21 11:14:29.474764152 -0400
@@ -547,7 +547,7 @@
             } else {
                 char *strError = db_strerror(ret);
 
-                icalerror_warn("icalbdbset_delete faild: ");
+                icalerror_warn("icalbdbset_delete failed: ");
                 icalerror_warn(strError);
                 tid->abort(tid);
                 return ICAL_FILE_ERROR;
@@ -642,7 +642,7 @@
             } else {
                 char *strError = db_strerror(ret);
 
-                icalerror_warn("icalbdbset_put faild: ");
+                icalerror_warn("icalbdbset_put failed: ");
                 icalerror_warn(strError);
                 tid->abort(tid);
                 return ICAL_FILE_ERROR;
