Author: Luca Boccassi <bluca@debian.org>
Forwarded: https://github.com/USNavalResearchLaboratory/protolib/pull/34
--- a/protolib/CMakeLists.txt
+++ b/protolib/CMakeLists.txt
@@ -85,10 +85,15 @@
 # Check for libraries
 set(THREADS_PREFER_PTHREAD_FLAG ON)
 find_package(Threads REQUIRED)
-find_package(LibXml2)
+find_package(PkgConfig)
+if(PkgConfig_FOUND)
+	pkg_check_modules(LibXml2 "libxml-2.0")
+else()
+	find_package(LibXml2)
+endif()
 if(LibXml2_FOUND)
-	include_directories(${LIBXML2_INCLUDE_DIR})
-	list(APPEND PLATFORM_LIBS ${LIBXML2_LIBRARIES})
+	include_directories(${LIBXML2_INCLUDE_DIR} ${LibXml2_INCLUDE_DIRS})
+	list(APPEND PLATFORM_LIBS ${LIBXML2_LIBRARIES} ${LibXml2_LIBRARIES})
 	list(APPEND PUBLIC_HEADER_FILES include/protoXml.h)
 	list(APPEND COMMON_SOURCE_FILES ${COMMON}/protoXml.cpp)
 endif()
