Author: Luca Boccassi <bluca@debian.org>
Forwarded: https://github.com/USNavalResearchLaboratory/protolib/pull/34
--- a/protolib/CMakeLists.txt
+++ b/protolib/CMakeLists.txt
@@ -20,11 +20,6 @@
 include(CheckIncludeFiles)
 include(CheckLibraryExists)
 
-check_include_files( pcap.h HAVE_PCAP_H)
-if(HAVE_PCAP_H)
-	#list(APPEND PLATFORM_SOURCE_FILES ${COMMON}/pcapCap.cpp)
-endif()
-
 check_include_files( sys/timerfd.h HAVE_TIMERFD_H)
 if(HAVE_TIMERFD_H)
 	list(APPEND PLATFORM_DEFINITIONS USE_TIMERFD)
@@ -88,6 +83,9 @@
 find_package(PkgConfig)
 if(PkgConfig_FOUND)
 	pkg_check_modules(LibXml2 "libxml-2.0")
+	if(CMAKE_HOST_SYSTEM_NAME MATCHES "GNU")
+		pkg_check_modules(LIBPCAP "libpcap")
+	endif()
 else()
 	find_package(LibXml2)
 endif()
@@ -98,6 +96,12 @@
 	list(APPEND COMMON_SOURCE_FILES ${COMMON}/protoXml.cpp)
 endif()
 
+if(LIBPCAP_FOUND)
+	include_directories(${LIBPCAP_INCLUDE_DIRS})
+	list(APPEND PLATFORM_LIBS ${LIBPCAP_LIBRARIES})
+	list(APPEND PLATFORM_SOURCE_FILES ${COMMON}/pcapCap.cpp)
+endif()
+
 if(PROTOKIT_ENABLE_WX)
         find_package(wxWidgets REQUIRED COMPONENTS core base)
 	if(wxWidgets_FOUND)
