Author: Luca Boccassi <bluca@debian.org>
Forwarded: https://github.com/USNavalResearchLaboratory/protolib/pull/34
--- a/protolib/CMakeLists.txt
+++ b/protolib/CMakeLists.txt
@@ -94,14 +94,19 @@
 	list(APPEND PLATFORM_LIBS ${LIBXML2_LIBRARIES} ${LibXml2_LIBRARIES})
 	list(APPEND PUBLIC_HEADER_FILES include/protoXml.h)
 	list(APPEND COMMON_SOURCE_FILES ${COMMON}/protoXml.cpp)
+	set(PROTOLIB_REQUIRES_PRIVATE "${PROTOLIB_REQUIRES_PRIVATE} libxml-2.0")
 endif()
 
 if(LIBPCAP_FOUND)
 	include_directories(${LIBPCAP_INCLUDE_DIRS})
 	list(APPEND PLATFORM_LIBS ${LIBPCAP_LIBRARIES})
 	list(APPEND PLATFORM_SOURCE_FILES ${COMMON}/pcapCap.cpp)
+	set(PROTOLIB_REQUIRES_PRIVATE "${PROTOLIB_REQUIRES_PRIVATE} libpcap")
 endif()
 
+# Note: cannot append, need to use PARENT_SCOPE
+set(REQUIRES_PRIVATE "${REQUIRES_PRIVATE} ${PROTOLIB_REQUIRES_PRIVATE}" PARENT_SCOPE)
+
 if(PROTOKIT_ENABLE_WX)
         find_package(wxWidgets REQUIRED COMPONENTS core base)
 	if(wxWidgets_FOUND)
