Author: Luca Boccassi <bluca@debian.org>
Forwarded: https://github.com/USNavalResearchLaboratory/norm/pull/23
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -121,8 +121,11 @@
 target_include_directories(norm PRIVATE ${CMAKE_CURRENT_BINARY_DIR})
 target_include_directories(norm PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include> $<INSTALL_INTERFACE:include>)
 
-if(BUILD_SHARED_LIBS AND WIN32)
-	target_compile_definitions(norm PUBLIC NORM_USE_DLL)
+if(BUILD_SHARED_LIBS)
+	set_target_properties(norm PROPERTIES SOVERSION 1)
+	if(WIN32)
+		target_compile_definitions(norm PUBLIC NORM_USE_DLL)
+	endif()
 endif()
 
 # Install target
