From: Antonio Terceiro <terceiro@debian.org>
Date: Mon, 23 Nov 2020 21:44:44 -0300
Subject: Fix for new version of markdown

---
 yarnlib/mdparser.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/yarnlib/mdparser.py b/yarnlib/mdparser.py
index 9adf057..e16f7b3 100644
--- a/yarnlib/mdparser.py
+++ b/yarnlib/mdparser.py
@@ -42,7 +42,7 @@ class GatherCodeBlocks(Treeprocessor):
 
     def run(self, root):
         h = html.parser.HTMLParser()
-        for child in root.getchildren():
+        for child in root:
             if child.tag == 'pre':
                 code = child.find('code')
                 text = h.unescape(code.text)
