From: Antonio Terceiro <terceiro@debian.org>
Date: Mon, 23 Nov 2020 21:47:57 -0300
Subject: Handle removal of html.parser.HTMLParser.unescape in python3.9

---
 yarnlib/mdparser.py | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/yarnlib/mdparser.py b/yarnlib/mdparser.py
index e16f7b3..4e9ba4c 100644
--- a/yarnlib/mdparser.py
+++ b/yarnlib/mdparser.py
@@ -17,7 +17,7 @@
 
 
 import logging
-import html.parser
+import html
 import markdown
 import io
 from markdown.treeprocessors import Treeprocessor
@@ -41,11 +41,10 @@ class GatherCodeBlocks(Treeprocessor):
         self.blocks = blocks
 
     def run(self, root):
-        h = html.parser.HTMLParser()
         for child in root:
             if child.tag == 'pre':
                 code = child.find('code')
-                text = h.unescape(code.text)
+                text = html.unescape(code.text)
                 self.blocks.append(text)
         return root
 
