--- hsqldb-1.8.0.10/bin/hsqldb-old	2009-08-17 21:03:33.000000000 +0200
+++ hsqldb-1.8.0.10/bin/hsqldb	2009-08-17 21:04:14.000000000 +0200
@@ -461,11 +461,11 @@
         echo "Successful shutdown ${SHUTDOWN_OPTION} (for the $TARGET_CLASS process)!"
         exit 0
     ;;
-    restart|restartcompacted)
+    restart|restartcompacted|force-reload)
         STOP_COMMAND=stop
         [ "$COMMAND" = restartcompacted ] && STOP_COMMAND=stopcompact
-        "${INVOC_PATH}"hsqldb $STOP_COMMAND || exit $?
-        exec "${INVOC_PATH}"/hsqldb start
+        $0 $STOP_COMMAND || exit $?
+        $0 start
     ;;
     *)
         echo "$SYNTAX_MSG" 1>&2
