diff -Nur hsqldb-1.8.0.2/src/org/hsqldb/util/ConnectionDialogCommon.java hsqldb-1.8.0.2.new/src/org/hsqldb/util/ConnectionDialogCommon.java
--- hsqldb-1.8.0.2/src/org/hsqldb/util/ConnectionDialogCommon.java	2005-12-11 14:47:11.000000000 +0100
+++ hsqldb-1.8.0.2.new/src/org/hsqldb/util/ConnectionDialogCommon.java	2005-12-11 14:47:25.000000000 +0100
@@ -178,7 +178,7 @@
                 recentSettings = new File(homedir, fileName);
 
                 if (!recentSettings.exists()) {
-                    JavaSystem.createNewFile(recentSettings);
+//                    JavaSystem.createNewFile(recentSettings);
 
                     // Changed back to what I recived from you
 //                  recentSettings.createNewFile();
