--- hsqldb-1.8.0.10.orig/bin/hsqldb
+++ hsqldb-1.8.0.10/bin/hsqldb
@@ -110,7 +110,7 @@
 SU_ECHO_SECS=1
 # File used as semaphore.  If file is removed, a running pid checker
 # process will exit.
-PIDCHECKER_FLAGFILE=/tmp/pidchecker.run
+PIDCHECKER_FLAGFILE=`mktemp /tmp/hsqldb.XXXXXXXX`
 # The following settings get overridden by optional setting in the config file.
 # Time to allow for JVM to die after all HSQLDB instances stopped.
 MAX_TERMINATE_SECS=1
