--- hsqldb-1.8.0.10.orig/bin/hsqldb
+++ hsqldb-1.8.0.10/bin/hsqldb
@@ -46,13 +46,12 @@
 # pidfile: /var/run/hsqldb.pid
 # config: /etc/sysconfig/hsqldb
 
-# This stuff for SuSE Linux.  Contributed by M. Bisping.
 ### BEGIN INIT INFO
 # Provides:          hsqldb-server
 # Required-Start:    $syslog $remote_fs $network $named
-# Required-Stop:
+# Required-Stop:     $remote_fs
-# Default-Start:     3 5
-# Default-Stop:      0 1 2 6
+# Default-Start:     2 3 4 5
+# Default-Stop:      0 1 6
 # Short-Description: HSQLDB-Server
 # Description:       Hsqldb, A High Performance Java Database Server
 ### END INIT INFO
