Description: Fix path in documentation generation
Author: Anton Gladky <gladk@debian.org>
Last-Update: 2015-06-16

Index: gnuplot/docs/Makefile.am
===================================================================
--- gnuplot.orig/docs/Makefile.am
+++ gnuplot/docs/Makefile.am
@@ -326,7 +326,7 @@ ${ELCS}: gnuplot-eldoc.el
 # that only appears to work with a non-generated texi file for input.
 
 gnuplot.info: gnuplot.texi
-	$(AM_V_GEN) $(MAKEINFO) -I$(srcdir) $< --no-split --output=$@
+	$(AM_V_GEN) $(MAKEINFO) -I$(srcdir) $< --no-split  --force --output=$@
 
 install-info: gnuplot.info
 	$(AM_V_at) $(top_srcdir)/mkinstalldirs $(DESTDIR)$(infodir)
