Description: remove beos-specific code
Author: Anton Gladky <gladk@debian.org>
Last-Update: 2015-06-16

Index: gnuplot/src/Makefile.am
===================================================================
--- gnuplot.orig/src/Makefile.am
+++ gnuplot/src/Makefile.am
@@ -77,11 +77,7 @@ getcolor_x11.o: getcolor.c
 bf_test_SOURCES = bf_test.c
 bf_test_LIBS = -lm
 
-if BUILD_SRC_BEOS_SUBDIR
-d_beos = beos
-endif
-
-SUBDIRS = $(d_beos) wxterminal qtterminal
+SUBDIRS = wxterminal qtterminal
 
 EXTRA_gnuplot_SOURCES = strftime.c vms.c vms.h
 
Index: gnuplot/configure.ac
===================================================================
--- gnuplot.orig/configure.ac
+++ gnuplot/configure.ac
@@ -190,7 +190,6 @@ GP_MSDOS
 GP_APPLE
 GP_BEOS
 GP_ALPHA
-AM_CONDITIONAL(BUILD_SRC_BEOS_SUBDIR, test x$build_beos_subdir = xyes)
 
 dnl _instead_ of -lm ...
 AC_CHECK_FUNC(sin,,[AC_CHECK_LIB(m,sin)])
@@ -1111,7 +1110,6 @@ AC_CONFIG_FILES([Makefile
            share/Makefile
            share/LaTeX/Makefile
            src/Makefile
-           src/beos/Makefile
            src/wxterminal/Makefile
            src/qtterminal/Makefile
            term/Makefile])
@@ -1197,10 +1195,6 @@ if test "$is_msdos" = yes; then
   AC_MSG_RESULT([    svga terminal (MSDOS/djgpp): yes])
 fi
 
-if test "$build_src_beos_subdir" = yes; then
-  AC_MSG_RESULT([    be terminal (BeOS): yes])
-fi
-
 if test "$with_caca" = yes; then
   AC_MSG_RESULT([    caca terminal: yes (EXPERIMENTAL)])
 else
