From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Thu, 12 Oct 2017 14:50:23 +0200
Subject: hurd: definition of PATH_MAX must be included separately
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

Signed-off-by: Ferenc Wágner <wferi@debian.org>
---
 tests/_syslog_override.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/_syslog_override.h b/tests/_syslog_override.h
index 7b254ae..923caea 100644
--- a/tests/_syslog_override.h
+++ b/tests/_syslog_override.h
@@ -24,7 +24,7 @@
 #ifndef QB_SYSLOG_OVERRIDE_H_DEFINED
 #define QB_SYSLOG_OVERRIDE_H_DEFINED
 
-#include <limits.h>
+#include "os_base.h"
 
 extern int _syslog_opened;
 extern int _syslog_option;
