Description: Missing blank line in POD
Author: Xavier Guimard <x.guimard@free.fr>
Forwarded: https://github.com/rhandom/perl-net-server/pull/11
Last-Update: 2017-10-26

--- libnet-server-perl-2.009.orig/lib/Net/Server/PreFork.pm
+++ libnet-server-perl-2.009/lib/Net/Server/PreFork.pm
@@ -678,6 +678,7 @@ This hook is called when a dead child is
 A child is considered dead when the pid does no longer exist.
 This hook could be used to cleanup possible temporary files
 or locks left over by a dead child.
+
 =back
 
 =head1 HOT DEPLOY
