Description: corrects SIGCHLD vs. SIGCLD confusion
Origin: vendor
Bug-Debian: https://bugs.debian.org/708180
Forwarded: https://rt.cpan.org/Public/Bug/Display.html?id=85308
Reviewed-By: Xavier Guimard <x.guimard@free.fr>
Last-Update: 2017-10-26

--- a/lib/Net/Server/SIG.pm
+++ b/lib/Net/Server/SIG.pm
@@ -36,6 +36,9 @@
     my $code_ref = shift;
     my $ref = ref($code_ref);
 
+    # Added by Debian maintainer to close #708180
+    $sig = 'CHLD' if ($sig eq 'CLD');
+
     if (! $ref) {
         if ($code_ref eq 'DEFAULT') {
             delete $_SIG{$sig};
@@ -62,6 +65,10 @@
 sub check_sigs {
     my @found;
     foreach my $sig (keys %_SIG){
+
+    # Added by Debian maintainer to close #712714
+    $sig = 'CHLD' if ($sig eq 'CLD');
+
         next if ! $_SIG{$sig};
         $_SIG{$sig} = 0;
         push @found, $sig;
@@ -72,6 +79,10 @@
 
 sub sig_is_registered {
     my $sig = shift;
+
+    # Added by Debian maintainer to close #712714
+    $sig = 'CHLD' if ($sig eq 'CLD');
+
     return $_SIG_SUB{$sig};
 }
 
