From: Markus Koschany <apo@debian.org>
Date: Mon, 22 Jan 2018 23:49:10 +0100
Subject: java9

---
 src/bin/jpip/CMakeLists.txt | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/bin/jpip/CMakeLists.txt b/src/bin/jpip/CMakeLists.txt
index 301d885..b412510 100644
--- a/src/bin/jpip/CMakeLists.txt
+++ b/src/bin/jpip/CMakeLists.txt
@@ -61,10 +61,10 @@ find_package(Java 1.5 COMPONENTS Development) # javac, jar
 
 # User can override this:
 if(NOT DEFINED JAVA_SOURCE_VERSION)
-  set(JAVA_SOURCE_VERSION 1.6)
+  set(JAVA_SOURCE_VERSION 1.7)
 endif()
 if(NOT DEFINED JAVA_TARGET_VERSION)
-  set(JAVA_TARGET_VERSION 1.6)
+  set(JAVA_TARGET_VERSION 1.7)
 endif()
 
 # Only build the java viewer if dev is found:
