From: jemptymethod <jempty...@gmail.com>
Date: Fri, 4 Sep 2009 11:21:19 -0700
Subject: Fix the exception 'can't read "state(sock)": no such element in
 array'

Origin: http://groups.google.com/group/tkhtml3/browse_thread/thread/8e186f9355255d29
---
 hv/hv3_http.tcl | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hv/hv3_http.tcl b/hv/hv3_http.tcl
index 0f99f02..5629aed 100644
--- a/hv/hv3_http.tcl
+++ b/hv/hv3_http.tcl
@@ -626,7 +626,7 @@ if 0 {
     upvar 0 $token state
     catch {
       close $state(sock)
-      unset state(sock)
+#      unset state(sock)
     }
     eval [linsert $args 0 ::http::FinishReal $token]
   }
