From: Ole Streicher <olebole@debian.org>
Date: Fri, 24 Feb 2012 12:25:44 +0100
Subject: Updated bookmarks: freshmeat->freecode; CVSTrac->fossil

---
 hv/hv3_bookmarks.tcl | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hv/hv3_bookmarks.tcl b/hv/hv3_bookmarks.tcl
index eea3f1a..73548d8 100644
--- a/hv/hv3_bookmarks.tcl
+++ b/hv/hv3_bookmarks.tcl
@@ -155,8 +155,8 @@ pressing enter.
   "INSTRUCTIONS"                http://tkhtml.tcl.tk 1
   "Hv3/Tkhtml3 Home page"       http://tkhtml.tcl.tk 0
   "Hv3/Tkhtml3 Mailing List"    http://groups.google.com/group/tkhtml3 0
-  "Hv3/Tkhtml3 CVSTrac"         http://tkhtml.tcl.tk/cvstrac/timeline 0
-  "Hv3 @ freshmeat.net"         http://freshmeat.net/hv3 0
+  "Hv3/Tkhtml3 Repository"      http://tkhtml.tcl.tk/fossil 0
+  "Hv3 @ Freecode"              http://freecode.com/projects/hv3 0
         } {
           if {$C} {
             set zSnapshot "
