Description: don't try upstream install methods
 .
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Last-Update: 2014-01-24

---

--- a/Makefile
+++ b/Makefile
@@ -64,22 +64,22 @@
 	npm test
 
 $(JS_HINT):
-	npm install jshint
+#	npm install jshint
 
 $(CSS_MIN):
-	npm install clean-css
+#	npm install clean-css
 
 $(JS_MIN):
-	npm install uglify-js
+#	npm install uglify-js
 
 $(D3):
-	npm install d3
+#	npm install d3
 
 $(JSDOM):
-	npm install jsdom
+#	npm install jsdom
 
 $(NODEUNIT):
-	npm install nodeunit
+#	npm install nodeunit
 
 rickshaw.css: $(CSS_FILES)
 	cat $(CSS_FILES) > rickshaw.css
