Description: Ensure horizon test settings are imported.
Author: Corey Bryant <corey.bryant@canonical.com>

--- a/heat_dashboard/test/settings.py
+++ b/heat_dashboard/test/settings.py
@@ -13,6 +13,9 @@
 
 # Default to Horizons test settings to avoid any missing keys
 
+# Default to Horizons test settings to avoid any missing keys
+from horizon.test.settings import *  # noqa
+
 import openstack_dashboard.enabled  # noqa: F811
 from openstack_dashboard.test.settings import *  # noqa: F403,H303
 from openstack_dashboard.utils import settings
