From 87d13fc7c00c2b74674be56b5bc4e204a65caeb9 Mon Sep 17 00:00:00 2001
From: "H.J. Lu" <hjl.tools@gmail.com>
Date: Wed, 29 Jan 2020 07:57:53 -0800
Subject: [PATCH 01/13] Add .gitignore

---
 .gitignore | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 000000000..d2b12799a
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1,17 @@
+/COPYING
+Makefile.in
+/aclocal.m4
+/autom4te.cache
+/compile
+/config.in
+/configure
+doc/gmp.info*
+doc/mdate-sh
+doc/stamp-vti
+doc/texinfo.tex
+doc/version.texi
+/install-sh
+/ltmain.sh
+/missing
+/test-driver
+/ylwrap
-- 
2.27.0

