Description: spelling errors
Author: Xavier Guimard <x.guimard@free.fr
Bug: https://rt.cpan.org/Ticket/Display.html?id=83824
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=83824
Last-Update: 2013-03-08

--- a/lib/Class/Data/Inheritable.pm
+++ b/lib/Class/Data/Inheritable.pm
@@ -54,7 +54,7 @@
 Class::Data::Inheritable is for creating accessor/mutators to class
 data.  That is, if you want to store something about your class as a
 whole (instead of about a single object).  This data is then inherited
-by your subclasses and can be overriden.
+by your subclasses and can be overridden.
 
 For example:
 
@@ -87,12 +87,12 @@
 
 However, should Raygun decide to set its own Suitcase() it has now
 "overridden" Pere::Ubu and is on its own, just like if it had
-overriden a method:
+overridden a method:
 
   # Raygun has an orange suitcase, Pere::Ubu's is still Blue.
   Raygun->Suitcase('Orange');
 
-Now that Raygun has overridden Pere::Ubu futher changes by Pere::Ubu
+Now that Raygun has overridden Pere::Ubu further changes by Pere::Ubu
 no longer effect Raygun.
 
   # Raygun still has an orange suitcase, but Pere::Ubu is using Samsonite.
