From: Roland Rosenfeld <roland@debian.org>
Subject: Adds HTML entities to the german.aff file.
Date: Mon, 13 Mar 2000 20:13:27 +0100
Forwarded: Björn Jacke <bjoern.jacke@gmx.de> on Sat, 19 Feb 2000 00:35:17 +0100

--- a/ispell/de_DE.aff.in
+++ b/ispell/de_DE.aff.in
@@ -257,18 +257,18 @@ altstringchar	SS	SS
 #
 # HTML/SGML/XML `entities' (useful if you have the Ispell-HTML patch)
 #
-#altstringtype "HTML" "TeX" ".html" ".htm" ".sgml" ".xml"
+altstringtype "HTML" "TeX" ".html" ".htm" ".sgml" ".xml"
 
-#altstringchar	\&auml\;	a\"
-#altstringchar	\&Auml\;	A\"
-#altstringchar	\&ouml\;	o\"
-#altstringchar	\&Ouml\;	O\"
-#altstringchar	\&uuml\;	u\"
-#altstringchar	\&Uuml\;	U\"
-#altstringchar	\&szlig\;	sS
-#altstringchar	SS	Ss
-#altstringchar	ss	ss
-#altstringchar	SS	SS
+altstringchar	&auml;	a\"
+altstringchar	&Auml;	A\"
+altstringchar	&ouml;	o\"
+altstringchar	&Ouml;	O\"
+altstringchar	&uuml;	u\"
+altstringchar	&Uuml;	U\"
+altstringchar	&szlig;	sS
+altstringchar	SS	Ss
+altstringchar	ss	ss
+altstringchar	SS	SS
 
 #
 # N/Troff with -ms/-me macros packages
