From: Roland Rosenfeld <roland@debian.org>
Date: Thu, 03 Jan 2019 10:40:10 +0100
Subject: Export locale variables to make build reproducible.
 Without this I observed problems with Varieté expanded wrong.
Forwarded: Bjoern Jacke <bjoern@j3e.de> on Fri, 4 Jan 2019 13:33:35 +0100

--- a/Makefile
+++ b/Makefile
@@ -119,9 +119,10 @@ SUBDIRS = testdir
 # (like for de_DE) would be used, we don't need that anyway. See
 # http://mail.nl.linux.org/linux-utf8/2003-11/msg00041.html
 
-LANG=C
-LC_ALL=C
-LC_COLLATE=C
+export LANG=C
+export LC_ALL=C
+export LC_COLLATE=C
+export LANGUAGE=en_US:en
 
 
 #$(addprefix dicts/,$(WORTE)): unpack
