From: Roland Rosenfeld <roland@debian.org>
Subject: dic2iso now 7bit clean and supports more characters.
Date: Fri, 08 Jan 2021 19:31:46 +0100
Forwarded: Bjoern JACKE <bjoern@j3e.de> on Fri, 8 Jan 2021 21:11:49 +0100

--- a/bin/dic2iso
+++ b/bin/dic2iso
@@ -1,9 +1,23 @@
 #!/bin/sed -f
 
-s/a"/ä/g
-s/A"/Ä/g
-s/o"/ö/g
-s/O"/Ö/g
-s/u"/ü/g
-s/U"/Ü/g
-s/sS/ß/g
+s/a"/\xE4/g
+s/A"/\xC4/g
+s/o"/\xF6/g
+s/O"/\xD6/g
+s/u"/\xFC/g
+s/U"/\xDC/g
+s/sS/\xDF/g
+s/a`/\xE0/g
+s/A`/\xC0/g
+s/a^/\xE2/g
+s/A^/\xC2/g
+s/c~/\xE7/g
+s/C~/\xC7/g
+s/e'/\xE9/g
+s/E'/\xC9/g
+s/e"/\xEB/g
+s/E"/\xCB/g
+s/e^/\xEA/g
+s/E^/\xCA/g
+s/n~/\xF1/g
+s/N~/\xD1/g
