From: Roland Rosenfeld <roland@debian.org>
Subject: Comment debug output of myspellfixprefix
Date: Sun, 10 Jan 2021 18:01:06 +0100
Forwarded: not needed

--- a/bin/myspellfixprefix.pl
+++ b/bin/myspellfixprefix.pl
@@ -11,7 +11,7 @@
 while (<STDIN>) {
 	chomp;
 	if (m/^[A-ZÄÖÜ].*\/.*[GUV]/) {
-		print STDERR "Capital prefixes will be expanded: $_\n";
+		#print STDERR "Capital prefixes will be expanded: $_\n";
 		
 		($start,$rest) = split("",$_,2);
 		$start =~  tr/A-ZÄÖÜ/a-zäöü/;
