From: Roland Rosenfeld <roland@debian.org>
Subject: Add (computer) "Tablet" to dictionary.
Date: Tue, 12 Jan 2021 17:58:14 +0100
Bug-Debian: https://bugs.debian.org/977369
Forwarded: Bjoern Jacke <bjoern@j3e.de> on Tue, 12 Jan 2021 18:18:45 +0100

--- a/dicts/anglizismen-de_all.txt
+++ b/dicts/anglizismen-de_all.txt
@@ -225,6 +225,7 @@ Support/S
 Sweatshirt/S
 Swimmingpool/S
 Swing
+Tablet/S
 Tattoo/S
 Techno
 Teddy/S
