Description: Use x-terminal-emulator instead of hardcoded xterm.
Origin: vendor
Bug-Debian: https://bugs.debian.org/952548
Forwarded: not-needed
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2020-05-22

--- a/lib/File/DesktopEntry.pm
+++ b/lib/File/DesktopEntry.pm
@@ -187,7 +187,7 @@
 
 If the desktop entry specifies that the program needs to be executed in a
 terminal the $TERMINAL environment variable is used. If this variable is not
-set C<xterm -e> is used as default.
+set C<x-terminal-emulator -e> is used as default on Debian systems.
 
 (On Windows this method returns a L<Win32::Process> object.)
 
@@ -229,7 +229,7 @@
 
     my $t = $self->get('Terminal');
     if ($t and $t eq 'true') {
-        my $term = $ENV{TERMINAL} || 'xterm -e';
+        my $term = $ENV{TERMINAL} || 'x-terminal-emulator -e';
         unshift @exec, _split($term);
     }
 
