Description: Disables the build-scan plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -10,7 +10,6 @@
 }
 
 plugins {
-	id 'com.gradle.build-scan' version '1.15.1'
 	id 'net.nemerosa.versioning' version '2.7.1'
 	id 'com.github.ben-manes.versions' version '0.20.0' apply false
 	id 'com.diffplug.gradle.spotless' version '3.14.0' apply false
@@ -19,10 +18,6 @@
 	id 'de.marcphilipp.nexus-publish' version '0.1.0' apply false
 }
 
-buildScan {
-	licenseAgreementUrl = 'https://gradle.com/terms-of-service'
-	licenseAgree = 'yes'
-}
 
 def buildTimeAndDate = OffsetDateTime.now()
 ext {
