Description: Disables the net.nemerosa.versioning plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -10,7 +10,6 @@
 }
 
 plugins {
-	id 'net.nemerosa.versioning' version '2.7.1'
 	id 'com.github.ben-manes.versions' version '0.20.0' apply false
 	id 'com.diffplug.gradle.spotless' version '3.14.0' apply false
 	id 'org.ajoberstar.git-publish' version '1.0.1' apply false
@@ -29,7 +28,6 @@
 	generateManifest = false
 	buildDate = DateTimeFormatter.ISO_LOCAL_DATE.format(buildTimeAndDate)
 	buildTime = DateTimeFormatter.ofPattern('HH:mm:ss.SSSZ').format(buildTimeAndDate)
-	buildRevision = versioning.info.commit
 	builtByValue = project.hasProperty('builtBy') ? project.builtBy : project.defaultBuiltBy
 
 	docsVersion = project.version.contains('SNAPSHOT') ? 'snapshot' : project.version
@@ -298,7 +296,6 @@
 				'Built-By': builtByValue,
 				'Build-Date': buildDate,
 				'Build-Time': buildTime,
-				'Build-Revision': buildRevision,
 				'Specification-Title': project.name,
 				'Specification-Version': normalizeVersion(project.version),
 				'Specification-Vendor': 'junit.org',
