Description: Disables the spotless plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -10,7 +10,6 @@
 }
 
 plugins {
-	id 'com.diffplug.gradle.spotless' version '3.14.0' apply false
 	id 'org.ajoberstar.git-publish' version '1.0.1' apply false
 	id 'com.github.johnrengelman.shadow' version '2.0.4' apply false
 	id 'de.marcphilipp.nexus-publish' version '0.1.0' apply false
@@ -92,7 +91,6 @@
 	apply plugin: 'java-library'
 	apply plugin: 'eclipse'
 	apply plugin: 'idea'
-	apply plugin: 'com.diffplug.gradle.spotless'
 	apply plugin: 'checkstyle'
 	apply plugin: 'de.obqo.gradle.degraph'
 
@@ -341,29 +339,6 @@
 
 	}
 
-	spotless {
-		def headerFile = rootProject.file('src/spotless/' + licenseOf(project)['headerFile'])
-		def importOrderConfigFile = rootProject.file('src/eclipse/junit-eclipse.importorder')
-		def javaFormatterConfigFile = rootProject.file('src/eclipse/junit-eclipse-formatter-settings.xml')
-
-		java {
-			licenseHeaderFile headerFile, '(package|import|open|module) '
-			importOrderFile importOrderConfigFile
-			eclipse().configFile javaFormatterConfigFile
-			removeUnusedImports()
-
-			trimTrailingWhitespace()
-			endWithNewline()
-		}
-
-		kotlin {
-			ktlint('0.24.0')
-			licenseHeaderFile headerFile
-			trimTrailingWhitespace()
-			endWithNewline()
-		}
-	}
-
 	afterEvaluate {
 
 		if (project.hasProperty('enableJaCoCo')) {
@@ -450,20 +425,6 @@
 			.filter { it.isDirectory() || !it.absolutePath.startsWith(projectDir.absolutePath) }
 	}
 
-	spotless {
-		format 'misc', {
-			target '**/*.gradle', '**/*.gradle.kts', '**/*.gitignore'
-			indentWithTabs()
-			trimTrailingWhitespace()
-			endWithNewline()
-		}
-		format 'documentation', {
-			target '**/*.adoc', '**/*.md'
-			trimTrailingWhitespace()
-			endWithNewline()
-		}
-	}
-
 	if (project.hasProperty('enableJaCoCo')) {
 
 		task jacocoMerge(type: JacocoMerge) {
