Description: Disables the unit tests
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/junit-jupiter-engine/junit-jupiter-engine.gradle
+++ b/junit-jupiter-engine/junit-jupiter-engine.gradle
@@ -1,5 +1,3 @@
-apply from: "$rootDir/gradle/testing.gradle"
-
 description = 'JUnit Jupiter Engine'
 
 jar {
--- a/junit-jupiter-params/junit-jupiter-params.gradle
+++ b/junit-jupiter-params/junit-jupiter-params.gradle
@@ -1,5 +1,3 @@
-apply from: "$rootDir/gradle/testing.gradle"
-
 description = 'JUnit Jupiter Params'
 
 afterEvaluate {
--- a/junit-vintage-engine/junit-vintage-engine.gradle
+++ b/junit-vintage-engine/junit-vintage-engine.gradle
@@ -1,5 +1,3 @@
-apply from: "$rootDir/gradle/testing.gradle"
-
 description = 'JUnit Vintage Engine'
 
 dependencies {
--- a/junit-jupiter-migrationsupport/junit-jupiter-migrationsupport.gradle
+++ b/junit-jupiter-migrationsupport/junit-jupiter-migrationsupport.gradle
@@ -1,5 +1,3 @@
-apply from: "$rootDir/gradle/testing.gradle"
-
 description = 'JUnit Jupiter Migration Support'
 
 dependencies {
--- a/junit-platform-commons-java-9/junit-platform-commons-java-9.gradle
+++ b/junit-platform-commons-java-9/junit-platform-commons-java-9.gradle
@@ -16,13 +16,6 @@
 // Compiles against the public, supported and documented Java 9 API.
 javacRelease = 9
 
-test {
-	useJUnitPlatform()
-	// Exclude "integration" package from default "class-path based" test run.
-	// Tests in the "integration" package assume to be run on the module-path.
-	exclude 'integration/**'
-}
-
 // Create "junit-commons-integration-tests.jar" archive which will be later
 // mounted as an automatic jar into the integration test module-path.
 task generateIntegrationTestsJar(type: Jar, dependsOn: compileTestJava) {
--- a/junit-platform-surefire-provider/junit-platform-surefire-provider.gradle
+++ b/junit-platform-surefire-provider/junit-platform-surefire-provider.gradle
@@ -1,5 +1,3 @@
-apply from: "$rootDir/gradle/testing.gradle"
-
 description = 'JUnit Platform Surefire Provider'
 
 dependencies {
--- a/settings.gradle
+++ b/settings.gradle
@@ -21,8 +21,6 @@
 include("junit-platform-suite-api")
 include("junit-platform-surefire-provider")
 include("junit-vintage-engine")
-include("platform-tests")
-include("platform-tooling-support-tests")
 include("junit-bom")
 
 // check that every subproject has a custom build file
