Description: Makes the manifests reproducible
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/build.gradle
+++ b/build.gradle
@@ -263,9 +263,6 @@
 		manifest {
 			attributes(
 				'Created-By': "${System.properties['java.version']} (${System.properties['java.vendor']} ${System.properties['java.vm.version']})".toString(),
-				'Built-By': builtByValue,
-				'Build-Date': buildDate,
-				'Build-Time': buildTime,
 				'Specification-Title': project.name,
 				'Specification-Version': normalizeVersion(project.version),
 				'Specification-Vendor': 'junit.org',
