From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Wed, 20 Jan 2021 12:06:03 +0100
Subject: Current Pacemaker does not use ha_logd

---
 logd/logd.service.in | 2 --
 1 file changed, 2 deletions(-)

diff --git a/logd/logd.service.in b/logd/logd.service.in
index c321d22..506da3d 100644
--- a/logd/logd.service.in
+++ b/logd/logd.service.in
@@ -1,8 +1,6 @@
 [Unit]
 Description=ha_logd logging daemon
 Documentation=man:ha_logd(8)
-Before=pacemaker.service
-PartOf=pacemaker.service
 
 [Service]
 ExecStart=@libdir@/@HB_PKG@/ha_logd -c @sysconfdir@/logd.cf
