From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Wed, 20 Jan 2021 12:24:02 +0100
Subject: Fix typo: bounary -> boundary

---
 lib/clplumbing/cl_msg_types.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/clplumbing/cl_msg_types.c b/lib/clplumbing/cl_msg_types.c
index f13c61d..3786f1e 100644
--- a/lib/clplumbing/cl_msg_types.c
+++ b/lib/clplumbing/cl_msg_types.c
@@ -1128,7 +1128,7 @@ binary2string(char* buf, char* maxp, void* value, size_t len, int depth)
 	baselen = B64_stringlen(len) + 1;
 	
 	if ( buf + baselen > maxp){
-		cl_log(LOG_ERR, "binary2string: out of bounary");
+		cl_log(LOG_ERR, "binary2string: out of boundary");
 		return -1;
 	}
 	
