From 6fd3a46885edd81e62ae62cfc798c1e1b903e170 Mon Sep 17 00:00:00 2001
From: Karl Heinz Marbaise <khmarbaise@apache.org>
Date: Sun, 28 Apr 2019 14:06:25 +0200
Subject: [PATCH] [MCOMPILER-379] - Fatal error compiling: basedir .../target/generated-sources/annotations does not exist
--- a/src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java
+++ b/src/main/java/org/apache/maven/plugin/compiler/AbstractCompilerMojo.java
@@ -655,6 +655,11 @@
 
         if ( generatedSourcesDirectory != null )
         {
+            if ( !generatedSourcesDirectory.exists() )
+            {
+                generatedSourcesDirectory.mkdirs();
+            }
+
             String generatedSourcesPath = generatedSourcesDirectory.getAbsolutePath();
 
             compileSourceRoots.add( generatedSourcesPath );
