Description: Disable the debugger support in the tools component
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/tools/src/main/java/org/apache/pdfbox/tools/PDFBox.java
+++ b/tools/src/main/java/org/apache/pdfbox/tools/PDFBox.java
@@ -16,8 +16,6 @@
  */
 package org.apache.pdfbox.tools;
 
-import org.apache.pdfbox.debugger.PDFDebugger;
-
 /**
  * Simple wrapper around all the command line utilities included in PDFBox.
  * Used as the main class in the runnable standalone PDFBox jar.
@@ -71,11 +69,6 @@
             {
                 PrintPDF.main(arguments);
             }
-            else if (command.equals("PDFDebugger") || command.equals("PDFReader"))
-            {
-                PDFDebugger.main(arguments);
-                exitAfterCallingMain = false;
-            }
             else if (command.equals("PDFMerger"))
             {
                 PDFMerger.main(arguments);
@@ -122,7 +115,6 @@
                 + "  ExtractImages\n"
                 + "  OverlayPDF\n"
                 + "  PrintPDF\n"
-                + "  PDFDebugger\n"
                 + "  PDFMerger\n"
                 + "  PDFReader\n"
                 + "  PDFSplit\n"
--- a/tools/pom.xml
+++ b/tools/pom.xml
@@ -39,16 +39,16 @@
 
   <dependencies>
     <dependency>
+        <groupId>${project.groupId}</groupId>
+        <artifactId>pdfbox</artifactId>
+        <version>${project.version}</version>
+    </dependency>
+    <dependency>
       <groupId>org.bouncycastle</groupId>
       <artifactId>bcmail-jdk15on</artifactId>
       <optional>true</optional>
     </dependency>
     <dependency>
-        <groupId>${project.groupId}</groupId>
-        <artifactId>pdfbox-debugger</artifactId>
-        <version>${project.version}</version>
-    </dependency>
-    <dependency>
       <groupId>junit</groupId>
       <artifactId>junit</artifactId>
       <scope>test</scope>
