Description: Change sudoers directory to be sudoers.d
Author: Chuck Short <zulcss@ubuntu.com>
Forwarded: Not needed.
--- a/heat/engine/clients/os/nova.py
+++ b/heat/engine/clients/os/nova.py
@@ -358,8 +358,8 @@ class NovaClientPlugin(microversion_mixi
             #
             # See bug https://bugs.launchpad.net/heat/+bug/1257410
             boothook_custom_user = r"""useradd -m %s
-echo -e '%s\tALL=(ALL)\tNOPASSWD: ALL' >> /etc/sudoers
-""" % (instance_user, instance_user)
+echo -e '%s\tALL=(ALL)\tNOPASSWD: ALL' >> /etc/sudoers.d/heat-instance-%s
+""" % (instance_user, instance_user, instance_user)
         else:
             config_custom_user = ''
             boothook_custom_user = ''
