Description: Fixes the compatibility with Java 11
Origin: backport, https://github.com/randomizedtesting/randomizedtesting/commit/e8839732

--
diff --git a/junit4-ant/src/main/java/com/carrotsearch/ant/tasks/junit4/JUnit4.java b/junit4-ant/src/main/java/com/carrotsearch/ant/tasks/junit4/JUnit4.java
index f4b3e83..b703270 100644
--- a/junit4-ant/src/main/java/com/carrotsearch/ant/tasks/junit4/JUnit4.java
+++ b/junit4-ant/src/main/java/com/carrotsearch/ant/tasks/junit4/JUnit4.java
@@ -1835,7 +1835,7 @@ public class JUnit4 extends Task {
             final TestClass testClass = new TestClass();
             ClassReader reader = new ClassReader(is);
             @SuppressWarnings("deprecation")
-            ClassVisitor annotationVisitor = new ClassVisitor(Opcodes.ASM8) {
+            ClassVisitor annotationVisitor = new ClassVisitor(Opcodes.ASM7) {
               @Override
               public AnnotationVisitor visitAnnotation(String desc, boolean visible) {
                 String className = Type.getType(desc).getClassName();
