Description: don't check for node-canuise-lite update
Author: Pirate Praveen <praveen@onenetbeyond.org>
Bug-Debian: https://bugs.debian.org/991651
Forwarded: not-needed
Reviewed-By: Yadd <yadd@debian.org>
Last-Update: 2021-07-30

--- a/.babelrc.js
+++ b/.babelrc.js
@@ -5,7 +5,8 @@
       {
         loose: true,
         bugfixes: true,
-        modules: false
+        modules: false,
+        ignoreBrowserslistConfig: true
       }
     ]
   ],
