From d46533b207c0869317d7672b1eaf035476b0bbf7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristoffer=20Gr=C3=B6nlund?= <krig@koru.se>
Date: Tue, 18 Oct 2016 12:34:37 +0200
Subject: [PATCH] Low: apcmastersnmp: Convert gethostbyname() usage to
 getaddrinfo()

Origin: upstream, https://github.com/ClusterLabs/cluster-glue/commit/d46533b207c0869317d7672b1eaf035476b0bbf7
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1895355
Reviewed-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>
Last-Update: 2020-09-15
---
 lib/plugins/stonith/apcmastersnmp.c | 16 ++++++++++------
 1 file changed, 10 insertions(+), 6 deletions(-)

diff --git a/lib/plugins/stonith/apcmastersnmp.c b/lib/plugins/stonith/apcmastersnmp.c
index 4411e1c5..fa87bba4 100644
--- a/lib/plugins/stonith/apcmastersnmp.c
+++ b/lib/plugins/stonith/apcmastersnmp.c
@@ -716,6 +716,7 @@ apcmastersnmp_set_config(StonithPlugin * s, StonithNVpair * list)
 	struct pluginDevice* sd = (struct pluginDevice *)s;
 	int	rc;
 	int *	i;
+	struct addrinfo *res;
 	StonithNamesToGet	namestocopy [] =
 	{	{ST_IPADDR,	NULL}
 	,	{ST_PORT,	NULL}
@@ -737,9 +738,12 @@ apcmastersnmp_set_config(StonithPlugin * s, StonithNVpair * list)
 	PluginImports->mfree(namestocopy[1].s_value);
 	sd->community = namestocopy[2].s_value;
 
-        /* try to resolve the hostname/ip-address */
-	if (gethostbyname(sd->hostname) != NULL) {
-        	/* init snmp library */
+	/* try to resolve the hostname/ip-address */
+
+	rc = getaddrinfo(sd->hostname, NULL, NULL, &res);
+	if (rc == 0) {
+		freeaddrinfo(res);
+		/* init snmp library */
 		init_snmp("apcmastersnmp");
 
 		/* now try to get a snmp session */
@@ -747,7 +751,7 @@ apcmastersnmp_set_config(StonithPlugin * s, StonithNVpair * list)
 
 			/* ok, get the number of outlets from the masterswitch */
 			if ((i = APC_read(sd->sptr, OID_NUM_OUTLETS, ASN_INTEGER))
-                    		== NULL) {
+				== NULL) {
 				LOG(PIL_CRIT
 				, "%s: cannot read number of outlets."
 				,       __FUNCTION__);
@@ -767,8 +771,8 @@ apcmastersnmp_set_config(StonithPlugin * s, StonithNVpair * list)
 			,       __FUNCTION__);
 		}
 	}else{
-		LOG(PIL_CRIT, "%s: cannot resolve hostname '%s', h_errno %d."
-		,       __FUNCTION__, sd->hostname, h_errno);
+		LOG(PIL_CRIT, "%s: cannot resolve hostname '%s', %s."
+		,       __FUNCTION__, sd->hostname, gai_strerror(rc));
 	}
 
 	/* not a valid config */
-- 
2.27.0

