From b5be33ae29ae18ec78575af207c6aeb949e3f73d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristoffer=20Gr=C3=B6nlund?= <krig@koru.se>
Date: Sat, 15 Oct 2016 09:36:37 +0200
Subject: [PATCH] low: cl_plumbing: Fix incorrect negations

Origin: upstream, https://github.com/ClusterLabs/cluster-glue/commit/b5be33ae29ae18ec78575af207c6aeb949e3f73d
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1895355
Reviewed-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>
Last-Update: 2020-09-15
---
 lib/clplumbing/cl_msg.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/clplumbing/cl_msg.c b/lib/clplumbing/cl_msg.c
index c6705d3d..2ff4882c 100644
--- a/lib/clplumbing/cl_msg.c
+++ b/lib/clplumbing/cl_msg.c
@@ -1343,7 +1343,7 @@ cl_msg_add_list_str(struct ha_msg* msg, const char* name,
 	if (n <= 0  || buf == NULL|| name ==NULL ||msg == NULL){
 		cl_log(LOG_ERR, "%s:"
 		       "invalid parameter(%s)", 
-		       !n <= 0?"n is negative or zero": 
+		       n <= 0?"n is negative or zero": 
 		       !buf?"buf is NULL":
 		       !name?"name is NULL":
 		       "msg is NULL",__FUNCTION__);
@@ -1396,7 +1396,7 @@ cl_msg_add_list_int(struct ha_msg* msg, const char* name,
 	if (n <= 0  || buf == NULL|| name ==NULL ||msg == NULL){
 		cl_log(LOG_ERR, "cl_msg_add_list_int:"
 		       "invalid parameter(%s)", 
-		       !n <= 0?"n is negative or zero": 
+		       n <= 0?"n is negative or zero": 
 		       !buf?"buf is NULL":
 		       !name?"name is NULL":
 		       "msg is NULL");
-- 
2.27.0

