From bc4aab4482a771ebcd4b16cbb5f169f431173c96 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristoffer=20Gr=C3=B6nlund?= <krig@koru.se>
Date: Wed, 19 Oct 2016 12:12:53 +0200
Subject: [PATCH] Dev: Clean up AUTHORS

Origin: upstream, https://github.com/ClusterLabs/cluster-glue/commit/bc4aab4482a771ebcd4b16cbb5f169f431173c96
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1895355
Reviewed-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>
Last-Update: 2020-09-15
---
 AUTHORS | 38 +++++++++++++++++++-------------------
 1 file changed, 19 insertions(+), 19 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index 1344abd0..545fbac3 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -1,6 +1,6 @@
-abrugger <abrugger@localhost>
+abrugger <>
 Alan Robertson <alanr@unix.sh>
-Alessandro Federico <afederic@users.noreply.github.com>
+Alessandro Federico <afederic@gmail.com>
 Alex Tsariounov <alext@novell.com>
 Andreas Mock <andreas.mock@web.de>
 Andrew Beekhof <andrew@beekhof.net>
@@ -12,14 +12,14 @@ Dave Blaschke <debltc@us.ibm.com>
 David Lee <t.d.lee@durham.ac.uk>
 Dejan Muhamedagic <dejan@hello-penguin.com>
 Denis Chapligin <chollya@satgate.net>
-Florian Haas <fghaas@users.noreply.github.com>
+Florian Haas <florian.haas@linbit.com>
 Frederik Schüler <fs@debian.org>
 Gao,Yan <ygao@suse.com>
-Goldwyn Rodrigues <goldwynr@users.noreply.github.com>
+Goldwyn Rodrigues <rgoldwyn@novell.com>
 gshi <gshi@localhost>
 Hannes Eder <heder@google.com>
 Helmut Weymann <helmut@h-weymann.de>
-Hideo Yamauchi <HideoYamauchi@users.noreply.github.com>
+Hideo Yamauchi <renayama19661014@novell.com>
 Holger Teutsch <holger.teutsch@web.de>
 Huang Zhen <zhenhltc@cn.ibm.com>
 Jean Raby <jraby@inverse.ca>
@@ -29,21 +29,21 @@ Kacper Kowalik <xarthisius.kk@gmail.com>
 Kazuhiko Higashi <higashi.kazuhiko@lab.ntt.co.jp>
 Kazuhiko Higashi <kazuh@goo.jp>
 Kazunori INOUE <kazunori.inoue3@gmail.com>
-Keisuke MORI <kskmori@users.noreply.github.com>
-kevin <kevin@localhost>
-Kristoffer Grönlund <krig@koru.se>
-Lars Ellenberg <lge@users.noreply.github.com>
+Keisuke MORI <keisuke.mori+ha@gmail.com>
+kevin <>
+Kristoffer Grönlund <kgronlund@suse.com>
+Lars Ellenberg <lars.ellenberg@linbit.com>
 Lars Marowsky-Brée <lmb@suse.com>
-lclaudio <lclaudio@localhost>
+lclaudio <>
 Martin Bene <martin.bene@icomedias.com>
 Martin Gerhard Loschwitz <martin.loschwitz@hastexo.com>
-Michael Prokop <Michael Prokop@localhost>
-msoffen <msoffen@localhost>
+Michael Prokop <>
+msoffen <>
 Nhan Ngo Dinh <nngodinh@tiscali.it>
-panjiam <panjiam@localhost>
+panjiam <>
 Pádraig Brady <P@draigBrady.com>
 Phil Carns <carns@mcs.anl.gov>
-ram <ram@localhost>
+ram <>
 Raoul Bhatia <r.bhatia@ipax.at>
 RaSca <rasca@miamammausalinux.org>
 Ruediger Oertel <ro@suse.com>
@@ -53,17 +53,17 @@ Sebastian Reitenbach <itlistuser@rapideye.de>
 Senko Rasic <senko.rasic@dobarkod.hr>
 Serge Dubrouski <sergeyfd@gmail.com>
 Simon Horman <horms@verge.net.au>
-sunjd <sunjd@localhost>
+sunjd <>
 Takatoshi MATSUO <matsuo.tak@gmail.com>
-Tim Serong <tserong@users.noreply.github.com>
+Tim Serong <tserong@suse.com>
 Vadym Chepkov <vchepkov@gmail.com>
 William Seligman <seligman@nevis.columbia.edu>
-xiliu <xiliu@localhost>
+xiliu <>
 Xinwei Hu <hxinwei@gmail.com>
 yao zhao <yao.zhao@windriver.com>
 yixiong <yixiong@localhost>
 Yoshihiko SATO <satoyoshi@intellilink.co.jp>
 Yuichi SEINO <seino.cluster2@gmail.com>
 Yusuke Iida <yusk.iida@gmail.com>
-zhaokai <zhaokai@localhost>
-zhenh <zhenh@localhost>
+zhaokai <>
+zhenh <>
-- 
2.27.0

