From 1f5a2ace89d60e1ce93f337a874b5aa1cc7579c4 Mon Sep 17 00:00:00 2001
From: Dejan Muhamedagic <dejan@hello-penguin.com>
Date: Mon, 4 Jul 2016 18:11:37 +0200
Subject: [PATCH] stonith: describe -E option

Origin: upstream, https://github.com/ClusterLabs/cluster-glue/commit/1f5a2ace89d60e1ce93f337a874b5aa1cc7579c4
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1895355
Reviewed-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>
Last-Update: 2020-09-15
---
 doc/stonith.xml.in | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/doc/stonith.xml.in b/doc/stonith.xml.in
index 575c3393..b975b90a 100644
--- a/doc/stonith.xml.in
+++ b/doc/stonith.xml.in
@@ -64,6 +64,7 @@
 	  <arg choice="plain"><replaceable>name</replaceable>=<replaceable>value</replaceable></arg>
 	</group>
 	<arg choice="plain"><option>-p</option> <replaceable>stonith-device-parameters</replaceable></arg>
+	<arg choice="plain"><option>-E</option> </arg>
 	<arg choice="plain"><option>-F</option> <replaceable>stonith-device-parameters-file</replaceable></arg>
       </group>
       <arg choice="opt"><option>-c</option> <replaceable>count</replaceable></arg>
@@ -80,6 +81,7 @@
 	  <arg choice="plain"><replaceable>name</replaceable>=<replaceable>value</replaceable></arg>
 	</group>
 	<arg choice="plain"><option>-p</option> <replaceable>stonith-device-parameters</replaceable></arg>
+	<arg choice="plain"><option>-E</option> </arg>
 	<arg choice="plain"><option>-F</option> <replaceable>stonith-device-parameters-file</replaceable></arg>
       </group>
       <arg choice="opt"><option>-c</option> <replaceable>count</replaceable></arg>
@@ -132,6 +134,15 @@
 	  whitespace.</para>
 	</listitem>
       </varlistentry>
+      <varlistentry>
+	<term>
+	  <option>-E</option>
+	</term>
+	<listitem>
+	  <para>Read the configuration from the environment. The environment variable
+	  names are the same as the parameter names.</para>
+	</listitem>
+      </varlistentry>
       <varlistentry>
 	<term>
 	  <option>-h</option>
-- 
2.27.0

