From c90168062bc3e174c3dd0f382b1e9f633acf4e0b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristoffer=20Gr=C3=B6nlund?= <krig@koru.se>
Date: Tue, 11 Oct 2016 09:53:39 +0200
Subject: [PATCH] Dev: Set foreign option to handle README.md

Origin: upstream, https://github.com/ClusterLabs/cluster-glue/commit/c90168062bc3e174c3dd0f382b1e9f633acf4e0b
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1895355
Reviewed-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>
Last-Update: 2020-09-15
---
 Makefile.am | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile.am b/Makefile.am
index 3f1146c2..0d799b78 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -15,6 +15,8 @@
 # along with this program; if not, see <http://www.gnu.org/licenses/>.
 #
 
+AUTOMAKE_OPTIONS = foreign
+
 DISTCHECK_CONFIGURE_FLAGS = \
         --with-systemdsystemunitdir=$$dc_install_base/$(systemdsystemunitdir)
 
-- 
2.27.0

