From 788a1183a80310e29415967b7d850f353ce940b2 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Kristoffer=20Gr=C3=B6nlund?= <krig@koru.se>
Date: Tue, 11 Oct 2016 09:04:42 +0200
Subject: [PATCH] Dev: I always get this backwards

Origin: upstream, https://github.com/ClusterLabs/cluster-glue/commit/788a1183a80310e29415967b7d850f353ce940b2
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1895355
Reviewed-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>
Last-Update: 2020-09-15
---
 README.md | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/README.md b/README.md
index ce0bf458..04865079 100644
--- a/README.md
+++ b/README.md
@@ -43,7 +43,7 @@ frequently requested by the developers to isolate and fix bugs, and
 are commonly found as attachments to Bugzilla entries.
 
 NOTE: hb_report has been integrated into
-[https://crmsh.github.io](crmsh), and is now maintained as part of the
+[crmsh](https://github.com/ClusterLabs/crmsh.git), and is now maintained as part of the
 `crmsh` project.
 
 ### Cluster Plumbing Library
-- 
2.27.0

