From afaeeb2f3e2a576c8a50ee3ce98e83e4da635c76 Mon Sep 17 00:00:00 2001
From: Dejan Muhamedagic <dejan@hello-penguin.com>
Date: Wed, 2 Nov 2016 13:12:59 +0100
Subject: [PATCH] update AUTHORS (with names/addresses from source files)

Origin: upstream, https://github.com/ClusterLabs/cluster-glue/commit/afaeeb2f3e2a576c8a50ee3ce98e83e4da635c76
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1895355
Reviewed-by: Rafael David Tinoco <rafaeldtinoco@ubuntu.com>
Last-Update: 2020-09-15
---
 AUTHORS | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/AUTHORS b/AUTHORS
index 545fbac3..6bd73165 100644
--- a/AUTHORS
+++ b/AUTHORS
@@ -16,10 +16,10 @@ Florian Haas <florian.haas@linbit.com>
 Frederik Schüler <fs@debian.org>
 Gao,Yan <ygao@suse.com>
 Goldwyn Rodrigues <rgoldwyn@novell.com>
-gshi <gshi@localhost>
+Guochun Shi <gshi@ncsa.uiuc.edu>
 Hannes Eder <heder@google.com>
 Helmut Weymann <helmut@h-weymann.de>
-Hideo Yamauchi <renayama19661014@novell.com>
+Hideo Yamauchi <renayama19661014@ybb.ne.jp>
 Holger Teutsch <holger.teutsch@web.de>
 Huang Zhen <zhenhltc@cn.ibm.com>
 Jean Raby <jraby@inverse.ca>
@@ -38,7 +38,7 @@ lclaudio <>
 Martin Bene <martin.bene@icomedias.com>
 Martin Gerhard Loschwitz <martin.loschwitz@hastexo.com>
 Michael Prokop <>
-msoffen <>
+Matt Soffen <matt@soffen.com>
 Nhan Ngo Dinh <nngodinh@tiscali.it>
 panjiam <>
 Pádraig Brady <P@draigBrady.com>
@@ -53,17 +53,17 @@ Sebastian Reitenbach <itlistuser@rapideye.de>
 Senko Rasic <senko.rasic@dobarkod.hr>
 Serge Dubrouski <sergeyfd@gmail.com>
 Simon Horman <horms@verge.net.au>
-sunjd <>
+Sun Jiang Dong <sunjd@cn.ibm.com>
 Takatoshi MATSUO <matsuo.tak@gmail.com>
 Tim Serong <tserong@suse.com>
 Vadym Chepkov <vchepkov@gmail.com>
 William Seligman <seligman@nevis.columbia.edu>
-xiliu <>
+Xiaoxiang Liu <xiliu@ncsa.uiuc.edu>
 Xinwei Hu <hxinwei@gmail.com>
 yao zhao <yao.zhao@windriver.com>
-yixiong <yixiong@localhost>
+Yixiong Zou <yixiong.zou@intel.com>
 Yoshihiko SATO <satoyoshi@intellilink.co.jp>
 Yuichi SEINO <seino.cluster2@gmail.com>
 Yusuke Iida <yusk.iida@gmail.com>
-zhaokai <>
-zhenh <>
+Zhaokai <zhaokai@cn.ibm.com>
+Huang Zhen <zhenh@cn.ibm.com>
-- 
2.27.0

