From: Sebastian Ramacher <sramacher@debian.org>
Date: Wed, 24 Jun 2020 23:10:27 +0200
Subject: Use packaged copy of MathJax

Forwarded: not-needed
---
 documentation/source/conf.py | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/documentation/source/conf.py b/documentation/source/conf.py
index 2983409..8f2cdfc 100644
--- a/documentation/source/conf.py
+++ b/documentation/source/conf.py
@@ -125,9 +125,7 @@ spelling_lang = "en_US"
 # Set path for mathjax js to a https URL as sometimes the Breathe docs are displayed under https
 # and we can't load an http mathjax file from an https view of the docs. So we change to a https
 # mathjax file which we can load from http or https. We break the url over two lines.
-mathjax_path = (
-    "https://c328740.ssl.cf1.rackcdn.com/" "mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"
-)
+mathjax_path = 'mathjax/MathJax.js?config=TeX-AMS-MML_HTMLorMML'
 
 
 # Add any paths that contain templates here, relative to this directory.
