From: =?utf-8?q?Timo_R=C3=B6hling?= <roehling@debian.org>
Date: Sat, 19 Feb 2022 08:51:47 +0100
Subject: Enable version injection via BREATHE_GIT_TAG env variable

Forwarded: not-needed
---
 documentation/source/conf.py | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/documentation/source/conf.py b/documentation/source/conf.py
index fa0d202..40ec9ef 100644
--- a/documentation/source/conf.py
+++ b/documentation/source/conf.py
@@ -34,11 +34,13 @@ extensions = ["breathe", "sphinx.ext.mathjax", "sphinx.ext.ifconfig", "sphinx.ex
 read_the_docs_build = os.environ.get("READTHEDOCS", None) == "True"
 travis_build = os.environ.get("TRAVIS_CI", None) == "True"
 
-# Get a description of the current position. Use Popen for 2.6 compat
-git_tag = subprocess.Popen(["git", "describe", "--tags"], stdout=subprocess.PIPE).communicate()[0]
+git_tag = os.environ.get("BREATHE_GIT_TAG", None)
+if git_tag is None:
+    # Get a description of the current position. Use Popen for 2.6 compat
+    git_tag = subprocess.Popen(["git", "describe", "--tags"], stdout=subprocess.PIPE).communicate()[0]
 
-# convert from bytes to string
-git_tag = git_tag.decode("ascii")
+    # convert from bytes to string
+    git_tag = git_tag.decode("ascii")
 
 if travis_build:
 
