Index: nunit/nuget/nunit.mocks.nuspec
===================================================================
--- nunit.orig/nuget/nunit.mocks.nuspec	2015-06-29 12:49:10.953204696 +0100
+++ nunit/nuget/nunit.mocks.nuspec	2015-06-29 12:49:10.949204696 +0100
@@ -2,7 +2,7 @@
 <package xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 	<metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
 		<id>NUnit.Mocks</id>
-		<version>${package.version}</version>
+		<version>2.6.4</version>
 		<title>NUnit.Mocks</title>
 		<authors>Charlie Poole</authors>
 		<owners>Charlie Poole</owners>
@@ -19,7 +19,7 @@
 		<tags>nunit test testing tdd mock framework</tags>
 	</metadata>
 	<files>
-		<file src="${project.base.dir}\license.txt" />
-		<file src="${current.build.dir}\framework\nunit.mocks.dll" target="lib" />
+		<file src="../license.txt" />
+		<file src="../bin/Debug/framework/nunit.mocks.dll" target="lib" />
 	</files>
-</package>
\ No newline at end of file
+</package>
Index: nunit/nuget/nunit.nuspec
===================================================================
--- nunit.orig/nuget/nunit.nuspec	2015-06-29 12:49:10.953204696 +0100
+++ nunit/nuget/nunit.nuspec	2015-06-29 12:49:10.953204696 +0100
@@ -2,7 +2,7 @@
 <package xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 	<metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
 		<id>NUnit</id>
-		<version>${package.version}</version>
+		<version>2.6.4</version>
 		<title>NUnit</title>
 		<authors>Charlie Poole</authors>
 		<owners>Charlie Poole</owners>
@@ -17,8 +17,8 @@
 		<tags>nunit test testing tdd framework fluent assert theory plugin addin</tags>
 	</metadata>
 	<files>
-		<file src="${project.base.dir}\license.txt" />
-		<file src="${current.build.dir}\framework\nunit.framework.dll" target="lib" />
-		<file src="${current.build.dir}\framework\nunit.framework.xml" target="lib" />
+		<file src="../license.txt" />
+		<file src="../bin/Debug/framework/nunit.framework.dll" target="lib" />
+		<file src="../bin/Debug/framework/nunit.framework.xml" target="lib" />
 	</files>
-</package>
\ No newline at end of file
+</package>
Index: nunit/nuget/nunit.runners.nuspec
===================================================================
--- nunit.orig/nuget/nunit.runners.nuspec	2015-06-29 12:49:10.953204696 +0100
+++ nunit/nuget/nunit.runners.nuspec	2015-06-29 12:49:20.893204933 +0100
@@ -2,7 +2,7 @@
 <package xmlns:xsd="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
 	<metadata xmlns="http://schemas.microsoft.com/packaging/2010/07/nuspec.xsd">
 		<id>NUnit.Runners</id>
-		<version>${package.version}</version>
+		<version>2.6.4</version>
 		<title>NUnit.Runners</title>
 		<authors>Charlie Poole</authors>
 		<owners>Charlie Poole</owners>
@@ -16,41 +16,18 @@
 		<tags>nunit test testing tdd runner</tags>
 	</metadata>
 	<files>
-		<file src="${project.base.dir}\license.txt" />
-		<file src="${current.build.dir}\agent.conf" target="tools" />
-		<file src="${current.build.dir}\agent.log.conf" target="tools" />
-		<file src="${current.build.dir}\launcher.log.conf" target="tools" />
-		<file src="${current.build.dir}\nunit.exe" target="tools" />
-		<file src="${current.build.dir}\nunit.exe.config" target="tools" />
-		<file src="${current.build.dir}\nunit.framework.dll" target="tools" />
-		<file src="${current.build.dir}\nunit-agent.exe" target="tools" />
-		<file src="${current.build.dir}\nunit-agent.exe.config" target="tools" />
-		<file src="${current.build.dir}\nunit-agent-x86.exe" target="tools" />
-		<file src="${current.build.dir}\nunit-agent-x86.exe.config" target="tools" />
-		<file src="${current.build.dir}\nunit-console.exe" target="tools" />
-		<file src="${current.build.dir}\nunit-console.exe.config" target="tools" />
-		<file src="${current.build.dir}\nunit-console-x86.exe" target="tools" />
-		<file src="${current.build.dir}\nunit-console-x86.exe.config" target="tools" />
-		<file src="${current.build.dir}\nunit-editor.exe" target="tools" />
-		<file src="${current.build.dir}\nunit-x86.exe" target="tools" />
-		<file src="${current.build.dir}\nunit-x86.exe.config" target="tools" />
-		<file src="${current.build.dir}\pnunit.framework.dll" target="tools" />
-		<file src="${current.build.dir}\pnunit.tests.dll" target="tools" />
-		<file src="${current.build.dir}\pnunit-agent.exe" target="tools" />
-		<file src="${current.build.dir}\pnunit-agent.exe.config" target="tools" />
-		<file src="${current.build.dir}\pnunit-launcher.exe" target="tools" />
-		<file src="${current.build.dir}\pnunit-launcher.exe.config" target="tools" />
-		<file src="${current.build.dir}\runpnunit.bat" target="tools" />
-		<file src="${current.build.dir}\test.conf" target="tools" />
-		<file src="${current.build.dir}\lib\log4net.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\nunit.core.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\nunit.core.interfaces.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\nunit.uiexception.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\nunit.uikit.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\nunit.util.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\nunit-console-runner.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\nunit-gui-runner.dll" target="tools\lib" />
-		<file src="${current.build.dir}\lib\Images\**\*.jpg" target="tools\lib\Images" />
-		<file src="${current.build.dir}\lib\Images\**\*.png" target="tools\lib\Images" />
+		<file src="../license.txt" />
+		<file src="../bin/Debug/nunit.exe" target="tools" />
+		<file src="../bin/Debug/nunit.exe.config" target="tools" />
+		<file src="../bin/Debug/framework/nunit.framework.dll" target="tools" />
+		<file src="../bin/Debug/nunit-console.exe" target="tools" />
+		<file src="../bin/Debug/nunit-console.exe.config" target="tools" />
+		<file src="../bin/Debug/lib/nunit.core.dll" target="tools/lib" />
+		<file src="../bin/Debug/lib/nunit.core.interfaces.dll" target="tools/lib" />
+		<file src="../bin/Debug/lib/nunit.uiexception.dll" target="tools/lib" />
+		<file src="../bin/Debug/lib/nunit.uikit.dll" target="tools/lib" />
+		<file src="../bin/Debug/lib/nunit.util.dll" target="tools/lib" />
+		<file src="../bin/Debug/lib/nunit-console-runner.dll" target="tools/lib" />
+		<file src="../bin/Debug/lib/nunit-gui-runner.dll" target="tools/lib" />
 	</files>
-</package>
\ No newline at end of file
+</package>
