From e07919f350a25e2f9a5b136c3cdce813e61e1d71 Mon Sep 17 00:00:00 2001
From: Mathieu Parent <math.parent@gmail.com>
Date: Fri, 13 Nov 2020 09:31:26 +0100
Subject: [PATCH] Change script shbangs to python3

---
 smb2-quota   | 2 +-
 smb2-secdesc | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/smb2-quota b/smb2-quota
index 6d0b8a3..49207c7 100755
--- a/smb2-quota
+++ b/smb2-quota
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # coding: utf-8
 #
 # smb2-quota is a cmdline tool to display quota information for the
diff --git a/smb2-secdesc b/smb2-secdesc
index 5886091..534dd92 100755
--- a/smb2-secdesc
+++ b/smb2-secdesc
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/python3
 # coding: utf-8
 
 import array
-- 
2.28.0

