From: Eric Dorland <eric@debian.org>
Date: Sat, 24 Dec 2016 11:00:25 -0500
Subject: Fix unescaped left brace

---
 automake.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/automake.in b/automake.in
index 921a6ae..003ce31 100644
--- a/automake.in
+++ b/automake.in
@@ -4156,7 +4156,7 @@ sub substitute_ac_subst_variables_worker($)
 sub substitute_ac_subst_variables ($)
 {
   my ($text) = @_;
-  $text =~ s/\${([^ \t=:+{}]+)}/&substitute_ac_subst_variables_worker ($1)/ge;
+  $text =~ s/\$\{([^ \t=:+{}]+)\}/&substitute_ac_subst_variables_worker ($1)/ge;
   return $text;
 }
 
