From: Eric Dorland <eric@debian.org>
Date: Sat, 24 Dec 2016 10:58:25 -0500
Subject: compile_f90_c_cxx-fix

---
 tests/compile_f90_c_cxx.test | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/compile_f90_c_cxx.test b/tests/compile_f90_c_cxx.test
index 21ed65d..b2ec0fd 100755
--- a/tests/compile_f90_c_cxx.test
+++ b/tests/compile_f90_c_cxx.test
@@ -41,7 +41,7 @@ END
 : > baz.cc
 
 $ACLOCAL
-$AUTOMAKE
+$AUTOMAKE -a
 
 # Look for the macros at the beginning of rules.  Be careful, as there
 # are literal tabs at the beginning of the search strings.
