--- a/src/main/java/org/apache/maven/plugin/install/AbstractInstallMojo.java
+++ b/src/main/java/org/apache/maven/plugin/install/AbstractInstallMojo.java
@@ -80,14 +80,14 @@
     /**
      * Digester for MD5.
      *
-     * @component role-hint="md5"
+     * @component role-hint="md5" role="org.codehaus.plexus.digest.Digester"
      */
     protected Digester md5Digester;
 
     /**
      * Digester for SHA-1.
      *
-     * @component role-hint="sha1"
+     * @component role-hint="sha1" role="org.codehaus.plexus.digest.Digester"
      */
     protected Digester sha1Digester;
 
