Description: Fix POD typos found by lintian spell-checker
Author: Florian Schlichting <fsfs@debian.org>
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=123586

--- a/lib/HTML/Element.pm
+++ b/lib/HTML/Element.pm
@@ -4153,7 +4153,7 @@
   $h = HTML::Element->new_from_lol($array_ref);
   @elements = HTML::Element->new_from_lol($array_ref, ...);
 
-Resursively constructs a tree of nodes, based on the (non-cyclic)
+Recursively constructs a tree of nodes, based on the (non-cyclic)
 data structure represented by each C<$array_ref>, where that is a reference
 to an array of arrays (of arrays (of arrays (etc.))).
 
--- a/lib/HTML/Parse.pm
+++ b/lib/HTML/Parse.pm
@@ -120,7 +120,7 @@
 
 =item $HTML::Parse::IGNORE_UNKNOWN
 
-This variable contols whether unknow tags should be represented as
+This variable contols whether unknown tags should be represented as
 elements in the parse tree.  Default is true.
 
 =item $HTML::Parse::IGNORE_TEXT
--- a/lib/HTML/Tree/AboutObjects.pod
+++ b/lib/HTML/Tree/AboutObjects.pod
@@ -166,7 +166,7 @@
 use the other kind of method call syntax, the syntax for B<object method>
 calls.  Calling object methods is just like class methods, except
 that instead of the ClassName as the special argument,
-you use an expression that yeilds an "object".  Usually this is
+you use an expression that yields an "object".  Usually this is
 just a scalar variable that you earlier captured the
 output of the constructor in.  Object method calls look like this:
 
@@ -565,7 +565,7 @@
 accessible as normal.  But in the unlikely case that the loan object was
 saved in mid-calculation, the calculation may not be resumable.  This is
 because of the way that that I<particular> class does its calculations,
-but similar limitations may occur with objects from other classses.
+but similar limitations may occur with objects from other classes.
 
 But often, even I<wanting> to save an object is basically wrong -- what would
 saving an ftp I<session> even mean?  Saving the hostname, username, and
--- a/lib/HTML/TreeBuilder.pm
+++ b/lib/HTML/TreeBuilder.pm
@@ -912,7 +912,7 @@
     sub end {
         return if $_[0]{'_stunted'};
 
-       # Either: Acccept an end-tag signal from HTML::Parser
+       # Either: Accept an end-tag signal from HTML::Parser
        # Or: Method for closing currently open elements in some fairly complex
        #  way, as used by other methods in this class.
         my ( $self, $tag, @stop ) = @_;
@@ -2091,7 +2091,7 @@
 
 =head2 end
 
-Either: Acccept an end-tag signal from HTML::Parser
+Either: Accept an end-tag signal from HTML::Parser
 Or: Method for closing currently open elements in some fairly complex
 way, as used by other methods in this class.
 
