From: Nick Gasson <nick@nickg.me.uk>
Date: Sat, 27 Feb 2021 14:51:42 +0800
Subject: Disable use SIGALRM for DNS lookup timeout

Forwarded: no
---
 lib/cddb_net.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/lib/cddb_net.c b/lib/cddb_net.c
index 7adf37b..aac3967 100644
--- a/lib/cddb_net.c
+++ b/lib/cddb_net.c
@@ -228,6 +228,10 @@ int sock_vfprintf(cddb_conn_t *c, const char *format, va_list ap)
 
 /* Time-out enabled work-alikes */
 
+/* Do not use SIGALRM in Debian as it is not safe if used in a program
+   with multiple threads. */
+#undef HAVE_ALARM
+
 #ifdef HAVE_ALARM
 /* time-out jump buffer */
 static jmp_buf timeout_expired;
