Subject: fix webpack
Author: Bastien Roucariès <rouca@debian.org>

Fix webpack compilation

Forwarded: not-needed

Index: node-source-map/webpack.config.js
===================================================================
--- node-source-map.orig/webpack.config.js
+++ node-source-map/webpack.config.js
@@ -77,7 +77,7 @@ testFiles.forEach(function (file) {
       library: "SOURCE_MAP_TEST_MODULE"
     },
     plugins: [
-      new BannerPlugin(run_test.toString() + "\n\n", { raw: true })
+      new BannerPlugin(run_test.toString() + "\n\n")
     ],
     devtool: "#inline-source-map"
   });
