Adapt for newer webpack version in the archive

--- a/webpack.config.js
+++ b/webpack.config.js
@@ -2,6 +2,7 @@
 var fs = require("fs");
 var path = require("path");
 var webpack = require("webpack");
+var UglifyJsPlugin = require('uglifyjs-webpack-plugin');
 
 var distDir = path.join(__dirname, "dist");
 
@@ -14,6 +15,7 @@
       filename: "source-map.js",
       library: "sourceMap",
       libraryTarget: "umd",
+      globalObject: 'this',
     },
   },
 
@@ -25,6 +27,7 @@
       filename: "source-map.debug.js",
       library: "sourceMap",
       libraryTarget: "umd",
+      globalObject: 'this',
     },
     devtool: "#inline-source-map"
   },
@@ -37,13 +40,24 @@
       filename: "source-map.min.js",
       library: "sourceMap",
       libraryTarget: "umd",
+      globalObject: 'this',
     },
-    plugins: [
-      new webpack.optimize.UglifyJsPlugin({
-        sourceMap: true
-      })
-    ],
-    devtool: "#source-map"
+    devtool: "#source-map",
+    optimization: {
+      minimize: true,
+      minimizer: [
+        new UglifyJsPlugin({
+          cache: true,
+          parallel: true,
+          uglifyOptions: {
+            compress: false,
+            ecma: 6,
+            mangle: true
+          },
+          sourceMap: true
+        })
+      ]
+    }
   }
 ];
 
