Index: libqmi-1.31~git20220823/src/libqmi-glib/qmi-device.c
===================================================================
--- libqmi-1.31~git20220823.orig/src/libqmi-glib/qmi-device.c
+++ libqmi-1.31~git20220823/src/libqmi-glib/qmi-device.c
@@ -3448,11 +3448,6 @@ qmi_device_class_init (QmiDeviceClass *k
                            G_PARAM_READABLE);
     g_object_class_install_property (object_class, PROP_CONSECUTIVE_TIMEOUTS, properties[PROP_CONSECUTIVE_TIMEOUTS]);
 
-    /**
-     * QmiDevice:device-node:
-     *
-     * Since: 1.24
-     */
 #if QMI_QRTR_SUPPORTED
     properties[PROP_NODE] =
         g_param_spec_object (QMI_DEVICE_NODE,
