Description: Disables the net.nemerosa.versioning plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -6,7 +6,6 @@
 	id 'idea'
 	id 'maven'
 	id 'signing'
-	id 'net.nemerosa.versioning' version '2.6.1'
 	id 'org.ajoberstar.github-pages' version '1.7.2'
 }
 
@@ -59,7 +58,6 @@
 			'Built-By': builtByValue,
 			'Build-Date': buildDate,
 			'Build-Time': buildTime,
-			'Build-Revision': versioning.info.commit,
 			'Specification-Title': project.name,
 			'Specification-Version': normalizeVersion(project.version),
 			'Specification-Vendor': 'apiguardian.org',
