Description: Disables the github-pages plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -6,7 +6,6 @@
 	id 'idea'
 	id 'maven'
 	id 'signing'
-	id 'org.ajoberstar.github-pages' version '1.7.2'
 }
 
 Date buildTimeAndDate = new Date()
@@ -183,22 +182,3 @@
 }
 
 createCurrentDocsFolder.onlyIf { project.hasProperty('replaceCurrentDocs') }
-
-githubPages {
-	repoUri = 'https://github.com/apiguardian-team/apiguardian.git'
-
-	credentials {
-		username = project.hasProperty('githubToken') ? project.githubToken : ''
-		password = ''
-	}
-
-	pages {
-		from docsDir
-		into "docs"
-	}
-
-	deleteExistingFiles = false
-}
-
-prepareGhPages.dependsOn(prepareDocsForUploadToGhPages)
-prepareGhPages.dependsOn(createCurrentDocsFolder)
