diff --git a/src/tests/whitespace_test b/src/tests/whitespace_test
index f055ed4c2..fa95494be 100755
--- a/src/tests/whitespace_test
+++ b/src/tests/whitespace_test
@@ -16,7 +16,7 @@ fi
 
 {
     # Look for lines with trailing whitespace in all files tracked by Git
-    git grep -n -I '\s\+$' -- "$(git rev-parse --show-toplevel)" ||
+    git grep --full-name -n -I '\s\+$' -- "$(git rev-parse --show-toplevel)" ||
         # Don't fail if no such lines were found anywhere
         [[ $? == 1 ]]
 } |
