From: =?utf-8?b?T25kxZllaiBTdXLDvQ==?= <ondrej@sury.org>
Date: Mon, 31 Oct 2016 10:48:02 +0100
Subject: Fix error: ISO C99 requires at least one argument for the "..." in a
 variadic macro

---
 tests/gdimagefile/gdnametest.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tests/gdimagefile/gdnametest.c b/tests/gdimagefile/gdnametest.c
index fcd040a..17853ef 100644
--- a/tests/gdimagefile/gdnametest.c
+++ b/tests/gdimagefile/gdnametest.c
@@ -47,8 +47,8 @@ mkcross(void)
 static void
 do_test(void)
 {
-	gdTestAssertMsg(strchr("123",'2') != 0, "strchr() is not functional.\n");
-	gdTestAssertMsg(strcasecmp("123abC","123Abc") == 0, "strcasecmp() is not functional.\n");
+	gdTestAssertMsg(strchr("123",'2') != 0, "%s", "strchr() is not functional.\n");
+	gdTestAssertMsg(strcasecmp("123abC","123Abc") == 0, "%s", "strcasecmp() is not functional.\n");
 
 	int n;
 	struct {
