From: GD Team <team+gd@tracker.debian.org>
Date: Fri, 24 Apr 2020 09:37:13 +0200
Subject: tests-make-a-little-change-for-autopkgtest

===================================================================
---
 tests/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/Makefile.am b/tests/Makefile.am
index b3c0795..0a61ec3 100644
--- a/tests/Makefile.am
+++ b/tests/Makefile.am
@@ -86,7 +86,7 @@ include webp/Makemodule.am
 include xbm/Makemodule.am
 include xpm/Makemodule.am
 
-LDADD = libgdtest.a ../src/libgd.la
+LDADD = libgdtest.a -lgd
 AM_CPPFLAGS = -I$(top_srcdir)/src -I$(top_srcdir)/tests/gdtest
 
 check_PROGRAMS = $(libgd_test_programs) $(libgd_helper_programs)
