Description: use SOURCE_DATE_EPOCH to get reproducible builds
Origin: vendor
Forwarded: not-needed
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-10-26

--- a/bitmaps
+++ b/bitmaps
@@ -116,7 +116,7 @@
 
 $VERSION = '|. $VERSION .q|';
 
-# created by Makefile.PL |. scalar localtime(). q|
+# created by Makefile.PL |. scalar gmtime($ENV{SOURCE_DATE_EPOCH}). q|
 #
 # Do not edit this package, 
 #
