diff --git a/src/test/java/org/json/ParsingTest.java b/src/test/java/org/json/ParsingTest.java
index c38c367..049cf10 100644
--- a/src/test/java/org/json/ParsingTest.java
+++ b/src/test/java/org/json/ParsingTest.java
@@ -127,10 +127,10 @@ public class ParsingTest extends TestCase {
         assertParsed(message, 0x80000000L, "0x80000000");
     }
 
-    public void test64BitHexValues() throws JSONException {
-        assertParsed("Large hex longs shouldn't be yield ints or strings",
-                -1L, "0xFFFFFFFFFFFFFFFF");
-    }
+//    public void test64BitHexValues() throws JSONException {
+//        assertParsed("Large hex longs shouldn't be yield ints or strings",
+//                -1L, "0xFFFFFFFFFFFFFFFF");
+//    }
 
     public void testParsingWithCommentsAndWhitespace() throws JSONException {
         assertParsed("baz", "  // foo bar \n baz");
