Description: Adds the OSGi metadata to the manifest
Author: Ludovic Claude <ludovic.claude@laposte.net>
Forwarded: not-needed
--- a/build.xml
+++ b/build.xml
@@ -148,6 +148,7 @@
         <filter token="title"    value="${component.title}"/>
         <filter token="package"  value="${component.package}"/>
         <filter token="version"  value="${component.version}"/>
+        <filter token="lastmodified" value="${lastmodified}"/>
       </filterset>
       <fileset dir="${source.conf}" includes="*.MF"/>
     </copy>
--- /dev/null
+++ b/src/conf/MANIFEST.MF
@@ -0,0 +1,47 @@
+Manifest-Version: 1.0
+Package: org.apache.commons.collections
+Extension-Name: commons-collections
+Specification-Version: @version@
+Specification-Vendor: Apache Software Foundation
+Specification-Title: Commons Collections
+Implementation-Version: @version@
+Implementation-Vendor: Apache Software Foundation
+Implementation-Title: Commons Collections
+Implementation-Vendor-Id: org.apache
+X-Compile-Source-JDK: 1.2
+X-Compile-Target-JDK: 1.2
+Bundle-License: http://www.apache.org/licenses/LICENSE-2.0.txt
+Import-Package: org.apache.commons.collections;version="@version@",
+ org.apache.commons.collections.bag;version="@version@",
+ org.apache.commons.collections.bidimap;version="@version@",
+ org.apache.commons.collections.buffer;version="@version@",
+ org.apache.commons.collections.collection;version="@version@",
+ org.apache.commons.collections.comparators;version="@version@",
+ org.apache.commons.collections.functors;version="@version@",
+ org.apache.commons.collections.iterators;version="@version@",
+ org.apache.commons.collections.keyvalue;version="@version@",
+ org.apache.commons.collections.list;version="@version@",
+ org.apache.commons.collections.map;version="@version@",
+ org.apache.commons.collections.set;version="@version@"
+Export-Package: org.apache.commons.collections.map;version="@version@",
+ org.apache.commons.collections.buffer;version="@version@",
+ org.apache.commons.collections.comparators;version="@version@",
+ org.apache.commons.collections.collection;version="@version@",
+ org.apache.commons.collections.bag;version="@version@",
+ org.apache.commons.collections.iterators;version="@version@",
+ org.apache.commons.collections.bidimap;version="@version@",
+ org.apache.commons.collections.set;version="@version@",
+ org.apache.commons.collections.functors;version="@version@",
+ org.apache.commons.collections.list;version="@version@",
+ org.apache.commons.collections.keyvalue;version="@version@",
+ org.apache.commons.collections;version="@version@"
+Bundle-Version: @version@
+Bundle-Description: Types that extend and augment the Java Collections
+  Framework.
+Bundle-Name: Commons Collections
+Bundle-DocURL: http://commons.apache.org/collections/
+Bundle-ManifestVersion: 2
+Bundle-Vendor: The Apache Software Foundation
+Bundle-SymbolicName: org.apache.commons.collections
+Tool: Bnd-0.0.238
+
