From: Jochen Sprickerhof <git@jochen.sprickerhof.de>
Date: Mon, 30 Apr 2018 17:04:38 +0200
Subject: Replace com.sun.org.apache.xml.internal by org.apache.xml
--- a/plugin-core/src/main/java/org/jvnet/jaxb2/maven2/RawXJC2Mojo.java
+++ b/plugin-core/src/main/java/org/jvnet/jaxb2/maven2/RawXJC2Mojo.java
@@ -77,8 +77,8 @@
 import org.xml.sax.InputSource;
 import org.xml.sax.SAXException;
 
-import com.sun.org.apache.xml.internal.resolver.CatalogManager;
-import com.sun.org.apache.xml.internal.resolver.tools.CatalogResolver;
+import org.apache.xml.resolver.CatalogManager;
+import org.apache.xml.resolver.tools.CatalogResolver;
 import com.sun.xml.txw2.annotation.XmlNamespace;
 
 /**
--- a/plugin-core/src/main/java/org/jvnet/jaxb2/maven2/resolver/tools/ClasspathCatalogResolver.java
+++ b/plugin-core/src/main/java/org/jvnet/jaxb2/maven2/resolver/tools/ClasspathCatalogResolver.java
@@ -5,7 +5,7 @@
 import java.net.URL;
 
 public class ClasspathCatalogResolver extends
-		com.sun.org.apache.xml.internal.resolver.tools.CatalogResolver {
+		org.apache.xml.resolver.tools.CatalogResolver {
 
 	public static final String URI_SCHEME_CLASSPATH = "classpath";
 
--- a/plugin-core/src/main/java/org/jvnet/jaxb2/maven2/resolver/tools/MavenCatalogResolver.java
+++ b/plugin-core/src/main/java/org/jvnet/jaxb2/maven2/resolver/tools/MavenCatalogResolver.java
@@ -10,10 +10,10 @@
 import org.jvnet.jaxb2.maven2.DependencyResourceResolver;
 import org.jvnet.jaxb2.maven2.plugin.logging.NullLog;
 
-import com.sun.org.apache.xml.internal.resolver.CatalogManager;
+import org.apache.xml.resolver.CatalogManager;
 
 public class MavenCatalogResolver extends
-		com.sun.org.apache.xml.internal.resolver.tools.CatalogResolver {
+		org.apache.xml.resolver.tools.CatalogResolver {
 
 	public static final String URI_SCHEME_MAVEN = "maven";
 	private final DependencyResourceResolver dependencyResourceResolver;
--- a/plugin-core/src/test/java/org/jvnet/jaxb2/maven2/resolver/tools/tests/MavenCatalogResolverTest.java
+++ b/plugin-core/src/test/java/org/jvnet/jaxb2/maven2/resolver/tools/tests/MavenCatalogResolverTest.java
@@ -7,8 +7,8 @@
 import org.junit.Test;
 import org.xml.sax.InputSource;
 
-import com.sun.org.apache.xml.internal.resolver.CatalogManager;
-import com.sun.org.apache.xml.internal.resolver.tools.CatalogResolver;
+import org.apache.xml.resolver.CatalogManager;
+import org.apache.xml.resolver.tools.CatalogResolver;
 
 public class MavenCatalogResolverTest {
 
