From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 22 Apr 2022 00:49:38 +0200
Subject: notification: Avoid creating a new hint variant when nothing has
 changed

Origin: https://gitlab.gnome.org/GNOME/libnotify/-/commit/b2d6712

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libnotify/+bug/1802483
---
 libnotify/notification.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libnotify/notification.c b/libnotify/notification.c
index ba3e6e8..5589ff1 100644
--- a/libnotify/notification.c
+++ b/libnotify/notification.c
@@ -1021,7 +1021,7 @@ get_parsed_variant (NotifyNotification *notification,
         const char *str = g_variant_get_string (variant, NULL);
         gchar *parsed = str_parser (notification, str);
 
-        if (parsed != NULL) {
+        if (parsed != NULL && g_strcmp0 (str, parsed) != 0) {
                 g_variant_unref (variant);
                 variant = g_variant_new_take_string (parsed);
         }
