From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Thu, 21 Apr 2022 18:33:28 +0200
Subject: notification: Do not try to append snap name to desktop when it's a
 path

Origin: https://gitlab.gnome.org/GNOME/libnotify/-/commit/d21b814

Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/libnotify/+bug/1802483
---
 libnotify/notification.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libnotify/notification.c b/libnotify/notification.c
index 8259691..ed4d9d2 100644
--- a/libnotify/notification.c
+++ b/libnotify/notification.c
@@ -485,7 +485,7 @@ try_prepend_desktop (const gchar *desktop)
          */
         ret = try_prepend_path (desktop, g_getenv ("SNAP"));
 
-        if (ret == NULL) {
+        if (ret == NULL && strchr (desktop, G_DIR_SEPARATOR) == NULL) {
                 const gchar *snap_name = g_getenv ("SNAP_NAME");
 
                 if (snap_name != NULL && snap_name[0] != '\0') {
