From: Markus Koschany <apo@debian.org>
Date: Sun, 12 Nov 2017 01:24:42 +0100
Subject: disable_buildSrc_tests

---
 buildSrc/build.gradle | 2 ++
 1 file changed, 2 insertions(+)

--- a/buildSrc/build.gradle
+++ b/buildSrc/build.gradle
@@ -31,6 +31,8 @@
     maven { url "https://plugins.gradle.org/m2/" }
 }
 
+sourceSets { test { groovy { exclude '**' } } }
+
 dependencies {
     compile 'org.ow2.asm:asm-all:5.0.3'
     compile gradleApi()
