Description: Address FTBFS with JDK 7
Author: Shuxiong Ye <yeshuxiong@gmail.com>
Forwarded: http://bitronix-transaction-manager.10986.n7.nabble.com/libbtm-java-2-1-2-Fail-to-build-from-source-with-Java7-td1499.html

--- a/src/bitronix/tm/resource/jdbc/PoolingDataSource.java
+++ b/src/bitronix/tm/resource/jdbc/PoolingDataSource.java
@@ -47,6 +47,7 @@
 import java.lang.reflect.Proxy;
 import java.sql.Connection;
 import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
 import java.util.Iterator;
 import java.util.List;
 import java.util.concurrent.CopyOnWriteArrayList;
@@ -404,4 +405,10 @@
     public void reset() throws Exception {
         pool.reset();
     }
+
+	@Override
+	public java.util.logging.Logger getParentLogger()
+			throws SQLFeatureNotSupportedException {
+		throw new SQLFeatureNotSupportedException();
+	}
 }
--- a/src/bitronix/tm/resource/jdbc/lrc/LrcXADataSource.java
+++ b/src/bitronix/tm/resource/jdbc/lrc/LrcXADataSource.java
@@ -29,6 +29,7 @@
 import java.sql.Connection;
 import java.sql.Driver;
 import java.sql.SQLException;
+import java.sql.SQLFeatureNotSupportedException;
 import java.util.Properties;
 
 /**
@@ -127,4 +128,10 @@
     public String toString() {
         return "a JDBC LrcXADataSource on " + driverClassName + " with URL " + url;
     }
+
+	@Override
+	public java.util.logging.Logger getParentLogger()
+			throws SQLFeatureNotSupportedException {
+		throw new SQLFeatureNotSupportedException();
+	}
 }
--- a/test/bitronix/tm/mock/resource/jdbc/MockDriver.java
+++ b/test/bitronix/tm/mock/resource/jdbc/MockDriver.java
@@ -24,6 +24,7 @@
 import java.sql.SQLException;
 import java.sql.Connection;
 import java.sql.DriverPropertyInfo;
+import java.sql.SQLFeatureNotSupportedException;
 import java.util.Properties;
 
 /**
@@ -55,4 +56,10 @@
     public DriverPropertyInfo[] getPropertyInfo(String url, Properties info) throws SQLException {
         return new DriverPropertyInfo[0];
     }
+
+	@Override
+	public java.util.logging.Logger getParentLogger()
+			throws SQLFeatureNotSupportedException {
+		throw new SQLFeatureNotSupportedException();
+	}
 }
--- a/test/bitronix/tm/mock/resource/jdbc/MockitoXADataSource.java
+++ b/test/bitronix/tm/mock/resource/jdbc/MockitoXADataSource.java
@@ -218,4 +218,10 @@
     public void setUselessThing(Object uselessThing) {
         this.uselessThing = uselessThing;
     }
+
+	@Override
+	public java.util.logging.Logger getParentLogger()
+			throws SQLFeatureNotSupportedException {
+		throw new SQLFeatureNotSupportedException();
+	}
 }
