Description: Fixup POD formatting (over/back)
Author: Bart Martens <bartm@debian.org>
Forwarded: no
Reviewed-by: Nick Morrott <knowledgejunkie@gmail.com>
Last-Update: 2019-02-28
---
--- a/lib/IO/Wrap.pm
+++ b/lib/IO/Wrap.pm
@@ -215,6 +215,8 @@
 
 =head1 AUTHOR
 
+=over 4
+
 =item Primary Maintainer
 
 Dianne Skoll (F<dfs@roaringpenguin.com>).
@@ -224,5 +226,7 @@
 Eryq (F<eryq@zeegee.com>).
 President, ZeeGee Software Inc (F<http://www.zeegee.com>).
 
+=back
+
 =cut
 
--- a/lib/IO/WrapTie.pm
+++ b/lib/IO/WrapTie.pm
@@ -478,6 +478,8 @@
 
 =head1 AUTHOR
 
+=over 4
+
 =item Primary Maintainer
 
 Dianne Skoll (F<dfs@roaringpenguin.com>).
@@ -487,5 +489,7 @@
 Eryq (F<eryq@zeegee.com>).
 President, ZeeGee Software Inc (F<http://www.zeegee.com>).
 
+=back
+
 =cut
 
