Description: Fixup list of upstream bugs in POD
Author: Bart Martens <bartm@debian.org>
Forwarded: no
Reviewed-by: Nick Morrott <knowledgejunkie@gmail.com>
Last-Update: 2019-02-28
---
--- a/lib/IO/Stringy.pm
+++ b/lib/IO/Stringy.pm
@@ -150,35 +150,14 @@
 Closed the following bugs at
 https://rt.cpan.org/NoAuth/Bugs.html?Dist=IO-stringy:
 
-=item
-
-2208 IO::ScalarArray->getline does not return undef for EOF if undef($/)
-
-=item
-
-7132 IO-stringy/Makefile.PL bug - name should be module name
-
-=item
-
-11249 IO::Scalar flush shouldn't return undef
-
-=item
-
-2172 $\ (output record separator) not respected
-
-=item
-
-8605 IO::InnerFile::seek() should return 1 on success
-
-=item
-
-4798 *.html in lib/
-
-=item
-
-4369 Improvement: handling of fixed-size reads in IO::Scalar
-
-(Actually, bug 4369 was closed in Version 2.109)
+    2208 IO::ScalarArray->getline does not return undef for EOF if undef($/)
+    7132 IO-stringy/Makefile.PL bug - name should be module name
+    11249 IO::Scalar flush shouldn't return undef
+    2172 $\ (output record separator) not respected
+    8605 IO::InnerFile::seek() should return 1 on success
+    4798 *.html in lib/
+    4369 Improvement: handling of fixed-size reads in IO::Scalar
+         (Actually, bug 4369 was closed in Version 2.109)
 
 =item Version 2.109   (2003/12/21)
 
