From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Tue, 2 Feb 2021 19:20:45 -0500
Subject: fix embedded version number
Forwarded: https://github.com/wolfcw/libfaketime/pull/283

---
 src/faketime.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/faketime.c b/src/faketime.c
index 452a58f..5da1e67 100644
--- a/src/faketime.c
+++ b/src/faketime.c
@@ -48,7 +48,7 @@
 
 #include "faketime_common.h"
 
-const char version[] = "0.9.7";
+const char version[] = "0.9.8";
 
 #ifdef __APPLE__
 static const char *date_cmd = "gdate";
