Description: Fix FTBFS (unit-tests) with GCC-11
Author: Lukas Märdian <slyon@ubuntu.com>
Origin: vendor, Ubuntu
Bug: https://github.com/g-truc/glm/pull/1087
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/glm/+bug/1946750
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=996241
Forwarded: https://github.com/g-truc/glm/pull/1087
Last-Update: 2021-10-12

--- glm.orig/test/CMakeLists.txt
+++ glm/test/CMakeLists.txt
@@ -206,7 +206,7 @@ elseif(CMAKE_CXX_COMPILER_ID MATCHES "GN
 		message("GLM: GCC - ${CMAKE_CXX_COMPILER_ID} compiler")
 	endif()
 
-	add_compile_options(-O2)
+	add_compile_options(-O0)
 	add_compile_options(-Wno-long-long)
 
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "Intel")
