Description: Fix POD errors
Author: Franck Joncourt <franck@debian.org>
Bug: https://rt.cpan.org/Ticket/Display.html?id=93724
Forwarded: yes
Last-Update: 2014/03/10
--- a/lib/Authen/SASL/Perl/DIGEST_MD5.pm
+++ b/lib/Authen/SASL/Perl/DIGEST_MD5.pm
@@ -802,7 +802,7 @@
 
 =head3 server
 
-=over4
+=over 4
 
 =item realm
 
--- a/lib/Authen/SASL/Perl/LOGIN.pm
+++ b/lib/Authen/SASL/Perl/LOGIN.pm
@@ -172,7 +172,7 @@
 
 =head3 Server
 
-=over4
+=over 4
 
 =item getsecret(username)
 
--- a/lib/Authen/SASL/Perl/PLAIN.pm
+++ b/lib/Authen/SASL/Perl/PLAIN.pm
@@ -143,7 +143,7 @@
 
 =head3 Server
 
-=over4
+=over 4
 
 =item checkpass(username, password, realm)
 
