--- a/pom.xml
+++ b/pom.xml
@@ -132,7 +132,7 @@
         </executions>
         <configuration>
           <version>1.0.0</version>
-          <model>src/main/mdo/antlrOptions.mdo</model>
+          <models><model>src/main/mdo/antlrOptions.mdo</model></models>
         </configuration>
       </plugin>
     </plugins>
--- a/src/main/java/org/codehaus/mojo/antlr/AntlrHtmlReport.java
+++ b/src/main/java/org/codehaus/mojo/antlr/AntlrHtmlReport.java
@@ -25,8 +25,8 @@
 import java.util.List;
 import java.util.Locale;
 
+import org.apache.maven.doxia.sink.render.RenderingContext;
 import org.apache.maven.doxia.siterenderer.Renderer;
-import org.apache.maven.doxia.siterenderer.RendererException;
 import org.apache.maven.doxia.siterenderer.sink.SiteRendererSink;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
@@ -217,20 +217,11 @@
     {
         try
         {
-            SiteRendererSink sink = siteRenderer.createSink( getReportOutputDirectory(), getOutputName() + ".html" );
+            RenderingContext context = new RenderingContext( getReportOutputDirectory(), getOutputName() + ".html" );
+            SiteRendererSink sink = new SiteRendererSink( context );
 
             generate( sink, Locale.getDefault() );
         }
-        catch ( RendererException e )
-        {
-            throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
-                + " report generation.", e );
-        }
-        catch ( IOException e )
-        {
-            throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
-                + " report generation.", e );
-        }
         catch ( MavenReportException e )
         {
             throw new MojoExecutionException( "An error has occurred in " + getName( Locale.ENGLISH )
