Description: Fixed a NPE when the grammar options are null
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: yes
Bug: http://jira.codehaus.org/browse/MANTLR-34
--- a/src/main/java/org/codehaus/mojo/antlr/metadata/MetadataExtracter.java
+++ b/src/main/java/org/codehaus/mojo/antlr/metadata/MetadataExtracter.java
@@ -239,6 +239,8 @@
             Method getRHSMethod = helper.getAntlrOptionClass().getMethod( "getRHS", Helper.NO_ARG_SIGNATURE );
             getRHSMethod.setAccessible( true );
 
+            if (options == null) return;
+
             Object importVocabOption = getElementMethod.invoke( options, new Object[] { "importVocab" } );
             if ( importVocabOption != null )
             {
