From: Markus Koschany <apo@debian.org>
Date: Sun, 1 Nov 2020 19:20:34 +0100
Subject: Version

Forwarded: no
diff --git a/src/main/java/junit/runner/Version.java b/src/main/java/junit/runner/Version.java
index 9f26384..81e07de 100644
--- a/src/main/java/junit/runner/Version.java
+++ b/src/main/java/junit/runner/Version.java
@@ -9,7 +9,7 @@ public class Version {
 	}
 
 	public static String id() {
-		return "4.13.2-SNAPSHOT";
+		return "4.13.2";
 	}
 	
 	public static void main(String[] args) {
