From: Tristan Seligmann <mithrandi@debian.org>
Date: Thu, 23 Jul 2020 09:54:38 +0200
Subject: Devendor toml.

---
 isort/settings.py | 2 +-
 setup.py          | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/isort/settings.py b/isort/settings.py
index 96fc941..507abd3 100644
--- a/isort/settings.py
+++ b/isort/settings.py
@@ -17,7 +17,7 @@ from warnings import warn
 
 from . import stdlibs
 from ._future import dataclass, field
-from ._vendored import toml
+import toml
 from .exceptions import (
     FormattingPluginDoesNotExist,
     InvalidSettingsPath,
diff --git a/setup.py b/setup.py
index ad89790..1a3fb7b 100644
--- a/setup.py
+++ b/setup.py
@@ -4,7 +4,6 @@ from setuptools import setup
 packages = \
 ['isort',
  'isort._future',
- 'isort._vendored.toml',
  'isort.deprecated',
  'isort.stdlibs']
 
