From: Till Kampetter <till.kampetter@gmail.com>
Date: Tue, 9 Aug 2016 18:11:36 +0200
Subject: Keep foomatic PPDs "(recommended)" stanzas in NickNames

Bug-Upstream: https://github.com/OpenPrinting/cups/pull/33
Forwarded: yes
---
 scheduler/cups-driverd.cxx | 10 ----------
 1 file changed, 10 deletions(-)

Index: cups/scheduler/cups-driverd.cxx
===================================================================
--- cups.orig/scheduler/cups-driverd.cxx	2022-02-20 07:05:12.833613156 +0000
+++ cups/scheduler/cups-driverd.cxx	2022-02-20 07:05:12.825612607 +0000
@@ -241,7 +241,6 @@
 	const char *scheme)		/* I - PPD scheme */
 {
   ppd_info_t	*ppd;			/* PPD */
-  char		*recommended;		/* Foomatic driver string */
 
 
  /*
@@ -280,15 +279,6 @@
   strlcpy(ppd->record.scheme, scheme, sizeof(ppd->record.scheme));
 
  /*
-  * Strip confusing (and often wrong) "recommended" suffix added by
-  * Foomatic drivers...
-  */
-
-  if ((recommended = strstr(ppd->record.make_and_model,
-                            " (recommended)")) != NULL)
-    *recommended = '\0';
-
- /*
   * Add the PPD to the PPD arrays...
   */
 
