From: Martin Pitt <mpitt@debian.org>
Date: Tue, 9 Aug 2016 18:11:41 +0200
Subject: Debian: Do not alter the permissions of cupsd.conf, ever

Origin: vendor
Bug: https://github.com/OpenPrinting/cups/pull/36
---
 scheduler/conf.c | 2 ++
 1 file changed, 2 insertions(+)

Index: cups/scheduler/conf.c
===================================================================
--- cups.orig/scheduler/conf.c	2022-02-20 07:05:27.618627438 +0000
+++ cups/scheduler/conf.c	2022-02-20 07:05:27.606626616 +0000
@@ -1132,10 +1132,12 @@
 			     Group, 1, 1) < 0 ||
        cupsdCheckPermissions(ServerRoot, "ssl", 0700, RunUser,
 			     Group, 1, 0) < 0 ||
+       /* Never alter permissions of central conffile
        cupsdCheckPermissions(ConfigurationFile, NULL, ConfigFilePerm, RunUser,
 			     Group, 0, 0) < 0 ||
        cupsdCheckPermissions(CupsFilesFile, NULL, ConfigFilePerm, RunUser,
 			     Group, 0, 0) < 0 ||
+       */
        cupsdCheckPermissions(ServerRoot, "classes.conf", 0600, RunUser,
 			     Group, 0, 0) < 0 ||
        cupsdCheckPermissions(ServerRoot, "printers.conf", 0600, RunUser,
