Description: fix POD spelling problems
Author: Jonathan Yu <jawnsy@cpan.org>
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=125715

--- a/FreezeThaw.pm
+++ b/FreezeThaw.pm
@@ -15,8 +15,8 @@
 Converts data to/from stringified form, appropriate for
 saving-to/reading-from permanent storage.
 
-Deals with objects, circular lists, repeated appearence of the same
-refence. Does not deal with overloaded I<stringify> operator yet.
+Deals with objects, circular lists, repeated appearance of the same
+reference. Does not deal with overloaded I<stringify> operator yet.
 
 =head1 EXPORT
 
@@ -134,7 +134,7 @@
   $value = $cooky->repeatedOK;
   $cooky->noRepeated;		# Now repeated are prohibited
 
-allow to find out/change the current setting for allowing repeated
+allow one to find out/change the current setting for allowing repeated
 references.
 
 If you want to flush the cache of saved objects you can use
@@ -146,7 +146,7 @@
 
 =head2 Instantiating
 
-Sometimes, when an object from a package is recreated in presense of
+Sometimes, when an object from a package is recreated in presence of
 repeated references, it is not safe to recreate the internal structure
 of an object in one step. In such a situation recreation of an object
 is carried out in two steps: in the first the object is C<allocate>d,
