--- mdds-0.6.1.orig/Makefile.in
+++ mdds-0.6.1/Makefile.in
@@ -210,8 +210,6 @@
 	$(CXX) $(LDFLAGS) $< -o $@
 
 install: $(HEADERS)
-	cp -a LICENSE COPYING
-	cp -a CHANGELOG NEWS
 	install -d $(DESTDIR)$(INSTALL_INCDIR)/mdds
 	install -d $(DESTDIR)$(INSTALL_DOCDIR)
 	install -d $(DESTDIR)@datarootdir@
@@ -219,7 +219,6 @@
 	install -m 644 @top_srcdir@/$(INCDIR)/mdds/*.hpp $(DESTDIR)$(INSTALL_INCDIR)/mdds
 	install -m 644 @top_srcdir@/$(INCDIR)/mdds/*.inl $(DESTDIR)$(INSTALL_INCDIR)/mdds
 	install -m 644 $(MISCDIR)/mdds.pc $(DESTDIR)@datarootdir@/pkgconfig/mdds-@API_VERSION@.pc
-	install -m 644 @top_srcdir@/AUTHORS @top_srcdir@/COPYING @top_srcdir@/NEWS @top_srcdir@/README.md @top_srcdir@/VERSION $(DESTDIR)$(INSTALL_DOCDIR)
 
 check: $(ALL_TESTS)
