Description: Make the build reproducible
Author: Chris Lamb <lamby@debian.org>
Last-Update: 2018-01-23

--- mdds-1.3.1.orig/doc/doxygen.conf
+++ mdds-1.3.1/doc/doxygen.conf
@@ -132,7 +132,7 @@ INLINE_INHERITED_MEMB  = NO
 # shortest path that makes the file name unique will be used
 # The default value is: YES.
 
-FULL_PATH_NAMES        = YES
+FULL_PATH_NAMES        = NO
 
 # The STRIP_FROM_PATH tag can be used to strip a user-defined part of the path.
 # Stripping is only done if one of the specified strings matches the left-hand
