--- ./caja-extension/folder-color.py	(original)
+++ ./caja-extension/folder-color.py	(refactored)
@@ -15,7 +15,7 @@
 # along with Folder Color; if not, see http://www.gnu.org/licenses
 # for more information.
 
-import os, urllib, gettext
+import os, urllib.request, urllib.parse, urllib.error, gettext
 from gi.repository import Caja, Gtk, GObject, Gio, GLib
 
 # i18n
@@ -293,7 +293,7 @@
                 continue
             
             # Get metadata file/folder
-            item_path = urllib.unquote(item.get_uri()[7:])
+            item_path = urllib.parse.unquote(item.get_uri()[7:])
             item = Gio.File.new_for_path(item_path)
             info = item.query_info('metadata', 0, None)
             # If any metadata > restore menu
@@ -312,7 +312,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.set_color(item_path, self._get_skel_folder(item_path, icon)['filename'], True)
     
     def _menu_activate_restore_all(self, menu, items):
@@ -321,7 +321,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.restore_color(item_path)
             self.foldercolor.restore_emblem(item_path)
     
@@ -331,7 +331,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.restore_emblem(item_path)
     
     def _menu_activate_emblem(self, menu, emblem, items):
@@ -340,7 +340,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             self.foldercolor.set_emblem(item_path, emblem)
     
     def _menu_activate_custom_color(self, menu, icon, items):
@@ -353,7 +353,7 @@
             if each_item.is_gone():
                 continue
             
-            item_path = urllib.unquote(each_item.get_uri()[7:])
+            item_path = urllib.parse.unquote(each_item.get_uri()[7:])
             custom_icon = self._get_skel_folder(item_path, icon)
             
             src_file = custom_icon['filename']
