Description: fix paths in cucumber/platform.rb
Author: Kevin Locke <kevin@kevinlocke.name>
Reviewed-by: Cédric Boutillier <boutil@debian.org>
Last-Update: 2015-06-19

--- a/lib/cucumber/platform.rb
+++ b/lib/cucumber/platform.rb
@@ -6,8 +6,8 @@
 module Cucumber
   unless defined?(Cucumber::VERSION)
     VERSION       = File.read(File.expand_path("../version", __FILE__))
-    BINARY        = File.expand_path(File.dirname(__FILE__) + '/../../bin/cucumber')
-    LIBDIR        = File.expand_path(File.dirname(__FILE__) + '/../../lib')
+    BINARY        = '/usr/bin/cucumber'
+    LIBDIR        = File.expand_path(File.dirname(__FILE__) + '/..')
     RAILS         = defined?(Rails)
     RUBY_BINARY   = File.join(RbConfig::CONFIG['bindir'], RbConfig::CONFIG['ruby_install_name'])
     RUBY_2_2      = RUBY_VERSION =~ /^2\.2/
