From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Wed, 18 Nov 2020 22:43:38 +0100
Subject: Always run Inkscape under the C.UTF-8 locale

---
 doc/sphinx/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doc/sphinx/Makefile.am b/doc/sphinx/Makefile.am
index 5005998..be4b012 100644
--- a/doc/sphinx/Makefile.am
+++ b/doc/sphinx/Makefile.am
@@ -95,7 +95,7 @@ DEPS_Pacemaker_Remote		= $(PNGS_Pacemaker_Remote)
 
 if BUILD_SPHINX_DOCS
 
-INKSCAPE_CMD	= $(INKSCAPE) --export-dpi=90 -C
+INKSCAPE_CMD	= LC_ALL=C.UTF-8 $(INKSCAPE) --export-dpi=90 -C
 
 # Pattern rule to generate PNGs from SVGs
 # (--export-png works with Inkscape <1.0, --export-filename with >=1.0;
