Description: Compatibility with OSGI Compendium 6
Author: Kai-Chung Yan (殷啟聰) <seamlikok@gmail.com>
Bug: https://issues.apache.org/jira/browse/FELIX-5048
Forwarded: no
--- a/src/main/java/org/apache/felix/bundlerepository/impl/OSGiRepositoryImpl.java
+++ b/src/main/java/org/apache/felix/bundlerepository/impl/OSGiRepositoryImpl.java
@@ -179,4 +179,19 @@
         }
         return sb.toString();
     }
+
+    @Override
+    public org.osgi.service.repository.RequirementBuilder newRequirementBuilder(String namespace) {
+        throw new UnsupportedOperationException("felix-bundlerepository does not fully work with OSGI Compendium 6.");
+    }
+
+    @Override
+    public org.osgi.service.repository.ExpressionCombiner getExpressionCombiner() {
+        throw new UnsupportedOperationException("felix-bundlerepository does not fully work with OSGI Compendium 6.");
+    }
+
+    @Override
+    public org.osgi.util.promise.Promise<Collection<org.osgi.resource.Resource>> findProviders(org.osgi.service.repository.RequirementExpression expression) {
+        throw new UnsupportedOperationException("felix-bundlerepository does not fully work with OSGI Compendium 6.");
+    }
 }
