Index: net-luminis-build-plugin-0.2.0/src/net/luminis/build/plugin/bnd/BuildTask.java
===================================================================
--- net-luminis-build-plugin-0.2.0.orig/src/net/luminis/build/plugin/bnd/BuildTask.java	2011-09-12 14:57:52.000000000 +0200
+++ net-luminis-build-plugin-0.2.0/src/net/luminis/build/plugin/bnd/BuildTask.java	2011-09-12 14:57:55.000000000 +0200
@@ -338,8 +338,8 @@
      * task attribute.
      */
     public void setIncludeIncludeResourceHeader(boolean include) {
-        if (! include)
-            m_instructions.put(Analyzer.REMOVE_HEADERS, Analyzer.INCLUDE_RESOURCE);
+        //if (! include)
+        //    m_instructions.put(Analyzer.REMOVE_HEADERS, Analyzer.INCLUDE_RESOURCE);
         // default is do include
     }
     
