Description: Sort the classes to make the build reproducible
Author: Chris Lamb <lamby@debian.org>
Bug-Debian: https://bugs.debian.org/862183
--- a/src/org/jgroups/util/XMLSchemaGenerator.java
+++ b/src/org/jgroups/util/XMLSchemaGenerator.java
@@ -102,7 +102,7 @@
    private static Set<Class<?>> getClasses(String packageName, Class<?> assignableFrom)
             throws IOException, ClassNotFoundException {
       ClassLoader loader = Thread.currentThread().getContextClassLoader();
-      Set<Class<?>> classes = new HashSet<Class<?>>();
+      Set<Class<?>> classes = new java.util.TreeSet<Class<?>>();
       String path = packageName.replace('.', '/');
       URL resource = loader.getResource(path);
       if (resource != null) {
