From: Gordon Ball <gordon@chronitis.net>
Date: Fri, 24 Jan 2020 09:33:15 +0000
Subject: Fix mismatched attribute wheelDeltaY -> deltaY

---
 src/Terminal.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Terminal.ts b/src/Terminal.ts
index 3927d8b..728b884 100644
--- a/src/Terminal.ts
+++ b/src/Terminal.ts
@@ -980,7 +980,7 @@ export class Terminal extends EventEmitter implements ITerminal, IDisposable, II
           : 65;
           break;
         case 'wheel':
-          button = (<WheelEvent>ev).wheelDeltaY > 0
+          button = (<WheelEvent>ev).deltaY > 0
             ? 64
           : 65;
           break;
