From: Gordon Ball <gordon@chronitis.net>
Date: Fri, 24 Jan 2020 09:54:59 +0000
Subject: Fix renamed type MediaQueryListListener -> EventListener

---
 src/ui/ScreenDprMonitor.ts | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ui/ScreenDprMonitor.ts b/src/ui/ScreenDprMonitor.ts
index 9247a03..d3efad5 100644
--- a/src/ui/ScreenDprMonitor.ts
+++ b/src/ui/ScreenDprMonitor.ts
@@ -19,7 +19,7 @@ export type ScreenDprListener = (newDevicePixelRatio?: number, oldDevicePixelRat
  */
 export class ScreenDprMonitor extends Disposable {
   private _currentDevicePixelRatio: number;
-  private _outerListener: MediaQueryListListener;
+  private _outerListener: EventListener;
   private _listener: ScreenDprListener;
   private _resolutionMediaMatchList: MediaQueryList;
 
