Description: Support externally-prodided/Debian compile flags
Author: Eugene V. Lyubimkin <jackyf@debian.org>
Last-Update: 2013-05-25
Index: html2text/Makefile.in
===================================================================
--- html2text.orig/Makefile.in
+++ html2text/Makefile.in
@@ -43,13 +43,10 @@ LIBSTDCXX_LIBS      = @LIBSTDCXX_LIBS@
 AUTO_PTR_BROKEN     = @AUTO_PTR_BROKEN@
 MAKEDEPEND_INCLUDES = @MAKEDEPEND_INCLUDES@
 
-DEBUG=-O2 -g
-
 INCLUDES  = $(LIBSTDCXX_INCLUDES)
 DEFINES   = -DVERSION=$(VERSION) $(SYS_POLL_MISSING) $(BOOL_DEFINITION) $(EXPLICIT) $(AUTO_PTR_BROKEN)
-CPPFLAGS  = $(INCLUDES) $(DEFINES)
-CXXFLAGS  = $(CPPFLAGS) $(DEBUG)
-LDFLAGS   = $(DEBUG)
+CPPFLAGS  += $(INCLUDES) $(DEFINES)
+CXXFLAGS  += $(CPPFLAGS)
 LOADLIBES = $(LIBSTDCXX_LIBS) $(SOCKET_LIBRARIES)
 
 .SUFFIXES : .C .o
