Description: Support cross-building
Bug-Debian: https://bugs.debian.org/702413
Author: Matthias Klose <doko@ubuntu.com>
Last-Update: 2013-03-06
Index: html2text/configure
===================================================================
--- html2text.orig/configure
+++ html2text/configure
@@ -39,7 +39,7 @@ int main(int, char **) {
 }
 EOF
 CXX=unknown;
-for i in "CC" "g++" "cc" "$CC"; do
+for i in "$CC" "CC" "g++" "cc"; do
   if $i -c $tmp_file.C 2>/dev/null; then
     CXX="$i";
     break;
