Description: Skip numbers in html tag attributes
Bug-Debian: https://bugs.debian.org/582199
Author: Eric Krohn <krohn-debbug@ekrohn.com>
Last-Update: 2010-05-19
Index: html2text/HTMLControl.C
===================================================================
--- html2text.orig/HTMLControl.C
+++ html2text/HTMLControl.C
@@ -372,7 +372,7 @@ HTMLControl::yylex2(yy_HTMLParser_stype
             attribute.first = c;
             for (;;) {
               c = get_char();
-              if (!isalpha(c) && c != '-' && c != '_' && c != ':') break;
+              if (!isalnum(c) && c != '-' && c != '_' && c != ':') break;
 	      // Same as in line 352 - Arno
               attribute.first += c;
             }
