Correctly specify NULLs for 64-bit architectures.
unchanged:
Index: html2text/format.C
===================================================================
--- html2text.orig/format.C
+++ html2text/format.C
@@ -562,7 +562,7 @@ Heading::format(Area::size_type w, int h
     "LEFT",   Area::LEFT,
     "CENTER", Area::CENTER,
     "RIGHT",  Area::RIGHT,
-    0
+    NULL
   );
 
   static char cell_attributes[7];
@@ -684,7 +684,7 @@ Paragraph::format(Area::size_type w, int
     "LEFT",   Area::LEFT,
     "CENTER", Area::CENTER,
     "RIGHT",  Area::RIGHT,
-    0
+    NULL
   );
 
   static BlockFormat bf("P");
@@ -754,7 +754,7 @@ Applet::format(Area::size_type w, int /*
       "LEFT",   Area::LEFT,
       "MIDDLE", Area::CENTER,
       "RIGHT",  Area::RIGHT,
-      0
+      NULL
     );
     Area *a = ::format(content.get(), w, halign);
     if (a) return a;
@@ -804,7 +804,7 @@ Division::format(Area::size_type w, int
     "LEFT",   Area::LEFT,
     "CENTER", Area::CENTER,
     "RIGHT",  Area::RIGHT,
-    0
+    NULL
   ));
 }
 
@@ -1640,7 +1640,7 @@ ListFormat::get_type(
     "A",         UPPER_ALPHA,
     "i",         LOWER_ROMAN,
     "I",         UPPER_ROMAN,
-    0
+    NULL
   );
 }
 
Index: html2text/table.C
===================================================================
--- html2text.orig/table.C
+++ html2text/table.C
@@ -122,14 +122,14 @@ create_lcs(
       "LEFT",   Area::LEFT,
       "CENTER", Area::CENTER,
       "RIGHT",  Area::RIGHT,
-      0
+      NULL
     );
     int row_valign = get_attribute(
       row.attributes.get(), "VALIGN", Area::MIDDLE,
       "TOP",    Area::LEFT,
       "MIDDLE", Area::MIDDLE,
       "BOTTOM", Area::BOTTOM,
-      0
+      NULL
     );
 
     const list<auto_ptr<TableCell> >           &cl(*row.cells);
@@ -158,14 +158,14 @@ create_lcs(
         "LEFT",   Area::LEFT,
         "CENTER", Area::CENTER,
         "RIGHT",  Area::RIGHT,
-        0
+        NULL
       );
       p->valign    = get_attribute(
         cell.attributes.get(), "VALIGN", row_valign,
         "TOP",    Area::TOP,
         "MIDDLE", Area::MIDDLE,
         "BOTTOM", Area::BOTTOM,
-        0
+        NULL
       );
       {
 	auto_ptr<Area> tmp(cell.format(
@@ -386,7 +386,7 @@ Table::format(Area::size_type w, int hal
     "LEFT",   Area::LEFT,
     "CENTER", Area::CENTER,
     "RIGHT",  Area::RIGHT,
-    0
+    NULL
   );
 
   // <TABLE>          => default => no border
