From: Maximiliano Curia <maxy@gnuservers.com.ar>
Date: Fri, 26 Jan 2018 13:54:30 -0300
Subject: Ignore the svg files in .pc

---
 validate_svg.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/validate_svg.sh
+++ b/validate_svg.sh
@@ -1,6 +1,6 @@
 #!/bin/sh
 
-find "${1:-.}" -type f -name '*.svg' -exec xmllint --noout {} + 2> xmlerrors
+find "${1:-.}" -path ./.pc -prune -or -type f -name '*.svg' -exec xmllint --noout {} + 2> xmlerrors
 if [ -s xmlerrors ]; then
     cat xmlerrors
     rm xmlerrors
