From: Brian May <bam@debian.org>
Date: Thu, 21 Oct 2021 10:01:20 +1100
Subject: python3

===================================================================
---
 lib/wind/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/wind/Makefile.am b/lib/wind/Makefile.am
index 0b49311..f524d85 100644
--- a/lib/wind/Makefile.am
+++ b/lib/wind/Makefile.am
@@ -89,7 +89,7 @@ idn_lookup_SOURCES = idn-lookup.c
 
 LDADD = libwind.la $(LIB_roken)
 
-PYTHON = python
+PYTHON = python3
 
 if !MAINTAINER_MODE
 skip_python = test -f $@ ||
