From: Ximin Luo <infinity0@debian.org>
Date: Wed, 7 Dec 2016 18:09:01 +0100
Subject: Call lessc with --source-map-basepath option for reproducibility

---
 setupbase.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setupbase.py b/setupbase.py
index 9f935e0..dd5327f 100644
--- a/setupbase.py
+++ b/setupbase.py
@@ -483,7 +483,7 @@ class CompileCSS(Command):
         for src, dst in zip(self.sources, self.targets):
             try:
                 run(['lessc',
-                    '--source-map',
+                    '--source-map', '--source-map-basepath='+os.path.dirname(__file__),
                     '--include-path=%s' % pipes.quote(static),
                     src,
                     dst,
