From: Gordon Ball <gordon@chronitis.net>
Date: Tue, 24 Sep 2019 20:37:19 +0000
Subject: use po2json from package for translations

---
 setupbase.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setupbase.py b/setupbase.py
index 8849a85..9f935e0 100644
--- a/setupbase.py
+++ b/setupbase.py
@@ -559,7 +559,7 @@ class CompileJS(Command):
     def build_jstranslation(self, trd):
         lang = trd[-5:]
         run([
-            pjoin('node_modules', '.bin', 'po2json'),
+            '/usr/share/nodejs/po2json/bin/po2json',
             '-p', '-F',
             '-f', 'jed1.x',
             '-d', 'nbjs',
