From: Gordon Ball <gordon@chronitis.net>
Date: Tue, 24 Sep 2019 20:37:20 +0000
Subject: Elide sphinxcontrib_github_alt from documentation config

---
 docs/source/conf.py | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/docs/source/conf.py b/docs/source/conf.py
index f0b8261..cb23a8a 100644
--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -70,10 +70,7 @@ extensions = [
     'sphinx.ext.intersphinx',
     'sphinx.ext.autosummary',
     'sphinx.ext.mathjax',
-    'IPython.sphinxext.ipython_console_highlighting',
     'nbsphinx',
-    'sphinxcontrib_github_alt',
-    'myst_parser'
 ]
 
 myst_enable_extensions = ["html_image"]
