From: Gordon Ball <gordon@chronitis.net>
Date: Wed, 15 Sep 2021 11:16:13 +0200
Subject: Drop deprecated babel-polyfill

---
 .babelrc          | 12 ++++++++++--
 webpack.config.js |  2 +-
 2 files changed, 11 insertions(+), 3 deletions(-)

diff --git a/.babelrc b/.babelrc
index cea726b..bca4311 100644
--- a/.babelrc
+++ b/.babelrc
@@ -1,3 +1,11 @@
-{ 
-  "presets": ["es2015"],
+{
+  "presets": [
+    [
+      "@babel/preset-env",
+      {
+        "useBuiltIns": "usage"
+      }
+    ]
+  ]
 }
+
diff --git a/webpack.config.js b/webpack.config.js
index bdb252f..646125e 100644
--- a/webpack.config.js
+++ b/webpack.config.js
@@ -8,7 +8,7 @@ crypto.createHash = (algorithm) =>
   cryptoOrigCreateHash(algorithm == 'md4' ? 'sha256' : algorithm);
 
 module.exports = {
-  entry: ['babel-polyfill', '@jupyterlab/apputils/lib/sanitizer'],
+  entry: ['regenerator-runtime/runtime', '@jupyterlab/apputils/lib/sanitizer'],
   output: {
     filename: 'index.js',
     path: path.resolve(__dirname, 'notebook/static/components/sanitizer'),
