From d4c076d4de12c4715c13dc47113d9d7afa25138b Mon Sep 17 00:00:00 2001
From: Sebastian Krzyszkowiak <dos@dosowisko.net>
Date: Mon, 30 Nov 2020 02:57:25 +0100
Subject: [PATCH] as-app: Parse custom tags from DEP-11

---
 libappstream-glib/as-app.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/libappstream-glib/as-app.c b/libappstream-glib/as-app.c
index f5c8614c..293f0a06 100644
--- a/libappstream-glib/as-app.c
+++ b/libappstream-glib/as-app.c
@@ -5793,6 +5793,14 @@ as_app_node_parse_dep11 (AsApp *app, GNode *node,
 			}
 			continue;
 		}
+		if (g_strcmp0 (tmp, "Custom") == 0) {
+			for (c = n->children; c != NULL; c = c->next) {
+				as_app_add_metadata (app,
+						     as_yaml_node_get_key (c),
+						     as_yaml_node_get_value (c));
+			}
+			continue;
+		}
 	}
 	if (nonfatal_str != NULL) {
 		g_debug ("nonfatal warning from %s: %s",
