From: Thomas Bechtold <thomasbechtold@jpberlin.de>
Date: Fri, 21 Jun 2013 11:23:31 +0200
Subject: pkgconfig

---
 check.pc.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/check.pc.in b/check.pc.in
index 9aa76f0..4f8b76c 100644
--- a/check.pc.in
+++ b/check.pc.in
@@ -7,7 +7,7 @@ Name: Check
 Description: A unit test framework for C
 URL: https://libcheck.github.io/check/
 Version: @VERSION@
-Requires.private: @LIBSUBUNIT_PC@
-Libs: -L${libdir} -lcheck
-Libs.private: @GCOV_LIBS@ @PTHREAD_LIBS@ @LIBS@
-Cflags: -I${includedir} @PTHREAD_CFLAGS@
+Requires: @LIBSUBUNIT_PC@
+Libs: -L${libdir} -lcheck_pic @GCOV_LIBS@ @PTHREAD_CFLAGS@ @PTHREAD_LIBS@ @LIBS@
+Libs.private:
+Cflags: -I${includedir} -pthread
