From: Thomas Bechtold <thomasbechtold@jpberlin.de>
Date: Fri, 21 Jun 2013 11:23:31 +0200
Subject: awk

---
 checkmk/checkmk.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/checkmk/checkmk.in b/checkmk/checkmk.in
index 2100274..60aee5a 100644
--- a/checkmk/checkmk.in
+++ b/checkmk/checkmk.in
@@ -1,4 +1,4 @@
-#! @AWK_PATH@ -f
+#! /usr/bin/mawk -f
 # @configure_input@
 
 # checkmk - translate more concise versions of test suite specifications
