Description: Disables SSPI support for Windows
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/pom.xml
+++ b/pom.xml
@@ -146,6 +146,9 @@
           <compilerArgs>
             <arg>-Xlint:all,-options,-path</arg>
           </compilerArgs>
+          <excludes>
+            <exclude>**/Windows*</exclude>
+          </excludes>
         </configuration>
       </plugin>
 
--- a/src/main/java/org/mariadb/jdbc/internal/com/send/authentication/gssapi/GssUtility.java
+++ b/src/main/java/org/mariadb/jdbc/internal/com/send/authentication/gssapi/GssUtility.java
@@ -63,20 +63,6 @@
    * @return authentication method
    */
   public static GssapiAuth getAuthenticationMethod() {
-    try {
-      // Waffle-jna has jna as dependency, so if not available on classpath, just use standard
-      // authentication
-      if (Platform.isWindows()) {
-        try {
-          Class.forName("waffle.windows.auth.impl.WindowsAuthProviderImpl");
-          return new WindowsNativeSspiAuthentication();
-        } catch (ClassNotFoundException cle) {
-          // waffle not in the classpath
-        }
-      }
-    } catch (Throwable cle) {
-      // jna jar's are not in classpath
-    }
     return new StandardGssapiAuthentication();
   }
 }
