Index: libdevel-callchecker-perl-0.008/t/threads.t
===================================================================
--- libdevel-callchecker-perl-0.008.orig/t/threads.t
+++ libdevel-callchecker-perl-0.008/t/threads.t
@@ -30,7 +30,7 @@ use Test::More tests => 3;
 use Thread::Semaphore ();
 use threads::shared;
 
-alarm 10;   # failure mode may involve an infinite loop
+alarm 30;   # failure mode may involve an infinite loop
 
 sub tsub1 (@) { $_[0] }
 sub tsub2 (@) { $_[0] }
