From 87b2176f711e47f09de1664b4ece8eb38537278b Mon Sep 17 00:00:00 2001
From: Gordon Ball <gordon@chronitis.net>
Date: Sun, 6 Nov 2016 12:53:39 +0100
Subject: Unconditionally import setuptools to pick up dependencies

---
 setup.py                    | 3 +--
 widgetsnbextension/setup.py | 3 +--
 2 files changed, 2 insertions(+), 4 deletions(-)

diff --git a/setup.py b/setup.py
index 2294258..ac1a5a3 100644
--- a/setup.py
+++ b/setup.py
@@ -102,8 +102,7 @@ setup_args = dict(
     },
 )
 
-if 'develop' in sys.argv or any(a.startswith('bdist') for a in sys.argv):
-    import setuptools
+import setuptools
 
 setuptools_args = {}
 install_requires = setuptools_args['install_requires'] = [
diff --git a/widgetsnbextension/setup.py b/widgetsnbextension/setup.py
index dfac07a..7b88c33 100644
--- a/widgetsnbextension/setup.py
+++ b/widgetsnbextension/setup.py
@@ -210,8 +210,7 @@ setup_args = dict(
     include_package_data = True,
 )
 
-if 'develop' in sys.argv or any(a.startswith('bdist') for a in sys.argv):
-    import setuptools
+import setuptools
 
 if 'setuptools' in sys.modules:
     # setup.py develop should check for submodules
