From: Gordon Ball <gordon@chronitis.net>
Date: Sun, 7 Feb 2021 19:49:40 +0000
Subject: Simplify sphinx build

---
 docs/source/conf.py | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/docs/source/conf.py b/docs/source/conf.py
index 8a5bb7a..1500025 100644
--- a/docs/source/conf.py
+++ b/docs/source/conf.py
@@ -26,7 +26,7 @@ def bash(filename):
 
 # -- source files and parsers -----------------------------------
 
-source_suffix = ['.rst', '.md', '.ipynb']
+source_suffix = ['.rst', '.md']
 source_parsers = {
     '.md': recommonmark.parser.CommonMarkParser,
 }
@@ -39,9 +39,6 @@ extensions = [
     'sphinx.ext.autosummary',
     'sphinx.ext.intersphinx',
     'sphinx.ext.mathjax',
-    'nbsphinx',
-    'jupyter_sphinx.embed_widgets',
-    'IPython.sphinxext.ipython_console_highlighting',
 ]
 
 intersphinx_mapping = {
