Description: compatibility with python 3.10
Author: Steve Langasek <steve.langasek@ubuntu.com>
Author: Graham Inggs <ginggs@debian.org>
Bug-Debian: https://bugs.debian.org/1001343
Last-Update: 2022-02-07
Forwarded: no

--- a/ipywidgets/widgets/widget_selection.py
+++ b/ipywidgets/widgets/widget_selection.py
@@ -6,7 +6,7 @@
 Represents an enumeration using a widget.
 """
 
-from collections import Mapping
+from collections.abc import Mapping
 try:
     from itertools import izip
 except ImportError:  #python3.x
--- a/ipywidgets/widgets/interaction.py
+++ b/ipywidgets/widgets/interaction.py
@@ -26,7 +26,7 @@
 from traitlets import HasTraits, Any, Unicode, observe
 from numbers import Real, Integral
 from warnings import warn
-from collections import Iterable, Mapping
+from collections.abc import Iterable, Mapping
 
 empty = Parameter.empty
 
--- a/ipywidgets/widgets/tests/test_interaction.py
+++ b/ipywidgets/widgets/tests/test_interaction.py
@@ -259,7 +259,8 @@
     check_widgets(c, lis=d)
 
 def test_mapping():
-    from collections import Mapping, OrderedDict
+    from collections import OrderedDict
+    from collections.abc import Mapping
     class TestMapping(Mapping):
         def __init__(self, values):
             self.values = values
