--- a/src/guid.h
+++ b/src/guid.h
@@ -79,6 +79,8 @@ check_segment_sanity(const char *text, s
 static inline int UNUSED
 text_to_guid(const char *text, efi_guid_t *guid)
 {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wstringop-truncation"
 	/* these variables represent the length of the /string/ they hold,
 	 * not the interpreted length of the value from them.  Mostly the
 	 * names make it more obvious to verify that my bounds checking is
@@ -176,6 +178,7 @@ text_to_guid(const char *text, efi_guid_
 	guid->e[5] = (uint8_t)strtoul(twobytes, NULL, 16);
 
 	return 0;
+#pragma GCC diagnostic pop
 }
 
 struct guidname {
