From: Andrea Mazzoleni <amadvance@gmail.com>
Date: Fri, 1 Mar 2019 20:40:25 +0100
Subject: Fix a buffer overflow with image of invalid size
Origin: https://github.com/amadvance/advancecomp/commit/fcf71a89265c78fc26243574dda3a872574a5c02
Bug-Debian-Security: https://security-tracker.debian.org/tracker/CVE-2019-9210
Bug-Debian: https://bugs.debian.org/923416
Bug: https://sourceforge.net/p/advancemame/bugs/277/

---

diff --git a/lib/png.c b/lib/png.c
index cbf140b2ca6d..f888a28519d5 100644
--- a/lib/png.c
+++ b/lib/png.c
@@ -656,6 +656,11 @@ adv_error adv_png_read_ihdr(
 	}
 	*pix_pixel = pixel;
 
+	if (width_align < width) {
+		error_unsupported_set("Invalid image size");
+		goto err;
+	}
+
 	if (data[10] != 0) { /* compression */
 		error_unsupported_set("Unsupported compression, %d instead of 0", (unsigned)data[10]);
 		goto err;
-- 
2.11.0

