Description: suppress warning if connection is still alive
Origin: vendor
Bug-Debian: http://bugs.debian.org/399309
Forwarded: no
Author: Bart Martens <bartm@knars.be>
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2014-08-11

--- a/lib/Ima/DBI.pm
+++ b/lib/Ima/DBI.pm
@@ -396,7 +396,7 @@
 		# This is to do proper '%%' translation.
 		my $sql = $class->transform_sql($statement => @_);
 		return $cache
-			? $dbh->prepare_cached($sql)
+			? $dbh->prepare_cached($sql,undef,1)
 			: $dbh->prepare($sql);
 	};
 }
