Description: fix some spelling mistakes
Origin: vendor
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2014-08-11
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=97938
Bug: https://rt.cpan.org/Ticket/Display.html?id=97938

--- a/lib/Ima/DBI.pm
+++ b/lib/Ima/DBI.pm
@@ -49,7 +49,7 @@
 
 Ima::DBI attempts to organize and facilitate caching and more efficient
 use of database connections and statement handles by storing DBI and
-SQL information with your class (instead of as seperate objects).
+SQL information with your class (instead of as separate objects).
 This allows you to pass around just one object without worrying about
 a trail of DBI handles behind it.
 
@@ -155,7 +155,7 @@
 
 One program, one database connection (per database user).  One program,
 one prepared statement handle (per statement, per database user).
-That's what Ima::DBI enforces.  Extremely handy in persistant environments
+That's what Ima::DBI enforces.  Extremely handy in persistent environments
 (servers, daemons, mod_perl, FastCGI, etc...)
 
 =item * Encourages use of bind parameters and columns
@@ -189,7 +189,7 @@
 
 =item * Overkill for programs that only use their SQL statements once
 
-Ima::DBI's caching might prove to be an unecessary performance hog if
+Ima::DBI's caching might prove to be an unnecessary performance hog if
 you never use the same SQL statement twice.  Not sure, I haven't
 looked into it.
 
@@ -217,7 +217,7 @@
 =item 4
 
 Use sql_* to retrieve your statement handles ($sth) as needed and db_*
-to retreive database handles ($dbh).
+to retrieve database handles ($dbh).
 
 
 =back
