From: Maximiliano Curia <maxy@gnuservers.com.ar>
Date: Mon, 4 Sep 2017 22:12:24 +0200
Subject: Fix icons path

---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index f7ab620..fbcbb21 100755
--- a/setup.py
+++ b/setup.py
@@ -85,8 +85,8 @@ class KmjBuild(build):
         for binary in ['kajongg', 'kajonggserver']:
             open(binary, 'w').write('#!/bin/sh\nexec %skajongg/%s.py $*\n' % (kdeDirs['data'], binary))
             os.chmod(binary, 0o0755 )
-        call(['cp sc-apps-kajongg.svgz kajongg.svgz'], shell=True)
-        call(['cp sc-action-games-kajongg-law.svgz games-kajongg-law.svgz'], shell=True)
+        call(['cp icons/sc-apps-kajongg.svgz kajongg.svgz'], shell=True)
+        call(['cp icons/sc-actions-games-kajongg-law.svgz games-kajongg-law.svgz'], shell=True)
         build.run(self)
 
 setup(name='kajongg',
