Description: Compatibility with Junit3
Author: Miguel Landaeta <miguel@miguel.cc>
Forwarded: no
Last-Update: 2011-09-12

--- jmock-1.2.0.orig/jmock-core/org/jmock/expectation/AssertMo.java
+++ jmock-1.2.0/jmock-core/org/jmock/expectation/AssertMo.java
@@ -73,7 +73,7 @@ public class AssertMo extends Assert
         assertTrue("Should not have verified", threwException);
     }
 
-    static protected void failNotEquals( String message,
+    static public void failNotEquals( String message,
                                          Object expected,
                                          Object actual ) {
         String formatted = "";
