From: Dmitry Shachnev <mitya57@debian.org>
Date: Wed, 11 Nov 2020 14:08:44 +0300
Subject: Use local intersphinx inventory

To prevent Sphinx from downloading https://docs.python.org/3/objects.inv
during build.
---
 docs/conf.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docs/conf.py b/docs/conf.py
index aad6069..8b7f396 100644
--- a/docs/conf.py
+++ b/docs/conf.py
@@ -120,7 +120,7 @@ pygments_style = 'sphinx'
 todo_include_todos = False
 
 intersphinx_mapping = {
-    'python': ('https://docs.python.org/3', None),
+    'python': ('/usr/share/doc/python3-doc/html', None),
 }
 
 
