From: Armin Novak <armin.novak@thincast.com>
Date: Wed, 27 Apr 2022 17:06:48 +0200
Subject: Do not mark WinPR implementation of ExitThread as noreturn

---
 winpr/include/winpr/thread.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/winpr/include/winpr/thread.h b/winpr/include/winpr/thread.h
index 4e603b3..c7961e9 100644
--- a/winpr/include/winpr/thread.h
+++ b/winpr/include/winpr/thread.h
@@ -207,7 +207,7 @@ extern "C"
 	                                    LPVOID lpParameter, DWORD dwCreationFlags,
 	                                    LPDWORD lpThreadId);
 
-	WINPR_API DECLSPEC_NORETURN VOID ExitThread(DWORD dwExitCode);
+	WINPR_API VOID ExitThread(DWORD dwExitCode);
 	WINPR_API BOOL GetExitCodeThread(HANDLE hThread, LPDWORD lpExitCode);
 
 	WINPR_API HANDLE _GetCurrentThread(void);
