From: Armin Novak <armin.novak@thincast.com>
Date: Mon, 28 Mar 2022 11:34:24 +0200
Subject: Updated changelog

(cherry picked from commit 732e94628945c2c218968c80b3a7a041b0b8a6e9)
---
 ChangeLog | 1 +
 1 file changed, 1 insertion(+)

diff --git a/ChangeLog b/ChangeLog
index 608c697..3e41b9f 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,6 +1,7 @@
 # 2022-XX-YY Version 2.X.Y
 
 Noteworthy changes:
+* Backported WINPR_ASSERT to ease future backports
 
 Fixed issues:
 * Backported #6786: Use /network:auto by default
