From: Hideki Yamane <henrich@debian.org>
Date: Mon, 28 Aug 2017 20:10:24 +0900
Subject: add module path for automodule directive

---
 Doc/source/conf.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Doc/source/conf.py b/Doc/source/conf.py
index 82a5d57..8dcb774 100644
--- a/Doc/source/conf.py
+++ b/Doc/source/conf.py
@@ -20,6 +20,8 @@
 # import sys
 # sys.path.insert(0, os.path.abspath('.'))
 
+import sys,os
+sys.path.append(os.path.dirname(os.path.abspath(__file__)) + '/../../Lib')
 
 # -- General configuration ------------------------------------------------
 
