diff --git a/meson.build b/meson.build
index 65be8ec..dd6bb77 100644
--- a/meson.build
+++ b/meson.build
@@ -60,7 +60,8 @@ install_data('at-spi2-atk.desktop',
              install_dir: join_paths(get_option('libdir'), 'gnome-settings-daemon-3.0', 'gtk-modules'))
 
 subdir('droute')
-subdir('atk-adaptor')
+#subdir('atk-adaptor')
+libatk_bridge_dep = dependency('atk-bridge-2.0')
 if get_option('tests')
   subdir('tests')
 endif
