Description: load global tasks and disable unneeded tasks
Author: Pirate Praveen <praveen@debian.org>
Forwarded: <no|not-needed|url proving that it has been forwarded>
Reviewed-By: Xavier Guimard <yadd@debian.org>
Last-Update: 2020-02-21

--- a/Gruntfile.js
+++ b/Gruntfile.js
@@ -39,7 +39,7 @@
   });
 
   // These plugins provide necessary tasks.
-  grunt.loadNpmTasks('grunt-eslint');
+  //grunt.loadNpmTasks('grunt-eslint');
   grunt.loadNpmTasks('grunt-contrib-nodeunit');
   grunt.loadNpmTasks('grunt-contrib-watch');
 
@@ -48,7 +48,7 @@
 
   // "npm test" runs these tasks
   grunt.registerTask('test', '', function(reporter) {
-    grunt.task.run(['eslint', 'nodeunit:' + (reporter || 'all'), 'subgrunt']);
+    grunt.task.run(['nodeunit:' + (reporter || 'all'), 'subgrunt']);
   });
 
   // Default task.
