Description: fix coffeescript path
Author: Xavier Guimard <yadd@debian.org>
Forwarded: no
Last-Update: 2020-09-06

--- a/lib/grunt.js
+++ b/lib/grunt.js
@@ -7,7 +7,7 @@
 try {
   // Note: grunt no longer depends on CoffeeScript, it will only use it if it is intentionally
   // installed in the project.
-  require('coffeescript/register');
+  require('coffee-script/register');
 } catch (e) {
   // This is fine, and will cause no problems so long as the user doesn't load .coffee files.
   // Print a useful error if we attempt to load a .coffee file.
