Description: Fix a spelling error in drvcfg.c.
Author: Hugh McMaster <hugh.mcmaster@outlook.com>
Last-Update: 2020-12-30
Forwarded: yes, via email

--- a/DRVConfig/txt/drvcfg.c
+++ b/DRVConfig/txt/drvcfg.c
@@ -75,7 +75,7 @@
 	hLastProperty->nPromptType			= ODBCINST_PROMPTTYPE_COMBOBOX;
 	hLastProperty->aPromptData          = malloc( sizeof( aColumnSeparators ) );
 	memcpy( hLastProperty->aPromptData, aColumnSeparators, sizeof( aColumnSeparators ) ); 
-	strncpy( hLastProperty->szName, "ColumnSeperator", INI_MAX_PROPERTY_NAME );
+	strncpy( hLastProperty->szName, "ColumnSeparator", INI_MAX_PROPERTY_NAME );
 	strncpy( hLastProperty->szValue, "|", INI_MAX_PROPERTY_VALUE );
 	hLastProperty->pszHelp				= strdup( "Column separator character used in table files.\nCANNOT EXIST IN COLUMN VALUES." );
 
