Description: Export additional odbcinst functions.
Author: Steve Langasek <vorlon@debian.org>
Last-Update: 2020-12-31
Forwarded: https://github.com/lurcher/unixODBC/pull/52

--- a/odbcinst/odbcinst.exp
+++ b/odbcinst/odbcinst.exp
@@ -65,3 +65,21 @@
 _SQLDriverConnectPrompt
 _SQLDriverConnectPromptW
 inst_logPushMsg
+iniClose
+iniCommit
+iniElement
+iniObject
+iniObjectDelete
+iniObjectEOL
+iniObjectFirst
+iniObjectNext
+iniObjectSeek
+iniObjectSeekSure
+iniOpen
+iniProperty
+iniPropertyEOL
+iniPropertyFirst
+iniPropertyInsert
+iniPropertyNext
+iniToUpper
+iniValue
