Description: Don't fail build intentionally on unknown systems
Author: Balint Reczey <balint@balintreczey.hu>

--- a/include/gpac/configuration.h
+++ b/include/gpac/configuration.h
@@ -188,8 +188,8 @@
 #define GPAC_HAS_JPEG
 #define GPAC_HAS_PNG
 
-#else
-#error "Unknown target platform used with static configuration file"
+//#else
+//#error "Unknown target platform used with static configuration file"
 #endif
 
 
