Subject: Add -O2 to CFLAGS by default, -O0 if noopt is set.
Author: Alessio Treglia <alessio@debian.org>
Forwarded: not-needed
---
 configure |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/configure
+++ b/configure
@@ -365,7 +365,7 @@
             ;;
         --cpu=*) cpu=`echo $opt | cut -d '=' -f 2`
             ;;
-        --enable-debug) debuginfo="yes"; no_gcc_opt="yes"
+        --enable-debug) debuginfo="yes";
             ;;
         --disable-opt) no_gcc_opt="yes"
             ;;
@@ -708,7 +708,7 @@
 
 #GCC opt
 if test "$no_gcc_opt" = "no"; then
-    CFLAGS="-O3 $CFLAGS"
+    CFLAGS="-O2 $CFLAGS"
 else
     CFLAGS="-O0 $CFLAGS"
 fi
