From: Logan Rosen <logan@ubuntu.com>
Date: Tue, 31 Dec 2013 23:35:00 -0500
Subject: Disable -Werror for automake to fix FTBFS while autoreconfing

Bug-Debian: https://bugs.debian.org/733834
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 58ff478..2bbf6dd 100644
--- a/configure.ac
+++ b/configure.ac
@@ -7,7 +7,7 @@ AC_PREREQ(2.53)
 
 AC_CANONICAL_TARGET
 
-AM_INIT_AUTOMAKE([-Wall -Werror foreign subdir-objects])
+AM_INIT_AUTOMAKE([-Wall foreign subdir-objects])
 AC_CONFIG_HEADERS([include/config.h])
 
 ifdef([AM_SILENT_RULES],[AM_SILENT_RULES])
