From 9c58b10cb53eede2b0a9af0e5322fdce7d779dc4 Mon Sep 17 00:00:00 2001
From: Balint Reczey <balint.reczey@canonical.com>
Date: Mon, 20 Nov 2017 12:44:23 +0100
Subject: [PATCH] Fix version to 1.0.1 which was tagged already

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 520be4c..444709a 100644
--- a/setup.py
+++ b/setup.py
@@ -14,7 +14,7 @@ with open("README.md", "r") as fp:
     hib_long_description = fp.read()
 
 setup(name="hibagent",
-      version='1.0.0',
+      version='1.0.1',
       author="Aleksei Besogonov",
       author_email="cyberax@amazon.com",
       url="https://github.com/awslabs/hibagent",
-- 
2.11.0

