Description: Add LSB init info to init script
Author: Balint Reczey <rbalint@ubuntu.com>
Forwarded: no
Last-Update: 2017-10-27

--- a/etc/init.d/hibagent
+++ b/etc/init.d/hibagent
@@ -4,6 +4,15 @@
 # description: EC2 instance hibernation agent
 # processname: hibagent
 
+### BEGIN INIT INFO
+# Provides:        hibagent
+# Required-Start:  $network $remote_fs $syslog
+# Required-Stop:   $network $remote_fs $syslog
+# Default-Start:   2 3 4 5
+# Default-Stop:    0 1 6
+# Short-Description: Start EC2 Spot hibernation agent
+### END INIT INFO
+
 . /lib/lsb/init-functions
 
 DAEMON_PATH="/usr/bin"
