From: Ansgar Burchardt <ansgar@43-1.org>
Date: Thu, 26 Aug 2010 17:15:34 +0900
Origin: vendor
Subject: Fix spelling errors in documentation

--- a/lib/Date/Calc/Object.pod
+++ b/lib/Date/Calc/Object.pod
@@ -483,7 +483,7 @@
 
 Invalid Dates
 
-Only "new()" allows to create objects containing possibly invalid
+Only "new()" allows one to create objects containing possibly invalid
 dates (needed for reading in and evaluating user input, for example).
 
 =item *
@@ -1074,7 +1074,7 @@
 dates subsequently printed by this module will automatically be
 in that language - provided that you use the built-in formats of
 this module, or that you pass the third parameter of the callback
-funtion to the funtions of the "Date::Calc" module which accept it.
+function to the functions of the "Date::Calc" module which accept it.
 
 However, this global language setting can be overridden for
 individual date (or delta) objects by using the B<OBJECT> method
--- a/lib/Date/Calendar/Profiles.pod
+++ b/lib/Date/Calendar/Profiles.pod
@@ -65,7 +65,7 @@
 responsible for parsing the calendar schemes contained
 here in the Date::Calendar::Profiles module.
 
-This method offers a "mini-language" which allows to
+This method offers a "mini-language" which allows one to
 specify common date formulas, like for instance a simple
 fixed date (in various different formats, e.g. american
 or european), or things like "the second Sunday of May"
--- a/lib/Date/Calc.pod
+++ b/lib/Date/Calc.pod
@@ -1133,7 +1133,7 @@
 one month is [1999,4,1].
 
 (Which is also the reason why the "C<Add_Delta_YM()>" function
-does not allow to add a days offset, because this would actually
+does not allow one to add a days offset, because this would actually
 require TWO functions: One for adding the days offset BEFORE and
 one for adding it AFTER applying the year/month offsets.)
 
