Author:		Edik Ponomarenko <edik@lantaris.ru>
Date:		Mon Jan 28 21:22:30 2019 +0300
Description:	Address potential 1-byte stack buffer overflow
Origin:		https://github.com/ndevilla/iniparser/commit/2412f165bcfde4ad8e3426fd59f2a920492b8c19
Bug-Ubuntu:	https://bugs.launchpad.net/bugs/1913321
Forwarded:	not-needed, obtained from upstream
Last-Update:	2019-01-28

    Fix size

diff --git a/src/iniparser.c b/src/iniparser.c
index fffdf9f..f1d1658 100644
--- a/src/iniparser.c
+++ b/src/iniparser.c
@@ -718,7 +718,7 @@ dictionary * iniparser_load(const char * ininame)
     char line    [ASCIILINESZ+1] ;
     char section [ASCIILINESZ+1] ;
     char key     [ASCIILINESZ+1] ;
-    char tmp     [(ASCIILINESZ * 2) + 1] ;
+    char tmp     [(ASCIILINESZ * 2) + 2] ;
     char val     [ASCIILINESZ+1] ;
 
     int  last=0 ;
