Index: iniparser-4.1/CMakeLists.txt
===================================================================
--- iniparser-4.1.orig/CMakeLists.txt
+++ iniparser-4.1/CMakeLists.txt
@@ -44,6 +44,10 @@ enable_testing()
 add_test(NAME testsuite
   COMMAND make
   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test)
+add_test(NAME testclean
+  COMMAND make clean
+  WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/test)
+set_tests_properties(testclean PROPERTIES DEPENDS testsuite)
 
 install (FILES ${INIPARSER_HDRS} DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/iniparser)
 
