Author: Hugo Osvaldo Barrera <hugo@barrera.io>
Origin: upstream, https://github.com/bus1/dbus-broker/commit/833a92c0aa262d467faed35d8cac3896f5bc352d
Description: run user service in session.slice
--- a/src/units/user/dbus-broker.service.in
+++ b/src/units/user/dbus-broker.service.in
@@ -11,6 +11,7 @@
 Sockets=dbus.socket
 ExecStart=@bindir@/dbus-broker-launch --scope user
 ExecReload=@bindir@/busctl --user call org.freedesktop.DBus /org/freedesktop/DBus org.freedesktop.DBus ReloadConfig
+Slice=session.slice
 
 [Install]
 Alias=dbus.service
