From: Sebastien Bacher <seb128@ubuntu.com>
Date: Thu, 25 Jul 2019 15:54:50 +0100
Subject: Ubuntu's dh-translations can't handle variables

Bug: https://bugzilla.gnome.org/784877

for GETTEXT_PACKAGE
===================================================================
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2aa1f32..055b7db 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -22,7 +22,7 @@ set(API_VERSION 1.2)
 set(CAL_API_VERSION 2.0)
 
 # Required for FindIntltool module
-set(GETTEXT_PACKAGE ${PROJECT_NAME})
+set(GETTEXT_PACKAGE "evolution-data-server")
 set(GETTEXT_PO_DIR ${CMAKE_SOURCE_DIR}/po)
 
 # Required for 'disttest' and 'distcheck' of DistTarget module
