From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Fri, 23 Aug 2019 22:13:14 -0400
Subject: drop python2 support

Forwarded: https://dev.gnupg.org/T5298
---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 4d32366..442bafb 100644
--- a/configure.ac
+++ b/configure.ac
@@ -425,7 +425,7 @@ if test "$found_py" = "1"; then
 	if test "$found_py" = "1" -o "$found_py3" = "1"; then
 	  # Reset everything, so that we can look for another Python.
           m4_foreach([mym4pythonver],
-                     [[2.7],[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[all]],
+                     [[3.4],[3.5],[3.6],[3.7],[3.8],[3.9],[all]],
            [unset PYTHON
 	    unset PYTHON_VERSION
 	    unset PYTHON_CPPFLAGS
