From b400fce0d3a8db7ed1038ca0645d513d28b88683 Mon Sep 17 00:00:00 2001
From: Alexandre Ghiti <alexandre.ghiti@canonical.com>
Date: Thu, 31 Mar 2022 14:09:55 +0200
Subject: [PATCH] lang: python: tests: Fix FTBFS caused by missing PYTHON

---
 lang/python/tests/Makefile.am | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/lang/python/tests/Makefile.am b/lang/python/tests/Makefile.am
index 62dddaa..6eec156 100644
--- a/lang/python/tests/Makefile.am
+++ b/lang/python/tests/Makefile.am
@@ -70,7 +70,8 @@ check: xcheck
 .PHONY: xcheck
 
 xcheck:	all
-	$(TESTS_ENVIRONMENT) $(PYTHON) $(srcdir)/run-tests.py \
+	for PYTHON in $(PYTHONS); do true; done; \
+	$(TESTS_ENVIRONMENT) $$PYTHON $(srcdir)/run-tests.py \
 	  --interpreters="$(PYTHONS)" --srcdir=$(srcdir) $(TESTFLAGS) \
 	  $(XTESTS)
 
-- 
2.32.0

