Description: Pull build-timestamp from debian/changelog.
Author: Chris Lamb <lamby@debian.org>
Origin: vendor
Forwarded: no
Bug-Debian: https://bugs.debian.org/783465
Last-Update: 2015-04-28

--- a/doc-docbook/GenLocalParams
+++ b/doc-docbook/GenLocalParams
@@ -3,7 +3,7 @@
 output="${1:-local_params}"
 : ${EXIM_VER:?Need the EXIM_VER environment variable}
 
-nicedate="$(date +"%d %b %Y")"
+nicedate="$(LC_ALL=C date --utc --date="$(dpkg-parsechangelog -l../debian/changelog -SDate)" +"%d %b %Y")"
 
 if which locale >/dev/null; then
     charset="$(locale | grep ^LC_CTYPE=)"
