Description: make test work with older EUMM versions
Origin: vendor
Forwarded: not-needed
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2017-10-21

--- a/t/27inline_maker.t
+++ b/t/27inline_maker.t
@@ -10,6 +10,7 @@
 use Cwd;
 use File::Copy::Recursive qw(rcopy);
 use autodie;
+use ExtUtils::MakeMaker;
 
 my @make_targets = qw(test install);
 my $CLEANUP = 1;
@@ -26,8 +27,14 @@
 
 my $lib_dir = File::Spec->rel2abs('lib');
 my $base_dir = File::Spec->rel2abs($TestInlineSetup::DIR);
-my $src_dir = File::Spec->catdir($base_dir, 'src dir');
-my $inst_dir = File::Spec->catdir($base_dir, 'inst dir');
+my ($src_dir, $inst_dir);
+if ($ExtUtils::MakeMaker::VERSION >= 7.00) {
+  $src_dir = File::Spec->catdir($base_dir, 'src dir');
+  $inst_dir = File::Spec->catdir($base_dir, 'inst dir');
+} else {
+  $src_dir = File::Spec->catdir($base_dir, 'srcdir');
+  $inst_dir = File::Spec->catdir($base_dir, 'instdir');
+}
 mkpath $inst_dir;
 
 my $cwd = getcwd;
