From: Jochen Friedrich <jochen@scram.de>
Subject: Clean up comiler warnings (Closes: #226561)
Forwarded: yes

--- a/src/toast.c
+++ b/src/toast.c
@@ -251,8 +251,8 @@ static char * emalloc P1((len), size_t l
 {
 	char * s;
 	if (!(s = malloc(len))) {
-		fprintf(stderr, "%s: failed to malloc %d bytes -- abort\n",
-			progname, (int)len);
+		fprintf(stderr, "%s: failed to malloc %ld bytes -- abort\n",
+			progname, (long)len);
 		onintr();
 		exit(1);
 	}
@@ -386,7 +386,7 @@ static void update_times P0()
 		ut[0] = instat.st_atime;
 		ut[1] = instat.st_mtime;
 
-		(void) utime(outname, ut);
+		(void) utime(outname, (struct utimbuf *)ut);
 
 #endif	/* UTIMBUF */
 	}
@@ -586,8 +586,8 @@ static int process_decode P0()
 
 		if (cc != sizeof(s)) {
 			if (cc >= 0) fprintf(stderr,
-			"%s: incomplete frame (%d byte%s missing) from %s\n",
-					progname, (int)(sizeof(s) - cc),
+			"%s: incomplete frame (%ld byte%s missing) from %s\n",
+					progname, (long)(sizeof(s) - cc),
 					"s" + (sizeof(s) - cc == 1),
 					inname ? inname : "stdin" );
 			gsm_destroy(r);
