Index: darts-0.32/Makefile.am
===================================================================
--- darts-0.32.orig/Makefile.am	2012-05-29 21:23:45.000000000 +0900
+++ darts-0.32/Makefile.am	2012-05-29 21:25:49.990745662 +0900
@@ -1,6 +1,6 @@
 SUBDIRS = tests
 EXTRA_DIRS = doc
-pkglibexecdir = ${libexecdir}/darts
+pkglibexecdir = ${prefix}/bin
 pkglibexec_PROGRAMS = mkdarts darts
 always_built_SUBDIRS = tests
 EXTRA_DIST = darts.spec BSD LGPL
