Index: darts-0.32/configure.in
===================================================================
--- darts-0.32.orig/configure.in	2012-05-29 21:31:50.607226294 +0900
+++ darts-0.32/configure.in	2012-05-29 21:31:50.619226312 +0900
@@ -33,7 +33,7 @@
 
 dnl check C++ features
 dnl
-AC_DEFUN(CHECK_CXX_STL, [
+AC_DEFUN([CHECK_CXX_STL], [
   AC_MSG_CHECKING(if ${CXX-c++} supports stl <$1> (required))
   AC_TRY_COMPILE(
 	[
@@ -110,7 +110,7 @@
 dnl
 dnl check C++ features
 dnl
-AC_DEFUN(CHECK_CXX_STL, [
+AC_DEFUN([CHECK_CXX_STL], [
   AC_MSG_CHECKING(if ${CXX-c++} supports stl <$1> (required))
   AC_TRY_COMPILE(
         [
