Index: darts-0.32/configure.in
===================================================================
--- darts-0.32.orig/configure.in	2008-03-22 20:07:51.000000000 +0900
+++ darts-0.32/configure.in	2012-05-29 20:19:42.251113458 +0900
@@ -27,8 +27,8 @@
 dnl check gcc
 dnl
 if test -n "$GCC"; then
-   CFLAGS="-O3 -Wall"
-   CXXFLAGS="-O3 -Wall"
+   AM_CFLAGS="-O3 -Wall"
+   AM_CXXFLAGS="-O3 -Wall"
 fi
 
 dnl check C++ features
