Description: Fixes a compilation error with Java 8
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: https://jira.terracotta.org/jira/browse/EHC-1081
--- a/src/main/java/net/sf/ehcache/config/CacheConfiguration.java
+++ b/src/main/java/net/sf/ehcache/config/CacheConfiguration.java
@@ -2543,7 +2543,7 @@
     /**
      * Accessor
      */
-    public List getCacheEventListenerConfigurations() {
+    public List<CacheEventListenerFactoryConfiguration> getCacheEventListenerConfigurations() {
         return cacheEventListenerConfigurations;
     }
 
