Description: Fixes the compatibility with the Portlet API 2.0
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/src/test/java/org/apache/commons/fileupload/portlet/MockPortletActionRequest.java
+++ b/src/test/java/org/apache/commons/fileupload/portlet/MockPortletActionRequest.java
@@ -268,4 +268,28 @@
         this.characterEncoding = characterEncoding;
     }
 
+    @Override
+    public String getMethod() {
+        return "GET";
+    }
+
+    @Override
+    public javax.servlet.http.Cookie[] getCookies() {
+        return null;
+    }
+
+    @Override
+    public Map<String,String[]> getPublicParameterMap() {
+        return null;
+    }
+
+    @Override
+    public Map<String,String[]> getPrivateParameterMap() {
+        return null;
+    }
+
+    @Override
+    public String getWindowID() {
+        return null;
+    }
 }
