description: time.clock is removed in python 3.8
author: Michael Gilbert <mgilbert@debian.org>

--- a/xml/genheaders.py
+++ b/xml/genheaders.py
@@ -75,10 +75,10 @@ if __name__ == '__main__':
 startTime = None
 def startTimer():
     global startTime
-    startTime = time.clock()
+    startTime = time.perf_counter()
 def endTimer(msg):
     global startTime
-    endTime = time.clock()
+    endTime = time.perf_counter()
     if (timeit):
         write(msg, endTime - startTime)
         startTime = None
