description: automatically create directories for output files
author: Michael Gilbert <mgilbert@debian.org>

--- a/xml/reg.py
+++ b/xml/reg.py
@@ -16,6 +16,7 @@
 
 import io,os,re,string,sys
 from lxml import etree
+from pathlib import Path
 
 def write(*args, **kwargs):
     file = kwargs.pop('file', sys.stdout)
@@ -386,6 +387,8 @@ class OutputGenerator:
         # Open specified output file. Not done in constructor since a
         # Generator can be used without writing to a file.
         if (self.genOpts.filename != None):
+            directory = os.path.dirname(self.genOpts.filename)
+            Path(directory).mkdir(parents=True, exist_ok=True)
             self.outFile = open(self.genOpts.filename, 'w')
         else:
             self.outFile = sys.stdout
