author: Michael Gilbert <mgilbert@debian.org>
description: use changelog date for timestamps

--- a/xml/genheaders.py
+++ b/xml/genheaders.py
@@ -17,6 +17,14 @@
 import sys, time, pdb, string, cProfile
 from reg import *
 
+import dateutil.parser
+import debian.changelog
+
+fid = open('../debian/changelog')
+changelog = debian.changelog.Changelog(fid)
+date = dateutil.parser.parse(changelog.date)
+fid.close()
+
 # debug - start header generation in debugger
 # dump - dump registry after loading
 # profile - enable Python profiling
@@ -238,28 +246,28 @@ glesProtoPrefixStrings = [
 
 # Insert generation date in a comment for headers not having *GLEXT_VERSION macros
 genDateCommentString = [
-    format('/* Generated on date %s */' % time.strftime('%Y%m%d')),
+    format('/* Generated on date %s */' % date.strftime('%Y%m%d')),
     ''
 ]
 
 # GL_GLEXT_VERSION is defined only in glext.h
 glextVersionStrings = [
-    format('#define GL_GLEXT_VERSION %s' % time.strftime('%Y%m%d')),
+    format('#define GL_GLEXT_VERSION %s' % date.strftime('%Y%m%d')),
     ''
 ]
 # WGL_WGLEXT_VERSION is defined only in wglext.h
 wglextVersionStrings = [
-    format('#define WGL_WGLEXT_VERSION %s' % time.strftime('%Y%m%d')),
+    format('#define WGL_WGLEXT_VERSION %s' % date.strftime('%Y%m%d')),
     ''
 ]
 # GLX_GLXEXT_VERSION is defined only in glxext.h
 glxextVersionStrings = [
-    format('#define GLX_GLXEXT_VERSION %s' % time.strftime('%Y%m%d')),
+    format('#define GLX_GLXEXT_VERSION %s' % date.strftime('%Y%m%d')),
     ''
 ]
 # EGL_EGLEXT_VERSION is defined only in eglext.h
 eglextVersionStrings = [
-    format('#define EGL_EGLEXT_VERSION %s' % time.strftime('%Y%m%d')),
+    format('#define EGL_EGLEXT_VERSION %s' % date.strftime('%Y%m%d')),
     ''
 ]
 
