Description: Fixes the compatibility with Java 11
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/test/src/java/org/jdom/test/cases/TestElement.java
+++ b/test/src/java/org/jdom/test/cases/TestElement.java
@@ -68,7 +68,6 @@
 import java.io.*;
 import java.util.*;
 import org.jdom.output.*;
-import org.omg.PortableInterceptor.SUCCESSFUL;
 
 public final class TestElement
         extends junit.framework.TestCase {
