Description: Ignore the google-java-format plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -35,7 +35,6 @@
   }
 }
 
-plugins { id 'com.github.sherter.google-java-format' version '0.9' apply false }
 plugins { id 'me.champeau.gradle.jmh' version '0.5.2' apply false }
 plugins { id 'org.sonarqube' version '3.0' apply false }
 
@@ -172,10 +171,6 @@
 // All projects are checked with googleJavaFormat, Checkstyle and PMD, 
 // and tested with :asm-test and JUnit.
 subprojects {
-  apply plugin: 'com.github.sherter.google-java-format'
-  googleJavaFormat.toolVersion = '1.9'
-  googleJavaFormat.exclude 'src/resources/java/**/*'
-  
   // Check the coding style with Checkstyle. Fail in case of error or warning.
   apply plugin: 'checkstyle'
   checkstyle.configFile = file("${rootDir}/tools/checkstyle.xml")
