Description: Ignore the SonarQube plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -35,7 +35,6 @@
   }
 }
 
-plugins { id 'org.sonarqube' version '3.0' apply false }
 
 description = 'ASM, a very small and fast Java bytecode manipulation framework'
 
@@ -298,13 +297,6 @@
       'Implementation-Version': "${version}")
   }
 
-  // Apply the SonarQube plugin to monitor the code quality of the project.
-  // Use with 'gradlew sonarqube -Dsonar.host.url=https://sonarqube.ow2.org'.
-  apply plugin: 'org.sonarqube'
-  sonarqube {
-    properties { property 'sonar.projectKey', "ASM:${project.name}" }
-  }
-
   // Add a task to generate a private javadoc and add it as a dependency of the
   // 'check' task.
   task privateJavadoc(type: Javadoc) {    
