Description: Disables the test coverage report
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -206,18 +206,6 @@
 // backward compatibility, retrofited to Java 1.5, packaged with biz.aQute.bnd
 // and uploaded to Maven with a POM, sources and Javadoc.
 configure(subprojects.findAll { it.provides }) {
-  // Code coverage configuration.
-  jacoco.toolVersion = '0.8.7'
-  jacocoTestReport {
-    reports { xml.enabled true }
-    classDirectories.setFrom(sourceSets.main.output.classesDirs)
-  }
-  jacocoTestCoverageVerification {
-    classDirectories.setFrom(sourceSets.main.output.classesDirs)
-    violationRules.rule { limit { minimum = 0.95; counter = 'INSTRUCTION' } }
-  }
-  check.dependsOn jacocoTestCoverageVerification
-
   // Create one backward compatibility checking task for each 'sigtest-*' file
   // in test/resources, and make the 'check' task depend on all these tasks.
   if (file('src/test/resources/').exists()) {
