Description: Skip debian directory.
Author: Alessio Treglia <quadrispro@ubuntu.com>
Forwarded: cuse@users.sourceforge.net

Index: libgig/Makefile.am
===================================================================
--- libgig.orig/Makefile.am
+++ libgig/Makefile.am
@@ -1,7 +1,7 @@
 # require automake >= 1.5 (mandatory for option dist-bzip2)
 AUTOMAKE_OPTIONS = 1.5 dist-bzip2
 
-SUBDIRS = doc man src debian win32 osx
+SUBDIRS = doc man src win32 osx
 
 # to prevent compile errors on some systems
 AM_CXXFLAGS = -pedantic
Index: libgig/configure.ac
===================================================================
--- libgig.orig/configure.ac
+++ libgig/configure.ac
@@ -149,7 +149,6 @@ AC_CONFIG_FILES([\
     man/korg2gig.1 \
     man/akaidump.1 \
     man/akaiextract.1 \
-    debian/Makefile \
     osx/Makefile \
     osx/libgig.xcodeproj/Makefile \
     win32/Makefile \
