Description: Fix the compatibility with CGLIB 3
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/project/hibernate-core/src/main/java/org/hibernate/bytecode/cglib/CglibClassTransformer.java
+++ b/project/hibernate-core/src/main/java/org/hibernate/bytecode/cglib/CglibClassTransformer.java
@@ -80,7 +80,7 @@
 		}
 
 		String[] names = ClassNameReader.getClassInfo( reader );
-		ClassWriter w = new DebuggingClassWriter( ClassWriter.COMPUTE_MAXS  );
+		DebuggingClassWriter w = new DebuggingClassWriter( ClassWriter.COMPUTE_MAXS  );
 		ClassTransformer t = getClassTransformer( names );
 		if ( t != null ) {
 			if ( log.isDebugEnabled() ) {
