Description: Ignore the code coverage plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -37,7 +37,6 @@
 apply from: 'gradle/root/java-compatibility-check.gradle'
 apply from: 'gradle/java-library.gradle'
 apply from: 'gradle/license.gradle'
-apply from: 'gradle/root/coverage.gradle'
 
 apply from: 'gradle/mockito-core/inline-mock.gradle'
 apply from: 'gradle/mockito-core/osgi.gradle'
