Description: Ignore the build scan plugin
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -16,7 +16,6 @@
 }
 
 plugins {
-    id 'com.gradle.build-scan' version '1.15.1'
 }
 
 description = 'Mockito mock objects library core API and implementation'
@@ -90,9 +89,4 @@
     distributionSha256Sum = 'e66e69dce8173dd2004b39ba93586a184628bc6c28461bc771d6835f7f9b0d28'
 }
 
-//Posting Build scans to https://scans.gradle.com
-buildScan {
-    licenseAgreementUrl = 'https://gradle.com/terms-of-service'
-    licenseAgree = 'yes'
-}
 
