diff -Nur graphicsmagick-1.3.20/Magick++/Makefile.am graphicsmagick-1.3.20/Magick++/Makefile.am
--- graphicsmagick-1.3.20/Magick++/Makefile.am	2014-08-16 20:33:23.000000000 +0000
+++ graphicsmagick-1.3.20/Magick++/Makefile.am	2014-08-23 19:32:23.198811403 +0000
@@ -168,7 +168,7 @@
 Magick___demo_button_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___demo_demo_SOURCES		= Magick++/demo/demo.cpp
-Magick___demo_demo_LDADD		= $(LIBMAGICKPP)
+Magick___demo_demo_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___demo_demo_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___demo_detrans_SOURCES		= Magick++/demo/detrans.cpp
@@ -176,69 +176,69 @@
 Magick___demo_detrans_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___demo_flip_SOURCES		= Magick++/demo/flip.cpp
-Magick___demo_flip_LDADD		= $(LIBMAGICKPP)
+Magick___demo_flip_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___demo_flip_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___demo_gravity_SOURCES		= Magick++/demo/gravity.cpp
-Magick___demo_gravity_LDADD		= $(LIBMAGICKPP)
+Magick___demo_gravity_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___demo_gravity_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___demo_piddle_SOURCES		= Magick++/demo/piddle.cpp
-Magick___demo_piddle_LDADD		= $(LIBMAGICKPP)
+Magick___demo_piddle_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___demo_piddle_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___demo_shapes_SOURCES		= Magick++/demo/shapes.cpp
-Magick___demo_shapes_LDADD		= $(LIBMAGICKPP)
+Magick___demo_shapes_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___demo_shapes_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___demo_zoom_SOURCES		= Magick++/demo/zoom.cpp
-Magick___demo_zoom_LDADD		= $(LIBMAGICKPP)
+Magick___demo_zoom_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___demo_zoom_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_appendImages_SOURCES	= Magick++/tests/appendImages.cpp
-Magick___tests_appendImages_LDADD	= $(LIBMAGICKPP)
+Magick___tests_appendImages_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_appendImages_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_attributes_SOURCES	= Magick++/tests/attributes.cpp
-Magick___tests_attributes_LDADD		= $(LIBMAGICKPP)
+Magick___tests_attributes_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_attributes_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_averageImages_SOURCES	= Magick++/tests/averageImages.cpp
-Magick___tests_averageImages_LDADD	= $(LIBMAGICKPP)
+Magick___tests_averageImages_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_averageImages_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_coalesceImages_SOURCES	= Magick++/tests/coalesceImages.cpp
-Magick___tests_coalesceImages_LDADD	= $(LIBMAGICKPP)
+Magick___tests_coalesceImages_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_coalesceImages_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_coderInfo_SOURCES	= Magick++/tests/coderInfo.cpp
-Magick___tests_coderInfo_LDADD		= $(LIBMAGICKPP)
+Magick___tests_coderInfo_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_coderInfo_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_color_SOURCES		= Magick++/tests/color.cpp
-Magick___tests_color_LDADD		= $(LIBMAGICKPP)
+Magick___tests_color_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_color_CPPFLAGS		= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_colorHistogram_SOURCES	= Magick++/tests/colorHistogram.cpp
-Magick___tests_colorHistogram_LDADD	= $(LIBMAGICKPP)
+Magick___tests_colorHistogram_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_colorHistogram_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_exceptions_SOURCES	= Magick++/tests/exceptions.cpp
-Magick___tests_exceptions_LDADD		= $(LIBMAGICKPP)
+Magick___tests_exceptions_LDADD		= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_exceptions_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_montageImages_SOURCES	= Magick++/tests/montageImages.cpp
-Magick___tests_montageImages_LDADD	= $(LIBMAGICKPP)
+Magick___tests_montageImages_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_montageImages_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_morphImages_SOURCES	= Magick++/tests/morphImages.cpp
-Magick___tests_morphImages_LDADD	= $(LIBMAGICKPP)
+Magick___tests_morphImages_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_morphImages_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_readWriteBlob_SOURCES	= Magick++/tests/readWriteBlob.cpp
-Magick___tests_readWriteBlob_LDADD	= $(LIBMAGICKPP)
+Magick___tests_readWriteBlob_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_readWriteBlob_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
 
 Magick___tests_readWriteImages_SOURCES	= Magick++/tests/readWriteImages.cpp
-Magick___tests_readWriteImages_LDADD	= $(LIBMAGICKPP)
+Magick___tests_readWriteImages_LDADD	= $(LIBMAGICKPP) $(LIBMAGICK)
 Magick___tests_readWriteImages_CPPFLAGS	= $(MAGICKPP_CPPFLAGS)
diff -Nur graphicsmagick-1.3.20/wand/Makefile.am graphicsmagick-1.3.20/wand/Makefile.am
--- graphicsmagick-1.3.20/wand/Makefile.am	2014-08-16 20:33:25.000000000 +0000
+++ graphicsmagick-1.3.20/wand/Makefile.am	2014-08-23 19:33:26.478814666 +0000
@@ -63,12 +63,12 @@
 
 wand_drawtest_LDADD        = $(LIBWAND)
 wand_drawtest_CPPFLAGS     = $(WAND_CPPFLAGS)
-wand_drawtest_LDFLAGS      = $(LDFLAGS)
+wand_drawtest_LDFLAGS      = $(LDFLAGS) $(LIBMAGICK)
 wand_drawtest_SOURCES      = wand/drawtest.c
 
 wand_wandtest_LDADD        = $(LIBWAND)
 wand_wandtest_CPPFLAGS     = $(WAND_CPPFLAGS)
-wand_wandtest_LDFLAGS      = $(LDFLAGS)
+wand_wandtest_LDFLAGS      = $(LDFLAGS) $(LIBMAGICK)
 wand_wandtest_SOURCES      = wand/wandtest.c
 
 WAND_CHECK_PGRMS = \
