Description: compile magick/semaphore.c with -O0 on ppc64el
Bug-Debian: https://bugs.debian.org/837719
Forwarded: no
Author: Laszlo Boszormenyi (GCS) <gcs@debian.org>
Last-Update: 2017-12-10

---

--- graphicsmagick-1.3.25.orig/magick/semaphore.c
+++ graphicsmagick-1.3.25/magick/semaphore.c
@@ -46,6 +46,11 @@
 #  define USE_OPENMP_LOCKS 1
 #endif
 
+#if defined(_ARCH_PPC64)
+# pragma GCC push_options
+# pragma GCC optimize("O0")
+#endif
+
 #if defined(USE_PTHREAD_LOCKS)
 #  include <pthread.h>
 #  define PTHREAD_MUTEX_DESTROY(semaphore_mutex)                        \
@@ -649,3 +654,7 @@ MagickExport void UnlockSemaphoreInfo(Se
   LeaveCriticalSection(&semaphore_info->mutex);
 #endif /* defined(USE_WIN32_LOCKS) */
 }
+
+#if defined(_ARCH_PPC64)
+# pragma GCC pop_options
+#endif
