Description: Remove wsmeext.sphinxext from doc's conf.py
Author: Thomas Goirand <zigo@debian.org>
Forwarded: not-needed
Bug-Debian: https://bugs.debian.org/963634
Last-Update: 2020-09-16

--- a/doc/source/conf.py
+++ b/doc/source/conf.py
@@ -40,7 +40,6 @@ os.environ['DJANGO_SETTINGS_MODULE'] = '
 extensions = [
     'openstackdocstheme',
     'sphinx.ext.autodoc',
-    'wsmeext.sphinxext',
     'sphinx.ext.coverage',
     'sphinx.ext.viewcode',
     'sphinxcontrib.pecanwsme.rest',
