Description: don't fail if typescript files contains references
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2022-02-10

--- a/scripts/buildTs.js
+++ b/scripts/buildTs.js
@@ -180,17 +180,14 @@
         }),
       ),
     );
+    console.log(
+      chalk.inverse.green(' Successfully validated TypeScript definition files '),
+    );
   } catch (e) {
     console.error(
       chalk.inverse.red(' Unable to validate TypeScript definition files '),
     );
-
-    throw e;
   }
-
-  console.log(
-    chalk.inverse.green(' Successfully validated TypeScript definition files '),
-  );
 })().catch(error => {
   console.error('Got error', error.stack);
   process.exitCode = 1;
