Description: force module path to match system paths
Author: Xavier Guimard <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2021-10-28

--- a/packages/jest-config/src/Defaults.ts
+++ b/packages/jest-config/src/Defaults.ts
@@ -43,7 +43,8 @@
   listTests: false,
   maxConcurrency: 5,
   maxWorkers: '50%',
-  moduleDirectories: ['node_modules'],
+// @ts-ignore
+  moduleDirectories: ['node_modules', ...process.config.variables.node_relative_path.split(':').map((str) => process.config.variables.node_prefix + '/' + str)],
   moduleFileExtensions: ['js', 'jsx', 'ts', 'tsx', 'json', 'node'],
   moduleNameMapper: {},
   modulePathIgnorePatterns: ['<rootDir>/.pc/'],
--- a/packages/jest-resolve/src/nodeModulesPaths.ts
+++ b/packages/jest-resolve/src/nodeModulesPaths.ts
@@ -23,7 +23,8 @@
   const modules =
     options && options.moduleDirectory
       ? Array.from(options.moduleDirectory)
-      : ['node_modules'];
+// @ts-ignore
+      : ['node_modules', ...process.config.variables.node_relative_path.split(':').map((str) => process.config.variables.node_prefix + '/' + str)];
 
   // ensure that `basedir` is an absolute path at this point,
   // resolving against the process' current working directory
--- a/packages/jest-resolve/src/resolver.ts
+++ b/packages/jest-resolve/src/resolver.ts
@@ -65,7 +65,8 @@
       extensions: options.extensions,
       hasCoreModules:
         options.hasCoreModules === undefined ? true : options.hasCoreModules,
-      moduleDirectories: options.moduleDirectories || ['node_modules'],
+// @ts-ignore
+      moduleDirectories: options.moduleDirectories || ['node_modules', ...process.config.variables.node_relative_path.split(':').map((str) => process.config.variables.node_prefix + '/' + str)],
       moduleNameMapper: options.moduleNameMapper,
       modulePaths: options.modulePaths,
       platforms: options.platforms,
