Description: prettier isn't available
Author: Xavier Guimard <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2021-01-11

--- a/packages/jest-config/src/ValidConfig.ts
+++ b/packages/jest-config/src/ValidConfig.ts
@@ -89,7 +89,7 @@
   onlyFailures: false,
   passWithNoTests: false,
   preset: 'react-native',
-  prettierPath: '<rootDir>/node_modules/prettier',
+  prettierPath: '',
   projects: ['project-a', 'project-b/'],
   reporters: [
     'default',
--- a/packages/jest-config/src/index.ts
+++ b/packages/jest-config/src/index.ts
@@ -201,7 +201,7 @@
     modulePathIgnorePatterns: options.modulePathIgnorePatterns,
     modulePaths: options.modulePaths,
     name: options.name,
-    prettierPath: options.prettierPath,
+    prettierPath: '',
     resetMocks: options.resetMocks,
     resetModules: options.resetModules,
     resolver: options.resolver,
