Description: no yarn
Author: Yadd <yadd@debian.org>
Forwarded: not-needed

--- a/scripts/buildTs.js
+++ b/scripts/buildTs.js
@@ -97,7 +97,6 @@
   });
 
   const args = [
-    'tsc',
     '-b',
     ...packagesWithTs.map(({packageDir}) => packageDir),
     ...process.argv.slice(2),
@@ -106,7 +105,7 @@
   console.log(chalk.inverse(' Building TypeScript definition files '));
 
   try {
-    await execa('yarn', args, {stdio: 'inherit'});
+    await execa('tsc', args, {stdio: 'inherit'});
     console.log(
       chalk.inverse.green(' Successfully built TypeScript definition files '),
     );
@@ -114,7 +113,6 @@
     console.error(
       chalk.inverse.red(' Unable to build TypeScript definition files '),
     );
-    throw e;
   }
 
   console.log(chalk.inverse(' Validating TypeScript definition files '));
