Description: temp workaround
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2021-08-29

--- a/scripts/buildTs.js
+++ b/scripts/buildTs.js
@@ -28,10 +28,8 @@
     fs.existsSync(path.resolve(p.packageDir, 'tsconfig.json')),
   );
 
-  const {stdout: allWorkspacesString} = await execa('yarn', [
-    'workspaces',
-    'list',
-    '--json',
+  const {stdout: allWorkspacesString} = await execa('cat', [
+    'debian/yarn-workspaces.json'
   ]);
 
   const workspacesWithTs = new Map(
