Author: Reiner Herrmann <reiner@reiner-h.de>
Description: Sort object files for deterministic linking order

--- a/GNUmakefile
+++ b/GNUmakefile
@@ -31,7 +31,7 @@
 PO4A_CONFIG = man/po4a/po4a.cfg
 
 libsrc = $(wildcard lib/*.c)
-libobj = $(libsrc:.c=.o)
+libobj = $(sort $(libsrc:.c=.o))
 
 src = $(libsrc) \
       $(wildcard grep-dctrl/*.c) \
@@ -39,7 +39,7 @@
       $(wildcard tbl-dctrl/*.c) \
       $(wildcard join-dctrl/*.c)
 
-obj = $(src:.c=.o)
+obj = $(sort $(src:.c=.o))
 
 exe =  	grep-dctrl/grep-dctrl \
 	sort-dctrl/sort-dctrl \
