Description: Support for gcc8
Origin: https://github.com/linux-test-project/lcov/commit/a5dd9529f9232b8d901a4d6eb9ae54cae179e5b3
Last-Updated: 2018-05-14
Forwarded: no

--- a/bin/geninfo
+++ b/bin/geninfo
@@ -2018,6 +2018,9 @@
 				# Skip instance-specific counts
 				next if ($line <= (scalar(@result) / 3));
 
+				# Skip instance-specific counts
+				next if ($line == $last_line);
+
 				$last_line = $line;
 				$last_block = $UNNAMED_BLOCK;
 				# Check for exclusion markers
@@ -2063,6 +2066,10 @@
 				# Strip unexecuted basic block marker
 				$count =~ s/\*$//;
 
+				 
+				# Strip unexecuted basic block marker
+				$count =~ s/\*$//;
+				
 				# <exec count>:<line number>:<source code>
 				if ($line eq "0")
 				{
