Index: lcov-1.14/Makefile
===================================================================
--- lcov-1.14.orig/Makefile
+++ lcov-1.14/Makefile
@@ -25,8 +25,8 @@ CFG_DIR := $(PREFIX)/etc
 BIN_DIR := $(PREFIX)/bin
 MAN_DIR := $(PREFIX)/share/man
 TMP_DIR := $(shell mktemp -d)
-FILES   := $(wildcard bin/*) $(wildcard man/*) README Makefile \
-	   $(wildcard rpm/*) lcovrc
+FILES   := $(shell LC_ALL=C ls bin/*) $(shell LC_ALL=C ls man/*) README Makefile \
+	   $(shell LC_ALL=C ls rpm/*) lcovrc
 
 .PHONY: all info clean install uninstall rpms test
 
