From 230e5e64ac95794b174708caf1c879124ae28a28 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Sat, 26 Mar 2022 16:32:55 -0700
Subject: [PATCH 06/11] parser: Set the DEBUG1 flag on profiles that use mqueue
 rules.

Signed-off-by: John Johansen <john.johansen@canonical.com>
---
 parser/mqueue.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/parser/mqueue.cc b/parser/mqueue.cc
index 4b7ae34f5..92f075060 100644
--- a/parser/mqueue.cc
+++ b/parser/mqueue.cc
@@ -204,6 +204,7 @@ int mqueue_rule::gen_policy_re(Profile &prof)
 		//	return RULE_NOT_SUPPORTED;
 	}
 
+	prof.flags.flags |= FLAG_DEBUG1;
 	/* always generate a label and mqueue entry */
 
 	//buffer << "(" << "\\x" << std::setfill('0') << std::setw(2) << std::hex << AA_CLASS_LABEL << "|)"; //is this required?
-- 
2.34.1

