Description: fix typos in POD
Author: Florian Schlichting <fsfs@debian.org>
Forwarded: https://rt.cpan.org/Ticket/Display.html?id=119000

--- a/lib/Mail/Sendmail.pm
+++ b/lib/Mail/Sendmail.pm
@@ -546,7 +546,7 @@
             || return fail("send $header: error");
     };
 
-    #- test diconnecting from network here, to see what happens
+    #- test disconnecting from network here, to see what happens
     #- print STDERR "DISCONNECT NOW!\n";
     #- sleep 4;
     #- print STDERR "trying to continue, expecting an error... \n";
@@ -758,7 +758,7 @@
 
 =item $mail{auth}
 
-This must be a reference to a hash containg all your authentication options:
+This must be a reference to a hash containing all your authentication options:
 
 $mail{auth} = \%options;
 or
@@ -968,7 +968,7 @@
 
 
   $mail{Smtp} = 'special_server.for-this-message-only.domain.com';
-  $mail{'X-custom'} = 'My custom additionnal header';
+  $mail{'X-custom'} = 'My custom additional header';
   $mail{'mESSaGE : '} = "The message key looks terrible, but works.";
   # cheat on the date:
   $mail{Date} = Mail::Sendmail::time_to_date( time() - 86400 );
@@ -1019,7 +1019,7 @@
 spam, or ...?)
 
 Thanks to the many users who sent me feedback, bug reports, suggestions, etc.
-And please excuse me if I forgot to answer your mail. I am not always reliabe
+And please excuse me if I forgot to answer your mail. I am not always reliable
 in answering mail. I intend to set up a mailing list soon.
 
 Last revision: 06.02.2003. Latest version should be available on
