Index: dictionaries-common/support/emacsen/ispell.el
===================================================================
--- dictionaries-common.orig/support/emacsen/ispell.el	2014-10-14 18:27:15.447534550 +0200
+++ dictionaries-common/support/emacsen/ispell.el	2014-10-14 18:27:15.443534559 +0200
@@ -3234,8 +3234,10 @@
                      "ispell-region: First skip: %s at (pos,line,column): (%s,%s,%s).\n"
                      key
                      (save-excursion (goto-char skip-region-start) (point))
-                     (line-number-at-pos skip-region-start)
-                     (save-excursion (goto-char skip-region-start) (current-column)))))
+		     (if (featurep 'xemacs)
+			 (line-number skip-region-start)
+		       (line-number-at-pos skip-region-start))
+		     (save-excursion (goto-char skip-region-start) (current-column)))))
 	      (ispell-print-if-debug
                "ispell-region: Continue spell-checking with %s and %s dictionary...\n"
 		program-basename dictionary)
@@ -3276,7 +3278,9 @@
                              "ispell-region: Next skip: %s at (pos,line,column): (%s,%s,%s).\n"
                              key
                              (save-excursion (goto-char skip-region-start) (point))
-                             (line-number-at-pos skip-region-start)
+			     (if (featurep 'xemacs)
+				 (line-number skip-region-start)
+			       (line-number-at-pos skip-region-start))
                              (save-excursion (goto-char skip-region-start) (current-column))))
 			(set-marker skip-region-start nil))))
 		(setq reg-end (max (point)
@@ -3584,7 +3588,10 @@
 		;; This error occurs due to filter pipe problems
 		(let* ((ispell-pipe-word (car poss))
 		       (actual-point (marker-position word-start))
-		       (actual-line (line-number-at-pos actual-point))
+		       (actual-line
+			(if (featurep 'xemacs)
+			    (line-number actual-point)
+			  (line-number-at-pos actual-point)))
 		       (actual-column (save-excursion (goto-char actual-point)
                                                       (current-column))))
 		  (ispell-print-if-debug
