Author: Agustin Martin Domingo <agmartin@debian.org>
Description: Minor change in ispell pop-up menu

Index: dictionaries-common/support/emacsen/ispell.el
===================================================================
--- dictionaries-common.orig/support/emacsen/ispell.el	2014-10-14 18:27:19.235526382 +0200
+++ dictionaries-common/support/emacsen/ispell.el	2014-10-14 18:27:19.231526390 +0200
@@ -2238,7 +2238,7 @@
       (erase-buffer)
       (if guess
 	  (progn
-	    (insert "Affix rules generate and capitalize "
+	    (insert "Note: Affix rules generate and capitalize "
 		    "this word as shown below:\n\t")
 	    (while guess
 	      (if (> (+ 4 (current-column) (length (car guess)))
