Index: dictionaries-common/support/emacsen/ispell.el
===================================================================
--- dictionaries-common.orig/support/emacsen/ispell.el	2014-10-14 18:27:27.247508977 +0200
+++ dictionaries-common/support/emacsen/ispell.el	2014-10-14 18:27:27.239508995 +0200
@@ -1132,7 +1132,10 @@
 	(insert-file-contents alias-file)
 	;; Look for a line "add FOO.multi", extract FOO
 	(when (search-forward-regexp "^add \\([^.]+\\)\\.multi" nil t)
-	  (let* ((aliasname (file-name-base alias-file))
+	  ;; Use expanded (file-name-sans-extension (file-name-nondirectory ...))
+	  ;; instead of compact (file-name-base ...), unsupported by XEmacs
+	  (let* ((aliasname (file-name-sans-extension
+			     (file-name-nondirectory alias-file)))
 		 (already-exists-p (assoc aliasname alist))
 		 (realname (match-string 1))
 		 (realdict (assoc realname alist)))
