Author: Agustin Martin Domingo <agmartin@debian.org>
Description: Change ispell-version to reflect Debian changes

Index: dictionaries-common/support/emacsen/ispell.el
===================================================================
--- dictionaries-common.orig/support/emacsen/ispell.el	2014-10-14 18:27:28.471506303 +0200
+++ dictionaries-common/support/emacsen/ispell.el	2014-10-14 18:27:28.467506311 +0200
@@ -832,6 +832,14 @@
 
 (defconst ispell-version "ispell.el 3.6 - 7-Jan-2003")
 
+;;; -----------------------------------------------------------------------
+;;; Tell people that this version has been patched for the Debian system
+;;; (dictionaries-common package)
+
+(defconst ispell-version (concat ispell-version " (+ Debian `dictionaries-common' changes)"))
+
+;;; End of Debian changes
+;;; -----------------------------------------------------------------------
 
 (defun ispell-check-version (&optional interactivep)
   "Ensure that `ispell-program-name' is valid and has the correct version.
