Description: Fixes the compatibility with Jetty 9.4
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: no
--- a/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java
+++ b/logback-access/src/main/java/ch/qos/logback/access/jetty/RequestLogImpl.java
@@ -209,11 +209,6 @@
         started = false;
     }
 
-    @Override
-    public boolean isRunning() {
-        return started;
-    }
-
     public void setFileName(String fileName) {
         this.fileName = fileName;
     }
@@ -227,26 +222,6 @@
         return started;
     }
 
-    @Override
-    public boolean isStarting() {
-        return false;
-    }
-
-    @Override
-    public boolean isStopping() {
-        return false;
-    }
-
-    @Override
-    public boolean isStopped() {
-        return !started;
-    }
-
-    @Override
-    public boolean isFailed() {
-        return false;
-    }
-
     public boolean isQuiet() {
         return quiet;
     }
@@ -310,14 +285,4 @@
         return fai.getFilterChainDecision(event);
     }
 
-    @Override
-    public void addLifeCycleListener(Listener listener) {
-        // we'll implement this when asked
-    }
-
-    @Override
-    public void removeLifeCycleListener(Listener listener) {
-        // we'll implement this when asked
-    }
-
 }
