Description: adjust path to script when run under autopkgtest
Origin: vendor
Forwarded: not-needed
Author: gregor herrmann <gregoa@debian.org>
Last-Update: 2015-10-30

--- a/t/03_translit.t
+++ b/t/03_translit.t
@@ -3,7 +3,8 @@
 
 my $truth;
 
-my $script = "./blib/script/translit";
+my $bindir = $ENV{ADTTMP} ? '/usr/bin' : './blib/script';
+my $script = "$bindir/translit";
 
 
 $truth = -e $script;
