--- a/crude.cfg	2022-01-18 11:54:12.004945370 +0100
+++ b/crude.cfg	2022-01-18 11:50:29.126071335 +0100
@@ -336,11 +336,11 @@

 # COMMON EXTENSIONS TO GENERAL MIDI DRUMSET 0 (notes 25-34)
 # Program  Snare Roll
-#      25  025_Snare_Roll.pat (unavailable)
+       25  Drum_000/025_Snare_Roll.pat
 # Program  Snap
-#      26  026_Snap.pat (unavailable)
+       26  Drum_000/026_Snap.pat
 # Program  High_Q
-#      27  027_High_Q.pat (unavailable)
+       27  Drum_000/027_High_Q.pat
 # Program  Slap (gus slap)
 #      28  028_Slap.pat (unavailable)
 # Program  Scratch1 (gus scratch1)
@@ -348,13 +348,13 @@
 # Program  Scratch2 (gus scratch2)
 #      30  030_Scratch_2_pull.pat (unavailable)
 # Program  Sticks (gus sticks)
-#      31  031_Sticks.pat (unavailable)
+       31  Drum_000/031_Sticks.pat
 # Program  Square Click (gus SqrClick)
-#      32  032_square_click.pat (unavailable)
+       32  Drum_000/032_Square_Click.pat
 # Program  Metronome Click (gus metclick)
-#      33  033_Metronome_Click (unavailable)
+       33  Drum_000/033_Metronome_Click.pat
 # Program  Metronome Bell (gus metbell)
-#      34  034_Metronome_Bell.pat (unavailable)
+       34  Drum_000/034_Metronome_Bell.pat

 # BEG GENERAL MIDI DRUMSET 0
 # Program  Acoustic Kick/Acoustic Bass Drum (gus kick1)
@@ -453,6 +453,8 @@
 # Program  Open Triangle (gus triangl2)
        81  Drum_000/081_Triangle_2_Open.pat
 # END GENERAL MIDI DRUMSET 0
+       82  Drum_000/082_Shaker.pat
+       84  Drum_000/084_Belltree.pat

 # COMMON EXTENSIONS TO GENERAL MIDI DRUMSET 0 (notes 82-87)
 # Program  Shaker (gus shaker)
