Description: Makes the build scripts behave properly.
Author: Kai-Chung Yan (殷啟聰)
Last-Update: 2018-02-07
--- a/build.gradle
+++ b/build.gradle
@@ -8,16 +8,12 @@
     }
 }
 
-plugins {
-    id "com.jfrog.bintray" version "1.2"
-}
-
-version = '1.58'
+group = 'com.beust'
+version = System.env.DEB_VERSION_UPSTREAM
 
 apply plugin: 'java'
 apply plugin: 'eclipse'
 apply plugin: 'idea'
-apply from: 'gradle/publishing.gradle'
 
 repositories {
     jcenter()
@@ -40,5 +36,4 @@
 
 test {
     // enable TestNG support (default is JUnit)
-    useTestNG()
 }
