Description: Preserves the Java 8 compatibility
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/build.gradle
+++ b/build.gradle
@@ -23,6 +23,11 @@
     testCompile 'org.testng:testng:6.9.13'
 }
 
+compileJava   {
+    sourceCompatibility = '1.8'
+    targetCompatibility = '1.8'
+}
+
 task sourceJar(type: Jar) {
     group 'Build'
     description 'An archive of the source code'
