From: =?utf-8?q?Ferenc_W=C3=A1gner?= <wferi@debian.org>
Date: Mon, 19 Dec 2016 22:57:22 +0100
Subject: Don't link dlm_stonith against libxml2

---
 fence/Makefile | 1 -
 1 file changed, 1 deletion(-)

diff --git a/fence/Makefile b/fence/Makefile
index 5cd711e..fe2b97b 100644
--- a/fence/Makefile
+++ b/fence/Makefile
@@ -23,7 +23,6 @@ CFLAGS += -I../include
 CFLAGS += $(shell pkg-config --cflags pacemaker-fencing)
 
 LDFLAGS += -Wl,-z,relro -Wl,-z,defs -pie
-LDFLAGS += `xml2-config --libs`
 LDFLAGS += -ldl
 
 all: $(BIN_TARGET)
