From: Vladimir Rutsky <altsysrq@gmail.com>
Subject: remove debug messages produced by pcxtoppm (about 241 line)

--- a/ppm/pcxtoppm.c
+++ b/ppm/pcxtoppm.c
@@ -639,8 +639,6 @@
 }
 
 
-static int rowsRead=0;
-
 static int
 GetByte(FILE * const fp) {
 
@@ -684,8 +682,6 @@
             pcxrow[bytesGenerated++] = c;
             --repetitionsLeft;
         } else {
- if (rowsRead==241)
-     pm_message("Reading Byte %ld", ftell(ifP));
             c = GetByte(ifP);
             if ((c & 0xc0) != 0xc0)
                 /* This is a 1-shot byte, not a repetition count */
@@ -693,12 +689,9 @@
             else {
                 /* This is a repetition count for the following byte */
                 repetitionsLeft = c & 0x3f;
-if (rowsRead == 241)
-pm_message("doing run of %d of Byte %ld in Row 241", repetitionsLeft, ftell(ifP));
                 c = GetByte(ifP);
             }
         }
     }
-rowsRead++;
 }
 
