From: Moritz Muehlenhoff <jmm@debian.org>
Subject: Inject hardened build flags into the build system

--- a/Makefile.config.in
+++ b/Makefile.config.in
@@ -102,7 +102,12 @@
 # -ansi and -Werror should work too, but are not included
 # by default because there's no point in daring the build to fail.
 # -pedantic isn't a problem because it causes at worst a warning.
-CFLAGS = -pedantic -O3 -Wall -Wno-uninitialized
+
+CFLAGS = `dpkg-buildflags --get CFLAGS`
+CFLAGS += -pedantic -O3 -Wall -Wno-uninitialized
+CFLAGS += `dpkg-buildflags --get CPPFLAGS`
+CFLAGS += -fcommon
+
 # On DEC Tru64 4.0F (at least), you need -DLONG_32 for ppmtompeg.
 #Tru64:
 #CFLAGS = -O2 -std1 -DLONG_32
@@ -134,7 +138,7 @@
 NEED_RUNTIME_PATH = N
 #NEED_RUNTIME_PATH = Y
 
-LDFLAGS =
+LDFLAGS = `dpkg-buildflags --get LDFLAGS`
 # Eunice users may want to use -noshare so that the executables can
 # run standalone:
 #LDFLAGS = -noshare
--- a/pbm/Makefile
+++ b/pbm/Makefile
@@ -20,6 +20,12 @@
 		pbmupc pi3topbm \
 		wbmptopbm xbmtopbm ybmtopbm	
 
+CFLAGS = `dpkg-buildflags --get CFLAGS`
+CFLAGS += `dpkg-buildflags --get CPPFLAGS`
+LDFLAGS = `dpkg-buildflags --get LDFLAGS`
+export CFLAGS
+export LDFLAGS
+
 ifneq ($(LEX)x,x)
   PORTBINARIES += thinkjettopbm
 endif
