Description: do not compare filenames not present under autopkgtest
Author: Florian Schlichting <fsfs@debian.org>
Forwarded: not-needed

--- a/t/03_no_index.t
+++ b/t/03_no_index.t
@@ -51,6 +51,9 @@
 			! $ignore{$_}
 		} $rule->in( curdir() );
 
+            SKIP: {
+                skip 'running under autopkgtest', 1 if -e 'lib/Debian/pkgperl/Foobar.pm';
+
 		is_deeply( \@files, [ qw{
 			Changes
                         META.json
@@ -58,6 +61,7 @@
 			Makefile.PL
 			lib/File/Find/Rule/Perl.pm
 		} ], 'Found the expected files' );
+            }
 	}
 }
 
