From a68d7aa59ff70aef9e4fec48daf1fffee7199711 Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <perex@perex.cz>
Date: Thu, 11 Mar 2021 15:00:20 +0100
Subject: [PATCH] Introduce Makefile to simplify the binary files availability

Signed-off-by: Jaroslav Kysela <perex@perex.cz>
---
 Makefile  | 16 ++++++++++++++++
 README.md |  3 +++
 2 files changed, 19 insertions(+)
 create mode 100644 Makefile

--- /dev/null
+++ b/Makefile
@@ -0,0 +1,16 @@
+#
+# Compile and install the firmware files to /lib/firmware tree
+# Requires the alsatplg tool (alsa-utils)
+#
+
+FWPATH ?= /lib/firmware
+
+targets = $(FWPATH)/skl_hda_dsp_generic-tplg.bin
+
+all: firmware
+
+$(FWPATH)/skl_hda_dsp_generic-tplg.bin: topology/hda-dsp/skl_hda_dsp_generic-tplg.conf
+	alsatplg -c $< -o $@
+
+.DUMMY: firmware
+firmware: $(targets)
