From: Andrew Shadura <andrew.shadura@collabora.co.uk>
Date: Thu, 7 Apr 2016 15:27:11 +0200
Subject: Update setup.py to install things into the right places.

--- a/setup.py
+++ b/setup.py
@@ -9,10 +9,8 @@
                  glob.glob('ubuntu.AdiumMessageStyle/Contents/Info.plist')),
 		('share/adium/message-styles/ubuntu.AdiumMessageStyle/Contents/Resources',
                  glob.glob('ubuntu.AdiumMessageStyle/Contents/Resources/*.*')),
-		('share/adium/message-styles/ubuntu.AdiumMessageStyle/Contents/Resources/balloons',
-                 glob.glob('ubuntu.AdiumMessageStyle/Contents/Resources/balloons/*.*')),
-		('share/adium/message-styles/ubuntu.AdiumMessageStyle/Contents/Resources/corners',
-                 glob.glob('ubuntu.AdiumMessageStyle/Contents/Resources/corners/*.*')),
+		('share/adium/message-styles/ubuntu.AdiumMessageStyle/Contents/Resources/Variants',
+                 glob.glob('ubuntu.AdiumMessageStyle/Contents/Resources/Variants/*.*')),
 		('share/adium/message-styles/ubuntu.AdiumMessageStyle/Contents/Resources/images',
                  glob.glob('ubuntu.AdiumMessageStyle/Contents/Resources/images/*.*')),
 		('share/adium/message-styles/ubuntu.AdiumMessageStyle/Contents/Resources/Incoming',
