Description: Disables protobuf nano support (missing dependency)
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/codec/pom.xml
+++ b/codec/pom.xml
@@ -119,6 +119,7 @@
             <exclude>**/marshalling/*.java</exclude>
             <exclude>**/compression/Lz4*.java</exclude>
             <exclude>**/compression/Lzma*.java</exclude>
+            <exclude>**/Protobuf*Nano.java</exclude>
           </excludes>
         </configuration>
       </plugin>
--- a/codec/src/main/java/io/netty/handler/codec/protobuf/ProtobufVarint32FrameDecoder.java
+++ b/codec/src/main/java/io/netty/handler/codec/protobuf/ProtobufVarint32FrameDecoder.java
@@ -16,7 +16,6 @@
 package io.netty.handler.codec.protobuf;
 
 import com.google.protobuf.CodedInputStream;
-import com.google.protobuf.nano.CodedInputByteBufferNano;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.ByteToMessageDecoder;
--- a/codec/src/main/java/io/netty/handler/codec/protobuf/ProtobufVarint32LengthFieldPrepender.java
+++ b/codec/src/main/java/io/netty/handler/codec/protobuf/ProtobufVarint32LengthFieldPrepender.java
@@ -16,7 +16,6 @@
 package io.netty.handler.codec.protobuf;
 
 import com.google.protobuf.CodedOutputStream;
-import com.google.protobuf.nano.CodedOutputByteBufferNano;
 import io.netty.buffer.ByteBuf;
 import io.netty.channel.ChannelHandler.Sharable;
 import io.netty.channel.ChannelHandlerContext;
