Description: Disables SubstrateVM support (missing dependencies)
Author: Emmanuel Bourg <ebourg@apache.org>
Forwarded: not-needed
--- a/common/src/main/java/io/netty/util/internal/svm/CleanerJava6Substitution.java
+++ b/common/src/main/java/io/netty/util/internal/svm/CleanerJava6Substitution.java
@@ -15,19 +15,9 @@
  */
 package io.netty.util.internal.svm;
 
-import com.oracle.svm.core.annotate.Alias;
-import com.oracle.svm.core.annotate.RecomputeFieldValue;
-import com.oracle.svm.core.annotate.TargetClass;
-
-@TargetClass(className = "io.netty.util.internal.CleanerJava6")
 final class CleanerJava6Substitution {
     private CleanerJava6Substitution() {
     }
 
-    @Alias
-    @RecomputeFieldValue(
-        kind = RecomputeFieldValue.Kind.FieldOffset,
-        declClassName = "java.nio.DirectByteBuffer",
-        name = "cleaner")
     private static long CLEANER_FIELD_OFFSET;
 }
--- a/common/src/main/java/io/netty/util/internal/svm/PlatformDependent0Substitution.java
+++ b/common/src/main/java/io/netty/util/internal/svm/PlatformDependent0Substitution.java
@@ -15,19 +15,9 @@
  */
 package io.netty.util.internal.svm;
 
-import com.oracle.svm.core.annotate.Alias;
-import com.oracle.svm.core.annotate.RecomputeFieldValue;
-import com.oracle.svm.core.annotate.TargetClass;
-
-@TargetClass(className = "io.netty.util.internal.PlatformDependent0")
 final class PlatformDependent0Substitution {
     private PlatformDependent0Substitution() {
     }
 
-    @Alias
-    @RecomputeFieldValue(
-        kind = RecomputeFieldValue.Kind.FieldOffset,
-        declClassName = "java.nio.Buffer",
-        name = "address")
     private static long ADDRESS_FIELD_OFFSET;
 }
--- a/common/src/main/java/io/netty/util/internal/svm/PlatformDependentSubstitution.java
+++ b/common/src/main/java/io/netty/util/internal/svm/PlatformDependentSubstitution.java
@@ -15,11 +15,6 @@
  */
 package io.netty.util.internal.svm;
 
-import com.oracle.svm.core.annotate.Alias;
-import com.oracle.svm.core.annotate.RecomputeFieldValue;
-import com.oracle.svm.core.annotate.TargetClass;
-
-@TargetClass(className = "io.netty.util.internal.PlatformDependent")
 final class PlatformDependentSubstitution {
     private PlatformDependentSubstitution() {
     }
@@ -31,9 +26,5 @@
      * in PlatformDependent happens during image building, the non-recomputed value
      * is cached.
      */
-    @Alias
-    @RecomputeFieldValue(
-        kind = RecomputeFieldValue.Kind.ArrayBaseOffset,
-        declClass = byte[].class)
     private static long BYTE_ARRAY_BASE_OFFSET;
 }
--- a/common/src/main/java/io/netty/util/internal/svm/UnsafeRefArrayAccessSubstitution.java
+++ b/common/src/main/java/io/netty/util/internal/svm/UnsafeRefArrayAccessSubstitution.java
@@ -15,18 +15,9 @@
  */
 package io.netty.util.internal.svm;
 
-import com.oracle.svm.core.annotate.Alias;
-import com.oracle.svm.core.annotate.RecomputeFieldValue;
-import com.oracle.svm.core.annotate.TargetClass;
-
-@TargetClass(className = "io.netty.util.internal.shaded.org.jctools.util.UnsafeRefArrayAccess")
 final class UnsafeRefArrayAccessSubstitution {
     private UnsafeRefArrayAccessSubstitution() {
     }
 
-    @Alias
-    @RecomputeFieldValue(
-        kind = RecomputeFieldValue.Kind.ArrayIndexShift,
-        declClass = Object[].class)
     public static int REF_ELEMENT_SHIFT;
 }
