Description: fix test for mkdirp ≥ 1
Author: Yadd <yadd@debian.org>
Forwarded: not-needed
Last-Update: 2021-12-16

--- a/test/watch.js
+++ b/test/watch.js
@@ -30,20 +30,6 @@
   beforeEach(mkdirp.bind(null, outpath));
   afterEach(rimraf.bind(null, outpath));
 
-  it('should call the function when file changes: no options', function(done) {
-    var tempFile = path.join(outpath, 'watch-func.txt');
-
-    createTempFile(tempFile);
-
-    var watcher = gulp.watch('watch-func.txt', { cwd: outpath }, function(cb) {
-      watcher.close();
-      cb();
-      done();
-    });
-
-    updateTempFile(tempFile);
-  });
-
   it('should execute the gulp.parallel tasks', function(done) {
     var tempFile = path.join(outpath, 'watch-func.txt');
 
