Description: fix for yargs ≥ 15
Author: Yadd <yadd@debian.org>
Forwarded: no
Last-Update: 2021-12-17

--- a/gulp-cli/index.js
+++ b/gulp-cli/index.js
@@ -58,7 +58,7 @@
   '\n' + ansi.bold('Usage:') +
   ' gulp ' + ansi.blue('[options]') + ' tasks';
 
-var parser = yargs.usage(usage, cliOptions);
+var parser = yargs.usage(usage).options(cliOptions);
 var opts = parser.argv;
 
 cli.on('require', function(name) {
