Description: Fix failing tests
--- a/index.js
+++ b/index.js
@@ -22,7 +22,11 @@
 }
 util.inherits(Gulp, Undertaker);
 
-Gulp.prototype.src = vfs.src;
+Gulp.prototype.src = function (glob, opt) {
+  opt = Object.assign({stripBOM: false}, opt);
+  return vfs.src(glob, opt);
+};
+
 Gulp.prototype.dest = vfs.dest;
 Gulp.prototype.symlink = vfs.symlink;
 Gulp.prototype.watch = function(glob, opt, task) {
