Description: No force local gulp
--- a/gulp-cli/index.js
+++ b/gulp-cli/index.js
@@ -178,7 +178,8 @@
           ? 'yarn add gulp'
         : 'npm install gulp';
     log.error(ansi.red('Try running: ' + installCommand));
-    exit(1);
+    env.modulePath = 'gulp';
+    log.error(ansi.red('Using globally installed gulp'));
   }
 
   if (!env.configPath) {
@@ -197,7 +198,9 @@
   }
 
   // Find the correct CLI version to run
-  var range = findRange(env.modulePackage.version, ranges);
+  // hard code range
+  //var range = findRange(env.modulePackage.version, ranges);
+  var range = '^4.0.0';
 
   if (!range) {
     log.error(
