Description: Launch marco with '--no-keybindings' cmdline option.
Author: Mike Gabriel <mike.gabriel@das-netzwerkteam.de>

diff --git a/src/arctica-greeter.vala b/src/arctica-greeter.vala
index 96a98ce..ea3b73e 100644
--- a/src/arctica-greeter.vala
+++ b/src/arctica-greeter.vala
@@ -842,7 +842,7 @@ public class ArcticaGreeter
             {
                 string[] argv;

-                Shell.parse_argv ("marco", out argv);
+                Shell.parse_argv ("marco --no-keybindings", out argv);
                 Process.spawn_async (null,
                                      argv,
                                      null,
