diff --git a/src/shutdown-dialog.vala b/src/shutdown-dialog.vala
index cf5e4a0..b466af1 100644
--- a/src/shutdown-dialog.vala
+++ b/src/shutdown-dialog.vala
@@ -59,6 +59,9 @@ public class ShutdownDialog : Gtk.Fixed
     private AnimateTimer animation;
     private bool closing = false;
 
+    public static string font = AGSettings.get_string (AGSettings.KEY_FONT_NAME);
+    public static string font_family = font.split_set(" ")[0];
+    public static int font_size = int.parse(font.split_set(" ")[1]);
 
     public ShutdownDialog (ShutdownDialogType type, Background bg)
     {
@@ -106,7 +109,7 @@ public class ShutdownDialog : Gtk.Fixed
         {
             var title_label = new Gtk.Label (null);
             title_label.visible = true;
-            title_label.set_markup ("<span font=\"Cantarell 15\" fgcolor=\"%s\">%s</span>".printf (AGSettings.get_string (AGSettings.KEY_TOGGLEBOX_FONT_FGCOLOR), _("Shut Down")));
+            title_label.set_markup ("<span font=\"%s %d\" fgcolor=\"%s\">%s</span>".printf (font_family, font_size+2, AGSettings.get_string (AGSettings.KEY_TOGGLEBOX_FONT_FGCOLOR), _("Shut Down")));
             title_label.set_alignment (0.0f, 0.5f);
             vbox.pack_start (title_label, false, false, 0);
 
@@ -139,7 +142,7 @@ public class ShutdownDialog : Gtk.Fixed
 
         var label = new Gtk.Label (null);
         label.set_line_wrap (true);
-        label.set_markup ("<span font=\"Cantarell 12\" fgcolor=\"%s\">%s</span>".printf (AGSettings.get_string (AGSettings.KEY_TOGGLEBOX_FONT_FGCOLOR), text));
+        label.set_markup ("<span font=\"%s %d\" fgcolor=\"%s\">%s</span>".printf (font_family, font_size, AGSettings.get_string (AGSettings.KEY_TOGGLEBOX_FONT_FGCOLOR), text));
         label.set_alignment (0.0f, 0.5f);
         label.visible = true;
         vbox.pack_start (label, false, false, 0);
@@ -544,6 +547,10 @@ private class DialogButton : Gtk.Button
     private Gtk.Image i;
     private Gtk.Label? l;
 
+    public static string font = AGSettings.get_string (AGSettings.KEY_FONT_NAME);
+    public static string font_family = font.split_set(" ")[0];
+    public static int font_size = int.parse(font.split_set(" ")[1]);
+
     public DialogButton (string inactive_filename, string focused_filename, string? active_filename, Gtk.Label? label = null)
     {
         this.inactive_filename = inactive_filename;
@@ -565,14 +572,14 @@ private class DialogButton : Gtk.Button
             try
             {
                 var font_provider = new Gtk.CssProvider ();
-                var css = "* {font-family: Cantarell; font-size: 12pt;}";
+                var css = "* {font-family: %s; font-size: %dpt;}".printf (font_family, font_size);
                 font_provider.load_from_data (css, -1);
                 style_ctx.add_provider (font_provider,
                                         Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
             }
             catch (Error e)
             {
-                debug ("Internal error loading font style (Cantarell 12pt): %s", e.message);
+                debug ("Internal error loading font style (%s, %dpt): %s", font_family, font_size, e.message);
             }
 
             l.override_color (Gtk.StateFlags.NORMAL, { 1.0f, 1.0f, 1.0f, 0.0f });
